```{r setupmutations, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```


# Mutations detection

This chapter briefly introduced the next chapters on the detection of mutations:

* Using an *in silico* virtual experiment
* Using swiss data from @Schmid-Siegert2017
* Using french data from @Plomion2018
* Using virtual data for Angela
* Using Angela sequences
* Using sequences from the *Sextonia rubra*
* Using sequences from the ["Faux de Verzy"](https://en.wikipedia.org/wiki/Faux_de_Verzy)

The three first analyses resulted in the manuscript in preparation:

* [manuscript](https://drive.google.com/file/d/1wKsOdGNDb1m1tBbbdIJYAinK1Ekrbs6p/view?usp=sharing)
* [SI](https://drive.google.com/file/d/13GgYO3wpa1iEgP-iwz11W6GGxBKLDFEe/view?usp=sharing)

## *In silico* experiment

The *in silico* experiment uses the [generateMutations](https://github.com/sylvainschmitt/generateMutations) and the [detectMutations](https://github.com/sylvainschmitt/detectMutations) workflows to test the effect of coverage/sequencing depth and allelic frequency on the performance of 7 tools to detect mutations either generalist or specific to mutations.

```{r generateMutations, fig.cap="Generate mutations workflow."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/generateMutations/main/dag/dag.minimal.svg")
```

```{r detectMutations, fig.cap="Detect mutations workflow."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/detectMutations/main/dag/dag.svg")
```

## Swiss data - @Schmid-Siegert2017

We re-analyzed mutations from @Schmid-Siegert2017 with `Strelka2` and  `GATK` to compare original and obtained mutations to show the interest of `Strelka2` and high sequencing depth.

```{r detectMutationsSwiss, fig.cap="Detect mutations workflow for Swiss data."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/detectMutations/swiss/dag/dag.minimal.svg")
```

## Bordeaux data - @Plomion2018

We re-analyzed mutations from @Plomion2018 with `Strelka2` and  `Mutect2` to compare original and obtained mutations to show the interest of `Strelka2` and high sequencing depth.

```{r detectMutationsBdx, fig.cap="Detect mutations workflow for Swiss data."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/detectMutations/bordeaux/dag/dag.minimal.svg")
```

## Angela virtual

We generated fake data corresponding to Angela architecture and sampling design, and tested our workflow ability to detect them back.

## Angela

Genome, heterozygous sites, cambial mutations and leaf mutations from sequences of Angela.

```{r detectMutationsAngela, fig.cap="Detect mutations workflow for Angela."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/detectMutations/angela/dag/dag.minimal.svg")
```

## *Sextonia rubra*

*Waiting for data.*

## Hetre

Mutations from sequences of the ["Faux de Verzy"](https://en.wikipedia.org/wiki/Faux_de_Verzy).

```{r detectMutationsVerzy, fig.cap="Detect mutations workflow for Angela."}
include_graphics("https://raw.githubusercontent.com/sylvainschmitt/detectMutations/hetre/dag/dag.svg")
```

## Project description

> For each species, we will extract DNA and construct 27 individually tagged genomic libraries to produce short read sequencing (see sampling strategy, WP2 above) with NovaSeq6000 SP technology, targeting aminimum  of  50x  coverage  per  library  (70-90x  should  be  realistic,  we  will  go  for  100x  if  possible).  The possible sequencing depth affordable in the project will depend on the genome size of chosen species and  some  extent  on  fluctuating  sequencing  prices.  Lab  work  will  be  conducted  at  Ecofog  and  PGTB (Biogeco); the sequencing will be subcontracted to CEA Genoscope, Evry.

> Detection of mutations (T4.2): For mutation detection, we will  minimise the impact of library preparation and  sequencing  errors  by  retaining  only  mutations  detected  by  comparison  with  the “zero  mutation reference” (Hanlon et al 2019) and in each branch (Orr et al. 2019), giving a total of 9 genotype sampling points in each tree. We will quantify the occurrence of mutations in each of the 24 leaf libraries per tree using a high-sensitivity method (Cibulskis et al., 2013). De novomutations appear in single meristem cells and  their  abundance  in  tree  tissues  will  depend  on  the  specific  cell’s  divisions  and  contribution  to generating tree tissues. Fixation of de novo mutations in tissues is expected to be rare and the expected time  to  fixation  is  long,  see (Nicholson  et  al.,  2018)for  an  example  in  humans.  The  detection  of  novel mutations must thus be sensitive to low allele copy number in tissues. For this reason, we will test and use mutation detection methods typically used in cancer mutation research, that are optimized for strong allelic frequency skews (Alioto et al., 2015).The artificial inclusion of in silicomutationsinto the reads and permutations among branch labels allows the estimation of the recovery rate of true mutations, the false negative rates, and the false positive rates (Orr et al., 2019). We will test whether the rate of accumulation of  mutations  is  increased  in  high-light-exposed  branches  compared  to  those  with  low-light  exposure accounting  for  the  physical  tree  branches  (tree  architecture -WP3),  the  mutation  history,  and  light environment (WP3).
