```{r setupmutationsangela, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(circlize)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r renameAngela, eval=F, echo=F}
sampleRef <- read_delim("data/mutations/angela/sampleRef.txt", delim = " ", 
                        col_names = c("idGeno", "id")) %>% 
  mutate(idGeno = gsub("DBR_", "", idGeno)) %>% 
  separate(id, c("sample", "leaf", "replicate"), remove = F)
libraries <- read_table("data/mutations/angela/raw_libraries.txt", col_names = "files") %>% 
  separate(files, c("idGeno", "file"), "/")
ids <- read_tsv("data/mutations/angela/idBordeaux.tsv") %>% 
  filter(EspeceBotanique == "Dicorynia guianensis") %>% 
  filter(Tissu != "graines") %>% 
  dplyr::select(id, Branch, Replicate, Lumiere) %>% 
  rename(sample = id, branchOld = Branch, light = Lumiere) %>% 
  mutate(branchOld = ifelse(is.na(branchOld), Replicate, branchOld)) %>% 
  dplyr::select(-Replicate) %>% 
  mutate(light = recode(light, "soleil" = "L", "ombre" = "S")) %>% 
    mutate(light = ifelse(is.na(light), "", light)) %>% 
  mutate(branch = recode(branchOld, "C1" = "T1", "C Sud" = "T3", "C Nord Est" = "T2", "C Nord Ouest" = "T4", 
                         "B31" = "A", "B32" = "B", "B2" = "C", "B4" = "D", "B1" = "E")) 
ids %>% 
  full_join(sampleRef) %>% 
  full_join(libraries) %>% 
  rename(sampleOld = sample, fileOld = file, idOld = id) %>% 
  mutate(leaf = ifelse(light == "", "", leaf)) %>% 
  mutate(sample = paste0(branch, light, leaf)) %>% 
  group_by(sample) %>% 
  mutate(library = paste0(sample, "_", 1:2)) %>% 
  mutate(file = paste0(library, ".fq.gz")) %>% 
  dplyr::select(sample, branch, light, leaf, library, file, branchOld, sampleOld, idOld, idGeno, fileOld) %>% 
  write_tsv("data/mutations/angela/samples.tsv")

read_tsv("data/mutations/angela/samples.tsv") %>% 
  mutate(command = paste("mv", fileOld, file)) %>% 
  dplyr::select(command) %>% 
  write_tsv("data/mutations/angela/rename.sh", col_names = F)

read_tsv("data/mutations/angela/samples.tsv") %>% 
  mutate(command = paste("touch", file)) %>% 
  dplyr::select(command) %>% 
  write_tsv("data/mutations/angela/dummy.sh", col_names = F)
```

# Mutations Angela

This chapter describes the analyses of Angela mutations currently done in the the [`angela` branch of the `detectMutations` repository](https://github.com/sylvainschmitt/detectMutations/tree/angela).

## Processing

**Results below are preliminary.**

To produce robust results we will proceed as follow:

* Compute BUSCO and coverage of DS2 on HS1 & HS2
* Choose the best reference
* Align if needed again the libraries on the best reference excluding T1
* Detect all heterozygous sites by joint-genotyping allowing a maximum of 1 missing library
* Detect all raw mutations in T2, T3, T4 with all cross comparisons
* Detect all raw mutations in all leaves compared to T2
* Remove all heterozygous sites from raw mutations
* Remove all raw cambium mutations from leaf raw mutations without heterozygous sites
* Test independently each filter on each cambium and leaf library
* Obtain a rboust dataset
* Explore sahred mutations across leaves, tips and branches

## Quality check

The results of multiQC are available for the haplotypes HS1 [here](data/mutations/angela/multiqc_report_HS1.html) and HS2 [here](data/mutations/angela/multiqc_report_HS2.html).

## Genome

```{r AngelaGenomeData}
hs1 <- read_tsv("data/mutations/angela/Dgu_HS1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(name, length) %>% 
  mutate(name = gsub("Super-Scaffold_", "", name)) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
hs2 <- read_tsv("data/mutations/angela/Dgu_HS2_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(name, length) %>% 
  mutate(name = gsub("Super-Scaffold_", "", name)) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
gc_hs1 <- vroom::vroom("data/mutations/angela/Dgu_HS1_HYBRID_SCAFFOLD.gc", skip = 1,
             col_names = c("chrom", "start", "stop", "AT", "GC", "A", "C", "G", "T", "N", "other", "length")) %>% 
  mutate(pos = start + 500) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
gc_hs2 <- vroom::vroom("data/mutations/angela/Dgu_HS2_HYBRID_SCAFFOLD.gc", skip = 1,
             col_names = c("chrom", "start", "stop", "AT", "GC", "A", "C", "G", "T", "N", "other", "length")) %>% 
  mutate(pos = start + 500) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
cov_hs1 <- vroom::vroom("data/mutations/angela/DS2_HS1.regions.bed", col_names = c("chrom", "start", "stop", "coverage")) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
cov_hs2 <- vroom::vroom("data/mutations/angela/DS2_HS2.regions.bed", col_names = c("chrom", "start", "stop", "coverage")) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
busco_hs1 <- read_tsv("data/mutations/angela/short_summary.specific.viridiplantae_odb10.Dgu_HS1_HYBRID_SCAFFOLD_busco.txt",
         skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1)
busco_hs2 <- read_tsv("data/mutations/angela/short_summary.specific.viridiplantae_odb10.Dgu_HS2_HYBRID_SCAFFOLD_busco.txt",
         skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1)
```

```{r AngelaGenomFig, fig.cap="Anchored scaffolds size-distribution for haplotypes HS1 and HS2 in Angela's genome."}
cowplot::plot_grid(
  plotlist = 
    lapply(list(HS1 = hs1, HS2 = hs2), function(genome)
      ggplot(genome, aes(reorder(name, end), end/10^6, fill = end > 2*10^6)) +
        geom_col() +
        coord_flip() +
        ylab("Length (mb)") + 
        xlab("Anchored super-scaffold") + 
        scale_fill_discrete(">2Mb")),
  labels = c("HS1", "HS2")
)
```

```{r AngelaBusco, fig.cap="BUSCO results for haplotypes HS1 and HS2."}
list(HS1 = busco_hs1, HS2 = busco_hs2) %>% 
  bind_rows(.id = "haploytpe") %>% 
  mutate(Type = factor(Type, levels= c("Complete and single-copy BUSCOs (S)",
                                       'Complete and duplicated BUSCOs (D)',
                                       "Fragmented BUSCOs (F)",
                                       "Missing BUSCOs (M)"))) %>% 
  ggplot(aes(haploytpe, N, fill = Type)) +
  geom_col() +
  geom_text(aes(y = N, label = N), col = "white", position = position_stack(vjust = .5)) +
  scale_y_sqrt() +
  theme(axis.title.x = element_blank())
```

```{r circosHS1, eval=F}
hs1_f <- filter(hs1, end > 2*10^6)
gc_hs1_f <- filter(gc_hs1, chrom %in% hs1_f$name)
cov_hs1_f <- filter(cov_hs1, chrom %in% hs1_f$name) %>% 
  mutate(coverage = ifelse(coverage > 500, 500, coverage))
hs2_f <- filter(hs2, end > 2*10^6)
gc_hs2_f <- filter(gc_hs2, chrom %in% hs2_f$name)
cov_hs2_f <- filter(cov_hs2, chrom %in% hs2_f$name) %>% 
  mutate(coverage = ifelse(coverage > 500, 500, coverage))
png("save/AngelaCircosHS1CovGC.png", width = 1000, height = 1000)
circos.initializeWithIdeogram(as.data.frame(hs1_f))
circos.genomicTrack(cov_hs1_f, 
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "blue", ...)
})

circos.genomicTrack(select(gc_hs1_f, chrom, start, stop, GC), 
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "red", ...)
})
text(0, 0, "HS1", cex = 1.5)
legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("coverage", "GC"), bty = "n")
dev.off()
png("save/AngelaCircosHS2CovGC.png", width = 1000, height = 1000)
circos.initializeWithIdeogram(as.data.frame(hs2_f))
circos.genomicTrack(cov_hs2_f,
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "blue", ...)
})

circos.genomicTrack(select(gc_hs2, chrom, start, stop, GC),
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "red", ...)
})
text(0, 0, "HS2", cex = 1.5)
legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("coverage", "GC"), bty = "n")
dev.off()
```

```{r circosHS1g, fig.cap="Genome coverage and GC-content on a 10-kb windows for the HS1 haplotype (scaffolds > 2-Mb)."}
include_graphics("save/AngelaCircosHS1CovGC.png")
```

```{r circosHS2g, fig.cap="Genome coverage and GC-content on a 10-kb windows for the HS2 haplotype (scaffolds > 2-Mb)."}
include_graphics("save/AngelaCircosHS2CovGC.png")
```

## Trunk

```{r, cache=FALSE}
trunk <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/angela/trunk_raw_mutations.sql") %>% tbl("mutations")
```

```{r, eval=F}
t <- trunk %>% 
  dplyr::select(tumor, CHROM, POS) %>% 
  collect() %>% 
  group_by(tumor) %>% 
  unique() %>% 
  summarise(N = n())
write_tsv(t, "data/mutations/angela/trunk_raw_mutations.stats")
```

```{r nmutTrunkRawTab}
read_tsv("data/mutations/angela/trunk_raw_mutations.stats") %>% 
  kable(caption = "Number of raw mutations per cambium sample.", format.args = list(big.mark = " "))
```

We filtered mutations with a robust filters, adjusting the coverage to the obtained variation of coverages all libraries:

* A read depth for the two sample between half and two times the mean coverage (`normal_DP <= 360, normal_DP >= 50, mutation_DP <= 360, mutation_DP >= 50`)
* A null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
* A minimum of 10 alternate allele count in the mutated sample (`mutation_altCountT1 >= 10`)
* An allelic frequency  inferior to 0.5 (`tumor_AF <= 0.5`)
* `Strelka 2` automatic filtering (`Filter == PASS`)

```{r mutTrunkFiltered}
trunk_mutations <- trunk %>% 
  filter(normal_DP <= 360, normal_DP >= 50) %>%
  filter(mutation_DP <= 360, mutation_DP >= 50) %>% 
  filter(normal_altCountT1 == 0) %>% 
  filter(mutation_altCountT1 >= 10) %>% 
  filter(mutation_AF <= 0.5) %>% 
  filter(FILTER == "PASS") %>% 
  collect() %>% 
  group_by(tumor, CHROM, POS, REF, ALT) %>% 
  summarise_at("mutation_AF", mean) %>% 
  mutate(SNV = paste0(CHROM, "_", "Pos", as.integer(POS)))
```

```{r nmutTrunkFiltTab}
trunk_mutations %>% 
  group_by(tumor) %>% 
  summarise(N = n()) %>% 
  kable(caption = "Number of filtered mutations per cambium.")
```


The third sample (South) show a intriguing increase in low-frequency mutations.

```{r afTrunkFiltFig, fig.cap="Alleles-frequencies of filtered mutations in cambium samples."}
trunk_mutations %>% 
  ggplot(aes(mutation_AF, col = tumor), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency")  
```

Only four mutations are shared by the three cambium samples.

```{r vennTrunkFiltFig, fig.cap="Filtered mutations sharing across cambium samples."}
ggvenn::ggvenn(lapply(split(trunk_mutations, trunk_mutations$tumor), `[[`, "SNV"), show_percentage = F)
```

```{r genomeTrunkFiltFig, fig.cap="Filtered mutations of cambium samples across the genome (scf>1Mb)."}
ggplot(filter(genome, length > 1*10^6), aes(x = CHROM, xend = CHROM)) +
  geom_segment(aes(y = start, yend = stop), size = 3, col = "lightgrey") +
  geom_point(aes(y = POS/10^6, col = tumor, size = (tumor == "T2_T3_T4")),
             data = trunk_mutations %>% 
               group_by(SNV, CHROM, POS) %>% 
               arrange(SNV) %>% 
               summarise(tumor = paste0(tumor, collapse = "_"))) +
  coord_flip() +
  ylab("Position (Mb)") +
  theme(axis.line.y = element_blank(), axis.title.y = element_blank(), axis.ticks.y = element_blank()) +
  scale_size_manual(guide = "none", values = c(1,4))
```

## Leaf

```{r, cache=FALSE}
leaf <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/angela/leaf_nontrunk_mutations.sql") %>% tbl("mutations")
```

```{r, eval=F}
t <- leaf %>% 
  dplyr::select(tumor, CHROM, POS) %>% 
  collect() %>% 
  group_by(tumor) %>% 
  unique() %>% 
  summarise(N = n())
write_tsv(t, "data/mutations/angela/leaf_nontrunk_mutations.stats")
```


```{r nmutLeafRawTab}
read_tsv("data/mutations/angela/leaf_nontrunk_mutations.stats") %>% 
  mutate(branch = str_sub(tumor, 1, 1), leaf = str_sub(tumor, 2, 3)) %>% 
  reshape2::dcast(branch ~ leaf, value.var = "N") %>% 
  kable(caption = "Number of raw mutations per leaf sample.", format.args = list(big.mark = " "))
```

We filtered mutations with a robust filters, adjusting the coverage to the obtained variation of coverages all libraries:

* A read depth for the two sample between half and two times the mean coverage (`normal_DP <= 360, normal_DP >= 50, mutation_DP <= 360, mutation_DP >= 50`)
* A null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
* A minimum of 10 alternate allele count in the mutated sample (`mutation_altCountT1 >= 10`)
* An allelic frequency  inferior to 0.5 (`tumor_AF <= 0.5`)
* `Strelka 2` automatic filtering (`Filter == PASS`)


```{r mutLeafFiltered}
leaf_mutations <- leaf %>% 
  filter(normal_DP <= 360, normal_DP >= 50) %>%
  filter(mutation_DP <= 360, mutation_DP >= 50) %>% 
  filter(normal_altCountT1 == 0) %>% 
  filter(mutation_altCountT1 >= 10) %>% 
  filter(mutation_AF <= 0.5) %>% 
  filter(FILTER == "PASS") %>% 
  collect() %>% 
  group_by(tumor, CHROM, POS, REF, ALT) %>% 
  summarise_at("mutation_AF", mean) %>% 
  mutate(branch = str_sub(tumor, 1, 1), tip = str_sub(tumor, 2, 2), leaf = str_sub(tumor, 3, 3)) %>% 
  mutate(SNV = paste0(CHROM, "_", "Pos", as.integer(POS)))
```

```{r nmutLeafFiltTab}
leaf_mutations %>% 
  group_by(tumor, branch, tip, leaf) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(branch ~ tip + leaf, value.var = "N") %>% 
  kable(caption = "Number of filtered mutations per leaf.")
```


```{r afLeafFiltFig, fig.cap="Alleles-frequencies of filtered mutations in leaf samples."}
leaf_mutations %>% 
  ggplot(aes(mutation_AF, col = tumor), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency")  
```

Most of mutations are shared by biological replicates.

```{r vennLeafFiltFig, fig.cap="Filtered mutations sharing across leaf samples within tips.", fig.height=8}
g1 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(leaf_mutations, branch == B, tip == "S"),
                              filter(leaf_mutations, branch == B, tip == "S")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
g2 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(leaf_mutations, branch == B, tip == "L"),
                              filter(leaf_mutations, branch == B, tip == "L")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
cowplot::plot_grid(plotlist = c(g1,g2))
```

## Tip

We further filtered mutations shared by all leaves in each tip.

```{r mutTipFiltered}
tip_mutations <- leaf_mutations %>% 
  group_by(SNV, CHROM, POS, REF, ALT) %>% 
  filter(n() == 3) %>% 
  dplyr::select(-leaf) %>% 
  mutate(tumor = paste0(branch, tip)) %>% 
  group_by(SNV, CHROM, POS, REF, ALT, tumor, branch, tip) %>% 
  summarise(mutation_AF = mean(mutation_AF))
write_tsv(tip_mutations, "data/mutations/angela/tip_mutations.tsv")
```

```{r nmutTipFiltTab}
tip_mutations %>%
  group_by(tumor, branch, tip) %>%
  summarise(N = n()) %>%
  reshape2::dcast(branch ~ tip, value.var = "N") %>%
  kable(caption = "Number of shared mutations per tip.")
```

Interestingly, shared mutations have higher allele frequency, increasing their chance to be transmitted to all cells in the meristem. But few mutations with low allele frequency are still shared across leaves !

```{r afTipFiltFig, fig.cap="Mean alleles-frequencies of filtered mutations shared in each tip."}
tip_mutations %>%
  ggplot(aes(mutation_AF, col = tumor), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency")
```

Only a few mutations are shared by tips within a branch.

```{r vennTipFiltFig, fig.cap="Sharing of filtered mutations shared in each tip across branches."}
g <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(tip_mutations, branch == B),
                            filter(tip_mutations, branch == B)$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
cowplot::plot_grid(plotlist = g)
```

## Branch

We further filtered mutations shared by all tips in each branch.

```{r mutBranchFiltered}
branch_mutations <- tip_mutations %>% 
  group_by(SNV, CHROM, POS, REF, ALT) %>% 
  filter(n() == 2) %>% 
  dplyr::select(-tip) %>% 
  group_by(SNV, CHROM, POS, REF, ALT, branch) %>% 
  summarise(mutation_AF = mean(mutation_AF))
```

```{r nmutBranchFiltTab}
branch_mutations %>%
  group_by(branch) %>%
  summarise(N = n()) %>% 
  kable(caption = "Number of shared mutations per tip.")
```

```{r afBranchFiltFig, fig.cap="Mean alleles-frequencies of filtered mutations shared in each tip."}
branch_mutations %>%
  ggplot(aes(mutation_AF, fill = branch), fill = NA) +
  geom_histogram(position = "dodge") +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency")
```
Interestingly the only pairs of branches sharing mutations A-B and C-E are architecturally close. 

```{r vennBranchFiltFig, fig.cap="Sharing of filtered mutations shared in each branch in the whole tree."}
ggvenn::ggvenn(lapply(split(branch_mutations,
                            branch_mutations$branch),
                        `[[`, "SNV"), show_percentage = F)
```

## Summary

In summary, each leaf sample contained approximately 250,000 candidate mutations, resulting in 58 to 126 robust mutations with both high and low allelic frequencies. Most robust mutations were shared by leaves at the branch tips, resulting in 19 to 57 mutations shared by leaves at the tip of a branch with high average allelic frequency in all samples (but some low-frequency mutations persisted). Only 2 to 6 mutations were shared between the two tips of a branch with allelic frequency diversity. And of these, only 7 were shared within a pair of branches: 2 between A and B and 5 between C and E, highlighting the greater architectural proximity of these pairs of branches in the tree.
