```{r setupmutationsangela, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(circlize)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r renameAngela, eval=F, echo=F}
sampleRef <- read_delim("data/mutations/angela/prep/sampleRef.txt", delim = " ", 
                        col_names = c("idGeno", "id")) %>% 
  mutate(idGeno = gsub("DBR_", "", idGeno)) %>% 
  separate(id, c("sample", "leaf", "replicate"), remove = F)
libraries <- read_table("data/mutations/angela/prep/raw_libraries.txt", col_names = "files") %>% 
  separate(files, c("idGeno", "file"), "/")
ids <- read_tsv("data/mutations/angela/prep/idBordeaux.tsv") %>% 
  filter(EspeceBotanique == "Dicorynia guianensis") %>% 
  filter(Tissu != "graines") %>% 
  dplyr::select(id, Branch, Replicate, Lumiere) %>% 
  rename(sample = id, branchOld = Branch, light = Lumiere) %>% 
  mutate(branchOld = ifelse(is.na(branchOld), Replicate, branchOld)) %>% 
  dplyr::select(-Replicate) %>% 
  mutate(light = recode(light, "soleil" = "L", "ombre" = "S")) %>% 
    mutate(light = ifelse(is.na(light), "", light)) %>% 
  mutate(branch = recode(branchOld, "C1" = "T1", "C Sud" = "T3", "C Nord Est" = "T2", "C Nord Ouest" = "T4", 
                         "B31" = "A", "B32" = "B", "B2" = "C", "B4" = "D", "B1" = "E")) 
ids %>% 
  full_join(sampleRef) %>% 
  full_join(libraries) %>% 
  rename(sampleOld = sample, fileOld = file, idOld = id) %>% 
  mutate(leaf = ifelse(light == "", "", leaf)) %>% 
  mutate(sample = paste0(branch, light, leaf)) %>% 
  group_by(sample) %>% 
  mutate(library = paste0(sample, "_", 1:2)) %>% 
  mutate(file = paste0(library, ".fq.gz")) %>% 
  dplyr::select(sample, branch, light, leaf, library, file, branchOld, sampleOld, idOld, idGeno, fileOld) %>% 
  write_tsv("data/mutations/angela/prep/samples.tsv")

read_tsv("data/mutations/angela/prep/samples.tsv") %>% 
  mutate(command = paste("mv", fileOld, file)) %>% 
  dplyr::select(command) %>% 
  write_tsv("data/mutations/angela/prep/rename.sh", col_names = F)

read_tsv("data/mutations/angela/prep/samples.tsv") %>% 
  mutate(command = paste("touch", file)) %>% 
  dplyr::select(command) %>% 
  write_tsv("data/mutations/angela/prep/dummy.sh", col_names = F)
```

# Mutations Angela

This chapter describes the analyses of Angela mutations currently done in the the [`angela` branch of the `detectMutations` repository](https://github.com/sylvainschmitt/detectMutations/tree/angela).

## Quality check

The results of multiQC are available for the haplotypes HS1 [here](data/mutations/angela/qc/multiqc_report_HS1.html) and HS2 [here](data/mutations/angela/qc/multiqc_report_HS2.html). They globally all gave a green light for subsequent analyses.

## Genome

We compared the two haplophases and chose HS1 for contiguity and length (Fig. \@ref(fig:AngelaGenomFig) and Fig. \@ref(fig:AngelaBusco)) as well as mean coverage and homogeneity (Fig. \@ref(fig:circosHS1g)).

```{r AngelaGenomeData}
hs1 <- read_tsv("data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(name, length) %>% 
  mutate(name = gsub("Super-Scaffold_", "", name)) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
hs2 <- read_tsv("data/mutations/angela/genome/Dgu_HS2_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(name, length) %>% 
  mutate(name = gsub("Super-Scaffold_", "", name)) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
gc_hs1 <- vroom::vroom("data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.gc", skip = 1,
             col_names = c("chrom", "start", "stop", "AT", "GC", "A", "C", "G", "T", "N", "other", "length")) %>% 
  mutate(pos = start + 500) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
gc_hs2 <- vroom::vroom("data/mutations/angela/genome/Dgu_HS2_HYBRID_SCAFFOLD.gc", skip = 1,
             col_names = c("chrom", "start", "stop", "AT", "GC", "A", "C", "G", "T", "N", "other", "length")) %>% 
  mutate(pos = start + 500) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
cov_hs1 <- vroom::vroom("data/mutations/angela/genome/DS2_HS1.regions.bed", col_names = c("chrom", "start", "stop", "coverage")) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
cov_hs2 <- vroom::vroom("data/mutations/angela/genome/DS2_HS2.regions.bed", col_names = c("chrom", "start", "stop", "coverage")) %>% 
  mutate(chrom = gsub("Super-Scaffold_", "", chrom))
busco_hs1 <- read_tsv("data/mutations/angela/genome/short_summary.specific.viridiplantae_odb10.Dgu_HS1_HYBRID_SCAFFOLD_busco.txt",
         skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1)
busco_hs2 <- read_tsv("data/mutations/angela/genome/short_summary.specific.viridiplantae_odb10.Dgu_HS2_HYBRID_SCAFFOLD_busco.txt",
         skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1)
covs_files <- list.files("data/mutations/angela/genome/dists/", full.names = T)
names(covs_files) <- list.files("data/mutations/angela/genome/dists/")
covs <- lapply(covs_files, read_tsv, col_names = c("chromosome", "coverage", "proportion"), cols(
  chromosome = col_character(),
  coverage = col_double(),
  proportion = col_double()
)) %>% 
  bind_rows(.id = "file") %>% 
  mutate(library = gsub(".mosdepth.global.dist.txt", "", file)) %>% 
  filter(chromosome == "total")
cov_lims <- covs %>% 
  filter(proportion == 0.1 | proportion == 0.9 | proportion == 0.5) %>% 
  group_by(library, proportion) %>% 
  summarise(coverage = mean(coverage)) %>% 
  mutate(proportion = recode(as.character(proportion), "0.1" = "high_cov", "0.9" = "low_cov", "0.5" = "med_cov")) %>% 
  reshape2::dcast(library ~ proportion)
```

```{r AngelaGenomFig, fig.cap="Anchored scaffolds size-distribution for haplotypes HS1 and HS2 in Angela's genome."}
cowplot::plot_grid(
  plotlist = 
    lapply(list(HS1 = hs1, HS2 = hs2), function(genome)
      ggplot(genome, aes(reorder(name, end), end/10^6, fill = end > 2*10^6)) +
        geom_col() +
        coord_flip() +
        ylab("Length (mb)") + 
        xlab("Anchored super-scaffold") + 
        scale_fill_discrete(">2Mb")),
  labels = c("HS1", "HS2")
)
```

```{r AngelaBusco, fig.cap="BUSCO results for haplotypes HS1 and HS2."}
list(HS1 = busco_hs1, HS2 = busco_hs2) %>% 
  bind_rows(.id = "haploytpe") %>% 
  mutate(Type = factor(Type, levels= c("Complete and single-copy BUSCOs (S)",
                                       'Complete and duplicated BUSCOs (D)',
                                       "Fragmented BUSCOs (F)",
                                       "Missing BUSCOs (M)"))) %>% 
  ggplot(aes(haploytpe, N, fill = Type)) +
  geom_col() +
  geom_text(aes(y = N, label = N), col = "white", position = position_stack(vjust = .5)) +
  scale_y_sqrt() +
  theme(axis.title.x = element_blank())
```

```{r AngelaCovs, fig.cap="Coverage distribution. Distribution of the number of locations in the reference genome with a given depth of coverage."}
ggplot(covs, aes(coverage, proportion*100, col = library)) +
  geom_line() +
  xlim(0, 500) +
  geom_hline(yintercept = 10, col = "red", linetype = "dashed") +
  geom_hline(yintercept = 90, col = "red", linetype = "dashed") +
  scale_color_discrete("") + 
  xlab("Coverage (X)") +
  ylab("% of base in the genome with at least X reads")
```

```{r circosHS1, eval=F}
hs1_f <- filter(hs1, end > 2*10^6)
gc_hs1_f <- filter(gc_hs1, chrom %in% hs1_f$name)
cov_hs1_f <- filter(cov_hs1, chrom %in% hs1_f$name) %>% 
  mutate(coverage = ifelse(coverage > 500, 500, coverage))
hs2_f <- filter(hs2, end > 2*10^6)
gc_hs2_f <- filter(gc_hs2, chrom %in% hs2_f$name)
cov_hs2_f <- filter(cov_hs2, chrom %in% hs2_f$name) %>% 
  mutate(coverage = ifelse(coverage > 500, 500, coverage))
png("save/AngelaCircosHS1CovGC.png", width = 1000, height = 1000)
circos.initializeWithIdeogram(as.data.frame(hs1_f))
circos.genomicTrack(cov_hs1_f, 
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "blue", ...)
})

circos.genomicTrack(select(gc_hs1_f, chrom, start, stop, GC), 
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "red", ...)
})
text(0, 0, "HS1", cex = 1.5)
legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("coverage", "GC"), bty = "n")
dev.off()
png("save/AngelaCircosHS2CovGC.png", width = 1000, height = 1000)
circos.initializeWithIdeogram(as.data.frame(hs2_f))
circos.genomicTrack(cov_hs2_f,
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "blue", ...)
})

circos.genomicTrack(select(gc_hs2, chrom, start, stop, GC),
    panel.fun = function(region, value, ...) {
        circos.genomicLines(region, value, col = "red", ...)
})
text(0, 0, "HS2", cex = 1.5)
legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("coverage", "GC"), bty = "n")
dev.off()
```

```{r circosHS1g, fig.cap="Genome coverage and GC-content on a 10-kb windows for the HS1 haplotype (scaffolds > 2-Mb)."}
include_graphics("save/AngelaCircosHS1CovGC.png")
```

```{r circosHS2g, fig.cap="Genome coverage and GC-content on a 10-kb windows for the HS2 haplotype (scaffolds > 2-Mb)."}
include_graphics("save/AngelaCircosHS2CovGC.png")
```

## Heterozygosity

We used k-mer analyses (21-mers) using `jellyfish` and the `GenomeScope` to estimate genetic diversity $\pi=0.901%$ (Fig. \@ref(fig:genomescopeAngela)).
**Should I also estimate non-synonymous diversity with ORF annotation using `transdecoder` and `SNPdat` (https://phdpages.netlify.app/6b189d4142e8224bff99e28abd11cbfdd50c51b1/symcapture/annotation.html#synonymy)?**
We detected and filtered SNPs as follow:

* Raw=5M : raw result from `GATK HaplotypeCaller + GenomicsDBImport + GenotypeGVCFs`
* Biallelic=5.3M : biallelic sites with `bcftools`
* SNP=4.7M : SNPs with `GATK`
* Filtered=3.85M : SNPs with QUAL < 30, QD < 2, FS > 60, SOR > 3 using `GATK`
* Non-missing=3.84M : SNPs in all genotypes and individuals with `plink`
* Shared=3.84M : shared SNPs by at least 32 out of 33 individuals with  `bcftools`

```{r genomescopeAngela, fig.cap="GenomeScope Profile. Full results here: http://genomescope.org/analysis.php?code=ayEfgONEP1cFWw2yjhfb."}
knitr::include_graphics("data/mutations/angela/hz/genomescope.png")
```

## Mutations

### Raw

We obtained between 292 and 407k candidates per library.

```{r, cache=FALSE}
trunk <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/angela/raw/cambium_nonhz_mutations.sql") %>% tbl("mutations")
leaf <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/angela/raw/leaf_nonhz_mutations.sql") %>% tbl("mutations")
```

```{r, eval=F}
t <- bind_rows(collect(trunk), collect(leaf)) %>% 
  dplyr::select(tumor, CHROM, POS) %>%
  group_by(tumor) %>% 
  unique() %>% 
  summarise(N = n())
write_tsv(t, "data/mutations/angela/raw/raw_mutations.stats")
```

```{r nmutRawTab}
read_tsv("data/mutations/angela/raw/raw_mutations.stats") %>% 
  mutate(sample = str_match_all(tumor, "[A-Z]+"), repetition = str_match_all(tumor, "[0-9]+")) %>% 
  unnest() %>% 
  mutate(repetition = as.numeric(repetition)) %>% 
  mutate(repetition = ifelse(sample == "T", repetition - 1, repetition)) %>% 
  reshape2::dcast(sample ~ repetition, value.var = "N") %>% 
  kable(caption = "Number of raw mutations per library.", format.args = list(big.mark = " "))
```

### Filters

We tested independantly the effect of 4 filters and look their effect on (1) the percentage kept, (2) the distribution of allelic frequencies, and (3) the overlap between filters:

1. **NAC**: a null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
1. **DP**: a read depth for the two sample between the 10th quantile and the 90th quantile of the coverage of the mutated library (`normal_DP <= high_cov, normal_DP >= low_cov, mutation_DP <= high_cov, mutation_DP >= low_cov`)
1. **MAC**: a minimum of 10 alternate allele count in the mutated sample (`mutation_altCountT1 >= 10`), corresponding to a minimum allele frequency of 0.1 for a mean coverage of 100X
1. **EVS**: `Strelka 2` automatic filtering based on the empiric variant score (`Filter == "PASS"`)

**EVS** is the most stringent filter (Fig. \@ref(fig:pctmutfilter), 
but only **MAC** filter changes the allele frequencies distribution (Fig. \@ref(fig:afmutfilter).
Most filters share mutations, except **EVS** that rejected 36 813 mutations shared by the three others (Fig. \@ref(fig:vennmutfilter).
And all filters except **MAC** are not sensitive to the library coverage (Fig. \@ref(fig:nmutcovfilter).
We will thus use two filters for next steps:

1. **base**: 
    * **NAC**: a null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
    * **DP**: a read depth for the two sample between the 10th quantile and the 90th quantile of the coverage of the mutated library (`normal_DP <= high_cov, normal_DP >= low_cov, mutation_DP <= high_cov, mutation_DP >= low_cov`)
    * **MAC**: a minimum of 10 alternate allele count in the mutated sample (`mutation_altCountT1 >= 10`), corresponding to a minimum allele frequency of 0.1 for a mean coverage of 100X
2. **evs**: 
    * **base**
    * **EVS**: `Strelka 2` automatic filtering based on the empiric variant score (`Filter == "PASS"`)

```{r, eval=F}
leaf_filters <- leaf %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  dplyr::select(tumor, SNV, normal_altCountT1, normal_DP, mutation_DP, mutation_altCountT1, FILTER, mutation_AF) %>% 
  collect()
trunk_filters <- trunk %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  dplyr::select(tumor, SNV, normal_altCountT1, normal_DP, mutation_DP, mutation_altCountT1, FILTER, mutation_AF) %>% 
  collect()
bind_rows(leaf_filters, trunk_filters) %>% 
  left_join(rename(cov_lims, tumor = library)) %>% 
  mutate(NAC = (normal_altCountT1 == 0)) %>% 
  mutate(DP = (normal_DP <= high_cov &
                         normal_DP >= low_cov &
                         mutation_DP <= high_cov &
                         mutation_DP >= low_cov)) %>% 
  mutate(MAC = (mutation_altCountT1 >= 10)) %>% 
  mutate(EVS = (FILTER == "PASS")) %>% 
  select(tumor, SNV, mutation_AF, NAC, DP, MAC, EVS) %>% 
  reshape2::melt(c("tumor", "SNV", "mutation_AF"), variable.name = "filter") %>% 
  vroom::vroom_write("data/mutations/angela/filters/filters_mutations.stats")
vroom::vroom("data/mutations/angela/filters/filters_mutations.stats") %>% 
  group_by(tumor, filter, value) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(tumor + filter ~ value, value.var = 'N') %>% 
  mutate(percentage = `TRUE`/(`TRUE`+`FALSE`)*100) %>% 
  vroom::vroom_write("data/mutations/angela/filters/filters_mutations.pct")
g <- vroom::vroom("data/mutations/angela/filters/filters_mutations.stats") %>% 
  ggplot(aes(mutation_AF, col = value)) +
  geom_density() +
  scale_x_sqrt() +
  facet_wrap(~ filter) +
  xlab("Mutation allelic frequency")
ggsave(g, filename = "data/mutations/angelafilters//filters_mutations_af.png", width = 4, height = 4, bg = "white")
t <- vroom::vroom("data/mutations/angela/filters/filters_mutations.stats") %>% 
  filter(value == TRUE) %>%
  dplyr::select(-mutation_AF, -tumor, -value) %>% 
  unique() 
g <- ggvenn::ggvenn(lapply(split(t, t$filter), `[[`, "SNV"), show_percentage = F)
ggsave(g, filename = "data/mutations/angela/filters/filters_mutations_venn.png", bg = "white")
```

```{r pctmutfilter, fig.cap="Percentage of kept mutations per filter."}
vroom::vroom("data/mutations/angela/filters/filters_mutations.pct") %>% 
  ggplot(aes(filter, percentage)) +
  geom_boxplot() +
  geom_jitter(aes(col = tumor)) +
  scale_y_log10() +
  scale_color_discrete("") +
  xlab("Filter") + ylab("Percentage of mutations kept")
```

```{r afmutfilter, fig.cap="Alleles frequencies of filtered mutations.", fig.width=4, fig.height=4}
knitr::include_graphics("data/mutations/angela/filters/filters_mutations_af.png")
```

```{r vennmutfilter, fig.cap="Filtered mutations sharing across filters.", fig.width=4, fig.height=4}
knitr::include_graphics("data/mutations/angela/filters/filters_mutations_venn.png")
```

```{r nmutcovfilter, fig.cap="Link between the number of detected mutations and the coverage across samples depending on the filter used."}
vroom::vroom("data/mutations/angela/filters/filters_mutations.pct")  %>%  
  left_join(rename(cov_lims, tumor = library)) %>% 
  group_by(tumor, filter) %>% 
  ggplot(aes(med_cov, `TRUE`, col = filter, label = tumor)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  geom_smooth(se = F, method = "lm") +
  scale_y_log10() +
  xlab("Coverage") + ylab("Number of mutations")
```

### Cambium

```{r, eval=F}
trunk %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  filter(normal_altCountT1 == 0) %>% 
  filter(mutation_altCountT1 >= 10) %>% 
  collect() %>% 
  left_join(rename(cov_lims, tumor = library)) %>% 
  filter(normal_DP <= high_cov &
                         normal_DP >= low_cov &
                         mutation_DP <= high_cov &
                         mutation_DP >= low_cov) %>% 
  mutate(Filter = ifelse(FILTER == "PASS", "evs", "base")) %>% 
  dplyr::select(-ChromKey, -NT, -ID, -QUAL, -FILTER, -SOMATIC, -PNOISE, -PNOISE2, -normal, -caller, -high_cov, -low_cov, -med_cov) %>% 
  group_by(tumor, SNV, Filter, CHROM, POS, REF, ALT, SGT) %>% 
  summarise_all(median) %>% 
  vroom::vroom_write("data/mutations/angela/filters/trunk_mutations.tsv")
```

Only 19 to 668 mutations passed evs filtering across samples (Tab. \@ref(tab:nmutTrunkFiltTab)).
Most of mutations are low frequency, with an enrichment of AF<0.15 in T3 (Fig. \@ref(fig:afTrunkFiltFig)).
Only a few mutations are shared by cambium samples, between T2 and T3 and between T3 and T4 (Fig. \@ref(fig:vennTrunkFiltFig)).

```{r nmutTrunkFiltTab}
vroom::vroom("data/mutations/angela/filters/trunk_mutations.tsv") %>%  
  dplyr::select(tumor, SNV, Filter) %>% 
  unique() %>% 
  group_by(tumor, Filter) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(tumor ~ Filter) %>% 
  mutate(base = base + evs) %>% 
  kable(caption = "Number of filtered mutations per cambium.")  
```

```{r afTrunkFiltFig, fig.cap="Alleles-frequencies of filtered mutations in cambium samples."}
vroom::vroom("data/mutations/angela/filters/trunk_mutations.tsv") %>% 
  ggplot(aes(mutation_AF, col = tumor), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency")  +
  facet_wrap(~ Filter) +
  scale_x_sqrt()
```

```{r vennTrunkFiltFig, fig.cap="Filtered mutations sharing across cambium samples."}
trunk_mutations <- vroom::vroom("data/mutations/angela/filters/trunk_mutations.tsv")
cowplot::plot_grid(
  ggvenn::ggvenn(lapply(split(trunk_mutations, trunk_mutations$tumor), `[[`, "SNV"), show_percentage = F), 
  ggvenn::ggvenn(lapply(split(filter(trunk_mutations, Filter == "evs"), filter(trunk_mutations, Filter == "evs")$tumor), `[[`, "SNV"), show_percentage = F),
  labels = c("base", "evs")
)
```

### Leaf

```{r, eval=F}
leaf %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  filter(normal_altCountT1 == 0) %>% 
  filter(mutation_altCountT1 >= 10) %>% 
  collect() %>% 
  left_join(rename(cov_lims, tumor = library)) %>% 
  filter(normal_DP <= high_cov &
                         normal_DP >= low_cov &
                         mutation_DP <= high_cov &
                         mutation_DP >= low_cov) %>% 
  mutate(Filter = ifelse(FILTER == "PASS", "evs", "base")) %>% 
  dplyr::select(-ChromKey, -NT, -ID, -QUAL, -FILTER, -SOMATIC, -PNOISE, -PNOISE2, -normal, -caller, -high_cov, -low_cov, -med_cov) %>% 
  group_by(tumor, SNV, Filter, CHROM, POS, REF, ALT, SGT) %>% 
  summarise_all(median) %>% 
  mutate(branch = str_sub(tumor, 1, 1), tip = str_sub(tumor, 2, 2), leaf = str_sub(tumor, 3, 3)) %>%
  vroom::vroom_write("data/mutations/angela/filters/leaf_mutations.tsv")
```

Only 28 to 63 mutations passed evs filtering across samples (Fig. \@ref(fig:nmutTrunkFiltTab)).
Most of mutations are low frequency (Fig. \@ref(fig:afLeafFiltFig)).
Most of mutations are not shared by biological replicates (Fig. \@ref(fig:vennLeafFiltevsFig) and Fig. \@ref(fig:vennLeafFiltbaseFig)).

```{r nmutLeafFiltTab, fig.cap="Number of filtered mutations per leaf."}
vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
  group_by(tumor, branch, tip, leaf, Filter) %>% 
  summarise(N = n()) %>% 
  ggplot(aes(branch, N, fill = tip)) +
  geom_boxplot() +
  facet_wrap(~ Filter, scales = "free_x") +
  coord_flip() +
  ylab("Number of mutations") + xlab("Branch")
```

```{r afLeafFiltFig, fig.cap="Alleles-frequencies of filtered mutations in leaf samples."}
vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
  ggplot(aes(mutation_AF, col = tumor), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency") +
  facet_wrap(~ Filter) 
```

```{r vennLeafFiltevsFig, fig.cap="Filtered mutations sharing across leaf samples within tips for evs filtering.", fig.height=8}
t <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% filter(Filter == "evs")
g1 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(t, branch == B, tip == "S"),
                              filter(t, branch == B, tip == "S")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
g2 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(t, branch == B, tip == "L"),
                              filter(t, branch == B, tip == "L")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
cowplot::plot_grid(plotlist = c(g1,g2))
```

```{r vennLeafFiltbaseFig, fig.cap="Filtered mutations sharing across leaf samples within tips for base filtering.", fig.height=8}
t <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv")
g1 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(t, branch == B, tip == "S"),
                              filter(t, branch == B, tip == "S")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
g2 <- lapply(LETTERS[1:5], function(B)
  ggvenn::ggvenn(lapply(split(filter(t, branch == B, tip == "L"),
                              filter(t, branch == B, tip == "L")$tumor),
                        `[[`, "SNV"), show_percentage = F)
)
cowplot::plot_grid(plotlist = c(g1,g2))
```

### Tip

We further filtered mutations shared by at least 2 leaves or all leaves in each tip (filters base2, base3, evs2 and evs3).
Only 1 to 15 evs3-filtered mutations where shared between all leaves across tips (Tab. \@ref(tab:nmutTipFiltTab))).
Most mutations, even shared, have low frequencies (Fig. \@ref(fig:afTipFiltFig))!
Only a few mutations are shared by tips within a branch (Fig. \@ref(fig:vennTipFiltFig)).

```{r mutTipFiltered}
tip_mutations <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
  group_by(branch, tip, SNV, CHROM, POS, REF, ALT) %>% 
  mutate(N = n()) %>% 
  filter(N > 1) %>% 
  arrange(SNV) %>% 
  dplyr::select(-leaf) %>% 
  mutate(tumor = paste0(branch, tip)) %>% 
  group_by(CHROM, POS, SNV, REF, ALT, 
           tumor, branch, tip, N) %>% 
  summarise(mutation_AF = mean(mutation_AF),
            Filter = ifelse("evs" %in% Filter, "evs", "base")) %>% 
  mutate(Filter = paste0(Filter, N)) %>% 
  dplyr::select(-N) %>% 
  ungroup()
```

```{r, eval=F}
library(vcfR)
# mutations as tsv
write_tsv(tip_mutations, "data/mutations/angela/tip_mutations.tsv")
# mutations back to vcf
ref_mut_vcf <- read.vcfR("data/mutations/angela/raw/AL1_vs_T2.raw.vcf")
header <- ref_mut_vcf@meta
header[64] <- "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">"
header[65] <- "##FILTER=<ID=base2,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (10th-90th), and mutations sharing across leaves (at least 2).\">"
header[66] <- "##FILTER=<ID=base3,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (10th-90th), and mutations sharing across leaves (at least 3).\">"
header[67] <- "##FILTER=<ID=evs2,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (10th-90th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 2).\">"
header[68] <- "##FILTER=<ID=evs3,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (10th-90th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 3).\">"
header <- header[-c(69:88)]
fix <- tip_mutations %>% 
  mutate(QUAL = NA, INFO=NA) %>% 
  dplyr::rename(ID = SNV, FILTER = Filter) %>% 
  mutate(FITLER1 = str_match_all(FILTER, "[a-z]+"), FITLER2 = str_match_all(FILTER, "[0-9]+")) %>% 
  unnest() %>% 
  group_by(CHROM, POS, ID, REF, ALT, QUAL, INFO) %>% 
  summarise(FITLER2 = max(FITLER2),
            FITLER1 = ifelse("evs" %in% FITLER1, "evs", "base")) %>% 
  ungroup() %>% 
  mutate(FILTER = paste0(FITLER1, FITLER2)) %>% 
  dplyr::select(-FITLER1, FITLER2) %>% 
  dplyr::select(CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO) %>% 
  arrange(CHROM, POS) %>% 
  mutate(POS = as.character(POS))
head(as.matrix(fix))
gt <- dplyr::select(tip_mutations, SNV) %>% 
  mutate(tumor = list(unique(tip_mutations$tumor))) %>% 
  unnest() %>% 
  left_join(select(tip_mutations, SNV, tumor, Filter)) %>% 
  mutate(GT = ifelse(is.na(Filter), "0/0", "0/1")) %>% 
  mutate(FORMAT = "GT") %>% 
  dplyr::select(SNV, FORMAT, GT, tumor) %>% 
  unique() %>% 
  reshape2::dcast(SNV + FORMAT ~ tumor, value.var = "GT") %>% 
  dplyr::rename(ID = SNV)
gt <- left_join(dplyr::select(fix, ID), gt) %>% 
  dplyr::select(-ID)
head(as.matrix(gt))
mutation_vcf <- ref_mut_vcf
mutation_vcf@meta <- header
mutation_vcf@fix <- as.matrix(fix)
mutation_vcf@gt <- as.matrix(gt)
write.vcf(mutation_vcf, "data/mutations/angela/filters/filtered_tip_mutations.vcf.gz")


```


```{r nmutTipFiltTab}
tip_mutations %>%
  group_by(tumor, branch, tip, Filter) %>%
  summarise(N = n()) %>%
  reshape2::dcast(branch + tip ~ Filter, value.var = "N") %>%
  mutate_all(funs(ifelse(is.na(.), 0, .))) %>% 
  mutate(base2 = base2 + base3 + evs2, evs2 = evs2 + evs3, base3 = base3 + evs3) %>% 
  kable(caption = "Number of shared mutations per tip.")  
```

```{r afTipFiltFig, fig.cap="Mean alleles-frequencies of filtered mutations shared in each tip."}
tip_mutations %>%
  ggplot(aes(mutation_AF, fill = tumor), fill = NA) +
  geom_histogram(position = "dodge") +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency") +
  facet_wrap(~ Filter, scales = "free_y")  
```

```{r vennTipFiltFig, fig.cap="Sharing of filtered mutations shared in each tip across branches. The different lines correspond in order tpo filters base2, evs2, base3, evs3."}
g.base2 <- lapply(LETTERS[1:5], function(B){
  ggvenn::ggvenn(lapply(split(filter(tip_mutations, branch == B),
                            filter(tip_mutations, branch == B)$tumor),
                        `[[`, "SNV"), show_percentage = F)
})
g.base3 <- lapply(LETTERS[1:5], function(B){
  t <- filter(tip_mutations, Filter %in% c("base3", "evs3"))
  ggvenn::ggvenn(lapply(split(filter(t, branch == B),
                            filter(t, branch == B)$tumor),
                        `[[`, "SNV"), show_percentage = F)
})
g.evs2 <- lapply(LETTERS[1:5], function(B){
  t <- filter(tip_mutations, Filter %in% c("evs2", "evs3"))
  ggvenn::ggvenn(lapply(split(filter(t, branch == B),
                            filter(t, branch == B)$tumor),
                        `[[`, "SNV"), show_percentage = F)
})
g.evs3 <- lapply(LETTERS[1:5], function(B){
  t <- filter(tip_mutations, Filter == "evs3")
  ggvenn::ggvenn(lapply(split(filter(t, branch == B),
                            filter(t, branch == B)$tumor),
                        `[[`, "SNV"), show_percentage = F)
})
cowplot::plot_grid(plotlist = c(g.base2, g.evs2, g.base3, g.evs3))  
```

### Branch

We further filtered mutations shared by all tips in each branch (still using filters base2, base3, evs2 and evs3).
Only 0 to 4 evs3-filtered mutations where shared between all leaves across tips (Tab. \@ref(tab:nmutBranchFiltTab))).
Most mutations, even shared, have low frequencies (Fig. \@ref(fig:afBranchFiltFig))!
Only a few mutations are shared by tips within a branch (Fig. \@ref(fig:vennTipFiltFig)), **but 6 mutations are shared across the crown with filter base2**.

```{r mutBranchFiltered}
branch_mutations <- tip_mutations %>% 
  group_by(branch, SNV, CHROM, POS, REF, ALT) %>% 
  filter(n() == 2) %>% 
  dplyr::select(-tip) %>% 
  group_by(SNV, CHROM, POS, REF, ALT, branch, Filter) %>% 
  summarise(mutation_AF = mean(mutation_AF)) 
```
```{r nmutBranchFiltTab}
branch_mutations %>%
  group_by(branch, Filter) %>%
  summarise(N = n()) %>% 
  reshape2::dcast(branch ~ Filter) %>% 
  mutate_all(funs(ifelse(is.na(.), 0, .))) %>% 
  mutate(base2 = base2 + base3 + evs2, evs2 = evs2 + evs3, base3 = base3 + evs3) %>% 
  kable(caption = "Number of shared mutations per tip.") 
```

```{r afBranchFiltFig, fig.cap="Mean alleles-frequencies of filtered mutations shared in each tip."}
branch_mutations %>%
  ggplot(aes(mutation_AF, fill = branch), fill = NA) +
  geom_histogram(position = "dodge") +
  scale_color_discrete("Sample") +
  xlab("Allelic frequency") + 
  facet_wrap(~Filter)
```

```{r vennBranchFiltFig, fig.cap="Sharing of filtered mutations shared in each branch in the whole tree."}
t <- branch_mutations
g.base2 <- ggvenn::ggvenn(lapply(split(t,
                            t$branch),
                        `[[`, "SNV"), show_percentage = F)
t <- filter(branch_mutations, Filter %in% c("base3", "evs3"))
g.base3 <- ggvenn::ggvenn(lapply(split(t,
                            t$branch),
                        `[[`, "SNV"), show_percentage = F)
t <- filter(branch_mutations, Filter %in% c("evs2", "evs3"))
g.evs2 <- ggvenn::ggvenn(lapply(split(t,
                            t$branch),
                        `[[`, "SNV"), show_percentage = F) 
t <- filter(branch_mutations, Filter == "evs3")
g.evs3 <- ggvenn::ggvenn(lapply(split(t,
                            t$branch),
                        `[[`, "SNV"), show_percentage = F) 
cowplot::plot_grid(plotlist = list(g.base2, g.evs2, g.base3, g.evs3), labels = c("base2", "evs2", "base3", "evs3"))  
```

## Summary

<!-- In summary, each leaf sample contained approximately 250,000 candidate mutations, resulting in 58 to 126 robust mutations with both high and low allelic frequencies. Most robust mutations were shared by leaves at the branch tips, resulting in 19 to 57 mutations shared by leaves at the tip of a branch with high average allelic frequency in all samples (but some low-frequency mutations persisted). Only 2 to 6 mutations were shared between the two tips of a branch with allelic frequency diversity. And of these, only 7 were shared within a pair of branches: 2 between A and B and 5 between C and E, highlighting the greater architectural proximity of these pairs of branches in the tree. -->
