```{r setupmutationsverzy, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(circlize)
library(dtplyr)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

# Mutations Hetre

This chapter describes the analyses of mutations from the ["Faux de Verzy"](https://en.wikipedia.org/wiki/Faux_de_Verzy)
currently done in the the [`hetre` branch of the `detectMutations` repository](https://github.com/sylvainschmitt/detectMutations/tree/hetre).

## Quality check

The results of `multiQC` are available [here](/data/mutations/hetre/multiqc_report_hetre.html).

## Genome

```{r hetreGenomeData}
genome <- read_tsv("data/mutations/hetre/Fagus_sylvatica_v3.fa.fai", 
         col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(name, length) %>% 
  mutate(name = gsub("Fsylvatica_scaffold_", "", name)) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
coverage <- list("mutant" = "data/mutations/hetre/mutant.regions.bed", 
     "revertant" = "data/mutations/hetre/revertant.regions.bed") %>% 
  lapply(vroom::vroom, col_names = c("chrom", "start", "stop", "coverage")) %>% 
  bind_rows(.id = "library") %>% 
  mutate(chrom = gsub("Fsylvatica_scaffold_", "", chrom))
```

```{r hetreGenomFig, fig.cap="Fagus sylvatica scaffolds ordered by size."}
ggplot(genome, aes(reorder(name, end), end/10^6, fill = end > 10*10^6)) +
  geom_col() +
  coord_flip() +
  ylab("Length (mb)") + 
  xlab("Fagus sylvatica scaffolds ordered by size") +
  scale_fill_discrete(">2Mb") +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  ggtitle(paste(nrow(genome), "scaffolds"), paste(nrow(filter(genome, end > 2*10^6)), "scaffolds >10Mb"))
```

```{r}
ggplot(coverage, aes(coverage)) +
  geom_histogram(fill = "lightgrey") +
  xlim(0, 200) +
  facet_wrap(~ library) +
  xlab("Mean coverage on a 10kb window") +
  geom_vline(aes(xintercept = coverage), col = "red", size = 1.2,
             data = group_by(coverage, library) %>% 
               summarise(coverage = mean(coverage))) +
  geom_text(aes(label = paste0(round(coverage), "X"),
                x = coverage - 15), y = 10000, col = "red",
             data = group_by(coverage, library) %>% 
               summarise(coverage = mean(coverage)))
```


```{r circosHetre, eval=F}
png("save/hetreCircosCov.png", width = 1000, height = 1000)
circos.initializeWithIdeogram(as.data.frame(filter(genome, end > 10*10^6)))
circos.genomicTrack(filter(coverage, library == "mutant", 
                           chrom %in% unique(filter(genome, end > 10*10^6)$name)) %>% 
                      mutate(coverage = ifelse(coverage > 200, 200, coverage)) %>% 
                      dplyr::select(-library) %>% 
                      as.data.frame(), 
                    panel.fun = function(region, value, ...) {
                      circos.genomicLines(region, value, col = "red", ...)
                    })
circos.genomicTrack(filter(coverage, library == "revertant", 
                           chrom %in% unique(filter(genome, end > 10*10^6)$name)) %>% 
                      mutate(coverage = ifelse(coverage > 200, 200, coverage)) %>% 
                      dplyr::select(-library) %>% 
                      as.data.frame(), 
                    panel.fun = function(region, value, ...) {
                      circos.genomicLines(region, value, col = "blue", ...)
                    })
legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("revertant", "mutant"), bty = "n")
dev.off()
```

```{r circosHetreg, fig.cap="Genome coverage on a 10-kb windows for mutant and revertant (scaffolds > 10-Mb)."}
include_graphics("save/hetreCircosCov.png")
```

## Mutations

```{r hetredb, cache=FALSE}
mutant <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/hetre/mutant_vs_revertant.raw.sql") %>% tbl("mutations")
revertant <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/hetre/revertant_vs_mutant.raw.sql") %>% tbl("mutations")
```

```{r hetrerawstats, eval=F}
tally(mutant)
tally(revertant)
lapply(list(mutant = mutant, revertant = revertant),
       function(tab)
         mutate(tab, SNV = paste0(CHROM, "_", "Pos", as.integer(POS))) %>% 
         dplyr::select(CHROM, POS, SNV) %>% 
         collect() %>% 
         unique()) %>% bind_rows(.id = "library") %>% 
  lazy_dt() %>% 
  group_by(CHROM, POS) %>% 
  filter(n() > 1) %>% 
  dplyr::select(CHROM, POS, SNV) %>% 
  unique()
```

We detected 1 143 611 raw candidate mutations in the mutant and 1 293 760 in the revertant, with only 372 219 in common.

We used the following filters:

* **Raw**
    * The mutation is detected in both directions of the comparison mutant versus revertant
    * A null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
* **Base**
   * All filters from raw
   * A read depth for the two sample between half and two times the mean coverage (`normal_DP <= 150, normal_DP >= 25.5, mutation_DP <= 150, mutation_DP >= 25.5`)
   * A minimum of 5 alternate allele count in the mutated sample (`mutation_altCountT1 >= 5`), corresponding to a minimum allele frequency of 0.1 for a mean coverage of 50X
* **Robust**
   * All filters from base
   * `Strelka 2` automatic filtering (`Filter == "PASS"`)
   
```{r hetrerawfilter}
raw <- lapply(list(mutant = mutant, revertant = revertant),
       function(tab)
         mutate(tab, SNV = paste0(CHROM, "_", "Pos", as.integer(POS))) %>% 
         filter(POS < 16468) %>%
         collect() %>% 
         unique()) %>% bind_rows(.id = "library") %>% 
  lazy_dt() %>% 
  group_by(CHROM, POS) %>% 
  filter(n() > 1) %>% 
  filter(normal_altCountT1 == min(normal_altCountT1)) %>% 
  filter(normal_altCountT1 == 0)
raw <- as_tibble(raw)
base <- filter(raw, 
               normal_DP <= 150, 
               normal_DP >= 25.5, 
               mutation_DP <= 150, 
               mutation_DP >= 25.5, 
               mutation_altCountT1 >= 5)
robust <- base[base$FILTER == "PASS",] 
```

```{r hetremuttab}
list(raw= raw, base = base, robust = robust) %>% 
  lapply(nrow) %>% 
  bind_rows(.id = "filter") %>% 
  kable(cpation = "Number of mutations per filter")
```


```{r hetremutfig, fig.cap="Alleles-frequencies of filtered mutations in hetre."}
list(raw= raw, base = base, robust = robust) %>% 
  bind_rows(.id = "filter") %>% 
  ggplot(aes(mutation_AF, col = filter), fill = NA) +
  geom_density(size = 1.3) +
  scale_color_discrete("Filter") +
  xlab("Allelic frequency")  
```

```{r hetremuttabfull}
mutate(base,filter = ifelse(SNV %in% robust$SNV, "robust", "base")) %>% 
  mutate(CHROM = gsub("Fsylvatica_scaffold_", "", CHROM)) %>% 
  dplyr::select(filter, library, CHROM, POS, REF, ALT, mutation_altCountT1, mutation_AF) %>% 
  DT::datatable(colnames = c("filter", "library", "scaffold", "position", "ref", "alt", "copies", "allelic frequency"))
```
