```{r setupmutationsverzy, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(circlize)
library(dtplyr)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

# Mutations Hetre

This chapter describes the analyses of mutations from the ["Faux de Verzy"](https://en.wikipedia.org/wiki/Faux_de_Verzy)
currently done in the the [`hetre` branch of the `detectMutations` repository](https://github.com/sylvainschmitt/detectMutations/tree/hetre).

## Quality check

The results of `multiQC` are available [here](/data/mutations/hetre/multiqc_report_hetre.html).

## Genome

```{r hetreGenomeData}
genomes <- list(
  "Ciron" = "data/mutations/hetre/Fagus_sylvatica_v3.fa.fai",
  "Mutant" = "data/mutations/hetre/Fagus_sylvatica_mutant_v0.fa.fai",
  "Revertant" = "data/mutations/hetre/Fagus_sylvatica_revertant_v0.fa.fai"
) %>% lapply(read_tsv, col_names = c("name", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  bind_rows(.id = "genome") %>% 
  dplyr::select(genome, name, length) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length) %>%  
  mutate(trsc = letters[1:nrow(.)], exon = LETTERS[1:nrow(.)])
coverages <- list(
  "mutant_Mutant" = "data/mutations/hetre/mutant_on_Fagus_sylvatica_mutant_v0.regions.bed", 
  "mutant_Revertant" = "data/mutations/hetre/mutant_on_Fagus_sylvatica_revertant_v0.regions.bed", 
  "mutant_Ciron" = "data/mutations/hetre/mutant_on_Fagus_sylvatica_v3.regions.bed",
  "revertant_Mutant" = "data/mutations/hetre/revertant_on_Fagus_sylvatica_mutant_v0.regions.bed", 
  "revertant_Revertant" = "data/mutations/hetre/revertant_on_Fagus_sylvatica_revertant_v0.regions.bed", 
  "revertant_Ciron" = "data/mutations/hetre/revertant_on_Fagus_sylvatica_v3.regions.bed"
  ) %>% 
  lapply(vroom::vroom, col_names = c("chrom", "start", "stop", "coverage")) %>% 
  bind_rows(.id = "library") %>% 
  separate(library, c("library", "genome"))
```

```{r hetreGenomFig, fig.cap="Fagus sylvatica scaffolds ordered by size."}
ggplot(genomes, aes(reorder(name, end), end/10^6, fill = end > 10*10^6)) +
  geom_col() +
  coord_flip() +
  ylab("Length (mb)") + 
  xlab("Fagus sylvatica scaffolds ordered by size") +
  scale_fill_discrete(">10Mb") +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  facet_wrap(~ genome, scales = "free")
```

```{r}
ggplot(coverages, aes(coverage)) +
  geom_histogram(fill = "lightgrey") +
  xlim(0, 200) +
  facet_grid(genome ~ library) +
  xlab("Mean coverage on a 10kb window") +
  geom_vline(aes(xintercept = coverage), col = "red", size = 1.2,
             data = group_by(coverages, genome, library) %>% 
               summarise(coverage = median(coverage))) +
  geom_text(aes(label = paste0(round(coverage), "X"),
                x = coverage - 15), y = 10000, col = "red",
             data = group_by(coverages, genome, library) %>% 
               summarise(coverage = median(coverage))) 
```


```{r g_name, eval=F}
circoscov <- function(g_name, coverage, depth, file){
  png(file, width = 1000, height = 1000)
  subgenome <- filter(genomes, genome == g_name, end > coverage*10^6) %>% dplyr::select(-genome)
  circos.initializeWithIdeogram(as.data.frame(subgenome))
  circos.genomicTrack(filter(coverages, genome == g_name, library == "mutant", chrom %in% subgenome$name) %>% 
                        mutate(coverage = ifelse(coverage > depth, depth, coverage)) %>% 
                        dplyr::select(-library, -genome) %>% 
                        as.data.frame(), 
                      panel.fun = function(region, value, ...) {
                        circos.genomicLines(region, value, col = "red", ...)
                      })
  circos.genomicTrack(filter(coverages, genome == g_name, library == "revertant", chrom %in% subgenome$name) %>% 
                        mutate(coverage = ifelse(coverage > depth, depth, coverage)) %>% 
                        dplyr::select(-library, -genome) %>% 
                        as.data.frame(), 
                      panel.fun = function(region, value, ...) {
                        circos.genomicLines(region, value, col = "blue", ...)
                      })
  legend("bottomleft", pch = 16, col = c("blue", "red"), legend = c("revertant", "mutant"), bty = "n")
  text(0, 0, g_name, cex = 1.5)
  dev.off()
}
circoscov("Ciron", 10, 200, "save/hetreCircosCovCiron.png")
circoscov("Revertant", 5, 200, "save/hetreCircosCovRevertant.png")
circoscov("Mutant", 5, 200, "save/hetreCircosCovMutant.png")
```

```{r circosHetreCiron, fig.cap="Genome coverage on a 10-kb windows for mutant and revertant (scaffolds > 10-Mb) on the Ciron's genome."}
include_graphics("save/hetreCircosCovCiron.png")
```

```{r circosHetreRevertant, fig.cap="Genome coverage on a 10-kb windows for mutant and revertant (scaffolds > 5-Mb) on the Revertant's genome."}
include_graphics("save/hetreCircosCovRevertant.png")
```

```{r circosHetreMutant, fig.cap="Genome coverage on a 10-kb windows for mutant and revertant (scaffolds > 5-Mb) on the Mutant's genome."}
include_graphics("save/hetreCircosCovMutant.png")
```

## Mutations

We used the following filters:

* **Base**
   * The mutation is detected in the revertant
   * A null number of alternate allele count in the normal sample (`normal_altCountT1 == 0`)
   * A read depth for the two sample between half and two times the mean coverage (`normal_DP <= 150, normal_DP >= 25.5, mutation_DP <= 150, mutation_DP >= 25.5`)
   * A minimum of 5 alternate allele count in the mutated sample (`mutation_altCountT1 >= 5`), corresponding to a minimum allele frequency of 0.1 for a mean coverage of 50X
   * A maximum allelic frequency of 0.5 (`mutation_AF <= 0.5`)
* **Robust**
   * All filters from base
   * `Strelka 2` automatic filtering (`Filter == "PASS"`)
   

```{r hetrerawfilter}
mutations <- DBI::dbConnect(RSQLite::SQLite(), "data/mutations/hetre/mutations.raw.sql") %>% 
  tbl("mutations") %>% 
  filter(
    tumor == "revertant",
    normal_altCountT1 == 0,
    normal_DP <= 150, 
    normal_DP >= 25.5, 
    mutation_DP <= 150, 
    mutation_DP >= 25.5, 
    mutation_altCountT1 >= 5,
    mutation_AF <= 0.5 
  ) %>% 
  collect() %>% 
  group_by(reference, CHROM, POS) %>% 
  filter(n() == 1) %>% 
  ungroup() %>% 
  mutate(Filter = ifelse(FILTER == "PASS", "robust", "base"))
```

```{r hetremuttab}
mutations %>% 
  group_by(reference, Filter) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(reference ~ Filter) %>% 
  kable(cpation = "Number of mutations per filter and reference", 
        format.args = list(big.mark = " "))
```


```{r hetremutfig, fig.cap="Alleles-frequencies of filtered mutations in hetre."}
ggplot(mutations, aes(mutation_AF, col = Filter), fill = NA) +
  geom_density(size = 1.3) +
  xlab("Allelic frequency")  +
  facet_wrap(~ reference, nrow = 2) +
  theme(legend.position = c(0.9, 0.1))
```

```{r hetremutmutantfig, fig.cap="Mutation position on the Mutant's genome (scaffolds > 5-Mb) for base (blue) and robust (red) candidates."}
ggplot(filter(genomes, genome == "Mutant", end > 5*10^6),
       aes(x = name, xend = name)) +
  geom_segment(aes(y = start, yend = end/10^6), size = 3, col = "lightgrey") +
  geom_point(aes(y = POS/10^6, col = mutation_AF),
             data = filter(mutations, reference == "Fagus_sylvatica_mutant_v0", Filter == "robust") %>%
               mutate(name = CHROM) %>%
               filter(name %in% filter(genomes, genome == "Mutant", end > 5*10^6)$name)) +
  coord_flip() +
  ylab("Position (Mb)") +
  theme(axis.line.y = element_blank(), axis.title.y = element_blank(), axis.ticks.y = element_blank()) +
  viridis::scale_color_viridis("AF")
```

```{r hetremutmutantfig2, fig.cap="Mutation position on the Mutant's genome (scaffolds > 5-Mb) for base (blue) and robust (red) candidates."}
ggplot(filter(genomes, genome == "Mutant", end > 5*10^6),
       aes(x = name, xend = name)) +
  geom_segment(aes(y = start, yend = end/10^6), size = 3, col = "lightgrey") +
  geom_point(aes(y = POS/10^6, col = mutation_AF),
             data = filter(mutations, reference == "Fagus_sylvatica_mutant_v0", Filter == "robust") %>%
               mutate(name = CHROM) %>%
               filter(name %in% filter(genomes, genome == "Mutant", end > 5*10^6)$name) %>% 
               filter(mutation_AF > 0.2)) +
  coord_flip() +
  ylab("Position (Mb)") +
  theme(axis.line.y = element_blank(), axis.title.y = element_blank(), axis.ticks.y = element_blank()) +
  viridis::scale_color_viridis("AF")

filter(mutations, reference == "Fagus_sylvatica_mutant_v0", Filter == "robust") %>%
               mutate(name = CHROM) %>%
               # filter(name %in% filter(genomes, genome == "Mutant", end > 5*10^6)$name) %>% 
               filter(mutation_AF > 0.2)
```


```{r}
filter(mutations, reference == "Fagus_sylvatica_mutant_v0", Filter == "robust") %>% 
  dplyr::select(CHROM, POS, REF, ALT, mutation_altCountT1, mutation_refCountT1, normal_altCountT1, normal_refCountT1, mutation_AF) %>% 
  DT::datatable(colnames = c("scf", "pos", "ref", "alt", "rev alt", "rev ref", "mut alt", "mut ref", "AF"))
```

