```{r setupmutationsanalyses, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(raster)
library(leaflet)
library(sf)
knit_hooks$set(webgl = hook_webgl)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```


# Mutations Analyses

*Chapter to be split*

## Mutations history

> Phylogenetic history of intraindividual mutations (T4.3): Once the confirmed mutations are available, we will use phylogenomic methods to establish the intra-tree evolutionary history of mutations and assess its fit with with the history expected from tree architecture (WP2) as illustrated by (Orr et al., 2019).

## Matiations & UVs

> Effects  of  UV  light  on  mutation  rates (T4.4): We  will  test  the  hypothesis  that  sunlight  enhances  the accumulation of mutation by comparing the number of mutations per physical branch meter in high and low light exposure branches and will compare mutation rates across species. Appropriate accounting for differences in light environment will be done based on modelling of light environment (T2.3) anddirect measurements of sunlight (T2.6).

## Mutations rate

> Evolution  of  mutation  rates(T5.1):  To  attempt  explaining  any  differences  in  observed  mutation  rates between  the  three  sequenced  species,  we  will  collect  data  from  the  literature  on  the  intra-individual accumulation of mutations (or somatic mutation rates) in plants and examine if a phylogenetic pattern (Soltis, Folk, & Soltis, 2019)can be observed after carefully ensuring the comparability of data considering the methods used in each study. (Soltis et al., 2019). We will then examine the evolution of gene families involved  in  UV  response  and  DNA  damage  tolerance  and  repair (Nakagawa  et  al.,  2011;  Oztas,  Selby, Sancar,  &  Adebali,  2018;  Ulm  &  Jenkins,  2015)and  any  other  relevant  pathways  to  explain  possible differences  in  the  rate  of  accumulation  of  mutations  between  plant  species,  and  the  three  sequenced species  in  particular.  Publically  available  data  on  the  same  gene  families  across  plant  lineages  will  be included (One Thousand Plant Transcriptomes Initiative, 2019). We will particularly address differences in gene  family  evolution  in  these  pathways    in  plants  with  different  life  history  traits  (e.g.  annual  plants, temperate trees).

## Germline segregation

> Timing  of  germline  segregation (T5.2):We  will  assess  whether  soma  and  germline  undergo  early segregation in tropical trees by assessing the transmission of mutations specific to each branch to gametic tissues, pollen or fruit pericarp, expected to carry a maternal genotype (Smouse, Sork, Scofield, & Grivet, 2012)and  to  zygotes  formed  on  the  branch  using  targeted  genotyping.  If  mutations  are  not  typically transmitted,  an  early  soma-germline  segregation  is  likely (Lanfear,  2018).  In  this  case,  mutations accumulated in tropical  tree tissues would indeed be  typically somatic, i.e., characteristic of vegetative tissues   only,   without   typically   entering   the   germline   andrepresenting   evolutionary   dead-ends. Conversely,  if  gametic  tissues  most  resemble  nearby  vegetative  tissues  (Figure  2  B)  and  mutations  are typically  transmitted  to  zygotes  in  proportions  expected  from  meiosis  as  in  oak (Plomion  et  al.,  2018), these  mutations  accumulated  through  an  individualâ€™s lifetime  could  not  be  called  purely  somatic:  their transmission  to  offspring  and  thus  their  evolutionary  relevance  would  be  demonstrated.  We  will  then compare the timing of germline segregation among the three species, considering their life history and species-richness of their families, and compare with recent studies from other scientists (this topic is an active field of research).

