```{r setuphistory, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(420)
library(knitr)
library(tidyverse)
library(ape)
library(phangorn)
library(ggtree)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r insilicomut}
n_branches <- 5
tips <- c("L", "O")
p_null <- 3
phylo <- rtree(n_branches, tip.label = paste0("B", 1:n_branches))
branches <- phylo$tip.label
for(b in branches){
  phylo <- add.tips(phylo, paste0(b, "_L"), b, edge.length = abs(rnorm(1, mean(phylo$edge.length)/4, sd(phylo$edge.length)/2)))
  phylo <- add.tips(phylo, paste0(b, "_O"), b, edge.length = abs(rnorm(1, mean(phylo$edge.length)/4, sd(phylo$edge.length)/2)))
  phylo <- drop.tip(phylo, b)
}
branches_list <- lapply(branches, function(b) c(paste0(b, "_L"), paste0(b, "_O")))
names(branches_list) <- branches
phylo <- makeNodeLabel(phylo, "u", prefix = "#", nodeList = branches_list)
rm(branches_list, b)
# orr et al. 2019 0.06 subs/site = 2.0 M
archi <- phylo
archi$edge.length <- (phylo$edge.length + 
                        abs(rnorm(length(phylo$edge.length), 0, sd(phylo$edge.length)))
                      ) * (1/0.06) # subs/site to meters 
nulltrees <- lapply(ape::rmtree(10^p_null-1, length(archi$tip.label), tip.label = archi$tip.label), 
                    treedist, archi) %>% 
  bind_rows() %>% 
  mutate(null = 1) %>% 
  bind_rows(treedist(phylo, archi)) %>% 
  mutate(null = ifelse(is.na(null), 0, null)) %>% 
  reshape2::melt("null") %>% 
  mutate(variable = gsub(".", " ", variable, fixed = T)) %>% 
  filter(variable %in% c("branch score difference", "path difference")) %>%
  arrange(value) %>% 
  group_by(variable) %>% 
  mutate(rank = 1:n()) %>% 
  mutate(pvalue = rank/n()) %>% 
  mutate(pvalue = ifelse(pvalue == min(pvalue), paste("P <", pvalue), paste("P =", pvalue))) 
```

```{r shuffle_tree, eval=F, echo=F}
shuffle_tree = function(tree){
  max_rf = 2*(length(tree$tip.label) - 3) # max RF dist
  rf = 0
  while(rf < max_rf){
    null_tree = tree
    null_tree$tip.label = sample(null_tree$tip.label)
    rf = RF.dist(null_tree, tree)
  }
  return(new.tree)
}
plot(shuffle_tree(phylo))
```

# Mutations history

> Phylogenetic history of intraindividual mutations (T4.3): Once the confirmed mutations are available, we will use phylogenomic methods to establish the intra-tree evolutionary history of mutations and assess its fit with with the history expected from tree architecture (WP2) as illustrated by (Orr et al., 2019).

## *in silico*

Let's generate a tree from 10 branch tips belonging to 5 branches with varying genetic distances due to mutations.
We can simulate a corresponding architecture with physical distance in meters instead of substitutions per site, and we will use a normal low to introduce noise in physical branches length compared to phylogeny length.
We thus obtained two similar trees (Fig. \@ref(fig:treeplot)), significantly more similar than random trees with the same number of tips (Fig. \@ref(fig:nulltreesinsilico), **the model null is too weak to me**).

```{r treeplot, fig.cap="Simulated phylogeny and archtiecture of sampled branches and tips."}
cowplot::plot_grid(
  ggtree(phylo) +
    geom_nodelab(nudge_x = 0.06) +
    geom_tiplab() +
    geom_treescale() +
    xlim(0,2) +
    ggtitle("Phylogeny", "subtitutions per site"),
  ggtree(archi) +
    geom_nodelab(nudge_x = 2) +
    geom_tiplab() +
    geom_treescale() +
    xlim(0,50) +
    ggtitle("Architecture", "meters"),
  nrow = 1
) 
```

```{r nulltreesinsilico, fig.cap="Observed differences between phylogeny and architecture compared to null trees with the same number of tips."}
ggplot(nulltrees, aes(value)) +
  geom_histogram(data = filter(nulltrees, null == 1), fill = "lightgrey") +
  geom_vline(aes(xintercept = value), col = "red", size = 1.2, data = filter(nulltrees, null == 0)) +
  geom_label(aes(x = value, label = pvalue),
             y = c(100, 100), x = c(5, 58.5),
             col = "red",
            data = filter(nulltrees, null == 0)) +
  facet_wrap(~ variable, scales = "free") 
```

## Angela

### Tips

```{r tipsvcfs, eval=F}
library(vcfR)
tip_mutations <- read_tsv("data/mutations/angela/filters/filtered_tip_mutations.tsv")
# vcf header
ref_mut_vcf <- read.vcfR("data/mutations/angela/raw/AL1_vs_T2.raw.vcf")
header <- ref_mut_vcf@meta
header[64] <- "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">"
header[65] <- "##FILTER=<ID=base2,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (5th-95th), and mutations sharing across leaves (at least 2).\">"
header[66] <- "##FILTER=<ID=base3,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (5th-95th), and mutations sharing across leaves (at least 3).\">"
header[67] <- "##FILTER=<ID=evs2,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (5th-95th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 2).\">"
header[68] <- "##FILTER=<ID=evs3,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (5th-95th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 3).\">"
header <- header[-c(69:88)]
# filters
mutations <- list(
  base1 = filter(tip_mutations, Filter %in% c("base1",  "evs1", "base2", "evs2", "base3", "evs3")),
  base2 = filter(tip_mutations, Filter %in% c("base2", "evs2", "base3", "evs3")),
  base3 = filter(tip_mutations, Filter %in% c("base3", "evs3")),
  evs1 = filter(tip_mutations, Filter %in% c("evs1", "evs2", "evs3")),
  evs2 = filter(tip_mutations, Filter %in% c("evs2", "evs3")),
  evs3 = filter(tip_mutations, Filter == "evs3"),
  af1 = filter(tip_mutations, mutation_AF > 0.1),
  af2 = filter(tip_mutations, mutation_AF > 0.2),
  af3 = filter(tip_mutations, mutation_AF > 0.3)
)
vcfs <- lapply(mutations, function(data){
  fix <- data %>% 
    mutate(QUAL = NA, INFO=NA) %>% 
    dplyr::rename(ID = SNV, FILTER = Filter) %>% 
    mutate(FITLER1 = str_match_all(FILTER, "[a-z]+"), FITLER2 = str_match_all(FILTER, "[0-9]+")) %>% 
    unnest() %>% 
    group_by(CHROM, POS, ID, REF, ALT, QUAL, INFO) %>% 
    summarise(FITLER2 = max(FITLER2),
              FITLER1 = ifelse("evs" %in% FITLER1, "evs", "base")) %>% 
    ungroup() %>% 
    mutate(FILTER = paste0(FITLER1, FITLER2)) %>% 
    dplyr::select(-FITLER1, FITLER2) %>% 
    dplyr::select(CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO) %>% 
    arrange(CHROM, POS) %>% 
    mutate(POS = as.character(POS))
  # head(as.matrix(fix))
  gt <- dplyr::select(data, SNV) %>% 
    mutate(tumor = list(unique(data$tumor))) %>% 
    unnest() %>% 
    left_join(select(data, SNV, tumor, Filter)) %>% 
    mutate(GT = ifelse(is.na(Filter), "0/0", "1/1")) %>% 
    mutate(FORMAT = "GT") %>% 
    dplyr::select(SNV, FORMAT, GT, tumor) %>% 
    unique() %>% 
    reshape2::dcast(SNV + FORMAT ~ tumor, value.var = "GT") %>% 
    dplyr::rename(ID = SNV)
  gt <- left_join(dplyr::select(fix, ID), gt) %>% 
    dplyr::select(-ID)
  # head(as.matrix(gt))
  mutation_vcf <- ref_mut_vcf
  mutation_vcf@meta <- header
  mutation_vcf@fix <- as.matrix(fix)
  mutation_vcf@gt <- as.matrix(gt)
  return(mutation_vcf)
})
invisible(lapply(names(vcfs), function(v) write.vcf(vcfs[[v]], paste0("data/mutations/angela/trees/tips/", v, ".vcf.gz"))))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/angela/trees/tips
wget https://raw.githubusercontent.com/edgardomortiz/vcf2phylip/master/vcf2phylip.py
for file in $(ls *.vcf.gz); do python vcf2phylip.py -i $file -n ; done
```

```{r, eval=F, echo=T}
files <- list.files(path = "data/mutations/angela/trees/tips/", pattern = ".nexus")
invisible(lapply(files, function(f){
  mutations <- read.phyDat(file.path("data/mutations/angela/trees/tips", f), format = "nexus")
  mutations <- as.DNAbin(mutations)
  ips::write.fas(mutations, file.path("data/mutations/angela/trees/tips", gsub("nexus", "fa", f)))
}))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/angela/trees/tips
for file in $(ls *.min4.fa); do iqtree -s $file -redo ; done
```

```{r angleatipsphylo, fig.width=12}
files <- list.files(path = "data/mutations/angela/trees/tips/", pattern = ".min4.fa.treefile")
phylos <- lapply(files, function(f) read.tree(file.path("data/mutations/angela/trees/tips", f)))
names(phylos) <- gsub(".min4.fa.treefile", "", files)
cowplot::plot_grid(plotlist = lapply(phylos, function(p) 
  ggtree(phylobase::phylo4d(p)) + geom_tippoint(aes(col = substr(label, 1, 1)), size = 4) + geom_tiplab()
), labels = names(phylos)) 
```

<!-- ### Comparison -->

<!-- **Only for the code as the tree are not good, using iqtree.** -->

<!-- ```{r } -->
<!-- library(dendextend) -->
<!-- archi <-  -->
<!--   "((((AS:3.83,AL:6.04)A:6.44,(BS:3.34,BL:10.64)B:1.6)AB:6.5,((CS:3.8,CL:5.13)C:14.45,(DS:3.5,DL:7.76)D:6.2)CD:1.7,(ES:0,EL:2.2)E:3.7)XE:26.5)T:1.3;" -->
<!-- archi <- ape::read.tree(text = archi) -->
<!-- archi.dendro <- phylogram::as.dendrogram(archi) -->
<!-- phylo <- ape::read.tree("data/mutations/angela/trees/iqtree/filtered_tip_mutations.min4.fa.treefile") -->
<!-- phylo <- ape::chronos(phylo) -->
<!-- phylo <- phytools::midpoint.root(phylo) -->
<!-- phylo.dendro <- phylogram::as.dendrogram(phylo) -->
<!-- dndlist <- dendlist(archi.dendro, phylo.dendro) %>%  -->
<!--   untangle(method = "ladderize") -->
<!-- dndlist %>%  -->
<!--   tanglegram(sort = T, common_subtrees_color_branches = TRUE, -->
<!--              main_right = "Phylogeny", main_left = "Architecture")    -->
<!-- ``` -->

### Leaves

```{r leavesvcfs, eval=F}
library(vcfR)
leaf_mutations <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv")
# vcf header
ref_mut_vcf <- read.vcfR("data/mutations/angela/raw/AL1_vs_T2.raw.vcf")
header <- ref_mut_vcf@meta
header[64] <- "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">"
header[65] <- "##FILTER=<ID=base,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), and sequencing depth (5th-95th).\">"
header[66] <- "##FILTER=<ID=evs,Description=\"Filter based on minimum allele count for mutation (>10) and normal sample (=0), sequencing depth (5th-95th), and empiric variant score of Strelka2.\">"
header <- header[-c(67:88)]
# filters
mutations <- list(
  base = filter(leaf_mutations, Filter %in% c("base",  "evs")),
  evs = filter(leaf_mutations, Filter %in% c("evs")),
  af1 = filter(leaf_mutations, mutation_AF > 0.1),
  af2 = filter(leaf_mutations, mutation_AF > 0.2)
)
vcfs <- lapply(mutations, function(data){
  fix <- data %>% 
    mutate(QUAL = NA, INFO=NA) %>% 
    dplyr::rename(ID = SNV, FILTER = Filter) %>% 
    group_by(CHROM, POS, ID, REF, ALT, QUAL, INFO) %>% 
    dplyr::select(CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO) %>% 
    arrange(CHROM, POS) %>% 
    mutate(POS = as.character(POS)) %>% 
    ungroup()
  # head(as.matrix(fix))
  gt <- dplyr::select(data, SNV) %>% 
    mutate(tumor = list(unique(data$tumor))) %>% 
    unnest() %>% 
    left_join(select(data, SNV, tumor, Filter)) %>% 
    mutate(GT = ifelse(is.na(Filter), "0/0", "1/1")) %>% 
    mutate(FORMAT = "GT") %>% 
    dplyr::select(SNV, FORMAT, GT, tumor) %>% 
    unique() %>% 
    reshape2::dcast(SNV + FORMAT ~ tumor, value.var = "GT") %>% 
    dplyr::rename(ID = SNV)
  gt <- left_join(dplyr::select(fix, ID), gt) %>% 
    dplyr::select(-ID)
  # head(as.matrix(gt))
  mutation_vcf <- ref_mut_vcf
  mutation_vcf@meta <- header
  mutation_vcf@fix <- as.matrix(fix)
  mutation_vcf@gt <- as.matrix(gt)
  return(mutation_vcf)
})
invisible(lapply(names(vcfs), function(v) write.vcf(vcfs[[v]], paste0("data/mutations/angela/trees/leaves/", v, ".vcf.gz"))))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/angela/trees/leaves
wget https://raw.githubusercontent.com/edgardomortiz/vcf2phylip/master/vcf2phylip.py
for file in $(ls *.vcf.gz); do python vcf2phylip.py -i $file -n ; done
```

```{r, eval=F, echo=T}
files <- list.files(path = "data/mutations/angela/trees/leaves/", pattern = ".nexus")
invisible(lapply(files, function(f){
  mutations <- read.phyDat(file.path("data/mutations/angela/trees/leaves", f), format = "nexus")
  mutations <- as.DNAbin(mutations)
  ips::write.fas(mutations, file.path("data/mutations/angela/trees/leaves", gsub("nexus", "fa", f)))
}))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/angela/trees/leaves
for file in $(ls *.min4.fa); do iqtree -s $file -redo ; done
```

```{r anglealeavessphylo, fig.width=12}
files <- list.files(path = "data/mutations/angela/trees/leaves/", pattern = ".min4.fa.treefile")
phylos <- lapply(files, function(f) read.tree(file.path("data/mutations/angela/trees/leaves", f)))
names(phylos) <- gsub(".min4.fa.treefile", "", files)
cowplot::plot_grid(plotlist = lapply(phylos, function(p) 
  ggtree(phylobase::phylo4d(p)) + geom_tippoint(aes(col = substr(label, 1, 1)), size = 4) + geom_tiplab()
  ), labels = names(phylos))
```

## Sixto

### Tips

```{r tipsvcfsSixto, eval=F}
library(vcfR)
tip_mutations <- read_tsv("data/mutations/sixto/filters/filtered_tip_mutations.tsv")
# vcf header
ref_mut_vcf <- read.vcfR("data/mutations/sixto/raw/B1L1_vs_T1.nonhz.vcf")
header <- ref_mut_vcf@meta
header[45] <- "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">"
header[46] <- "##FILTER=<ID=base2,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), sequencing depth (5th-95th), and mutations sharing across leaves (at least 2).\">"
header[47] <- "##FILTER=<ID=base3,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), sequencing depth (5th-95th), and mutations sharing across leaves (at least 3).\">"
header[48] <- "##FILTER=<ID=evs2,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), sequencing depth (5th-95th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 2).\">"
header[49] <- "##FILTER=<ID=evs3,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), sequencing depth (5th-95th), empiric variant score of Strelka2, and mutations sharing across leaves (at least 3).\">"
header <- header[-c(50:69)]
# filters
mutations <- list(
  base1 = filter(tip_mutations, Filter %in% c("base1",  "evs1", "base2", "evs2", "base3", "evs3")),
  base2 = filter(tip_mutations, Filter %in% c("base2", "evs2", "base3", "evs3")),
  base3 = filter(tip_mutations, Filter %in% c("base3", "evs3")),
  evs1 = filter(tip_mutations, Filter %in% c("evs1", "evs2", "evs3")),
  evs2 = filter(tip_mutations, Filter %in% c("evs2", "evs3")),
  evs3 = filter(tip_mutations, Filter == "evs3"),
  af1 = filter(tip_mutations, mutation_AF > 0.1),
  af2 = filter(tip_mutations, mutation_AF > 0.2),
  af3 = filter(tip_mutations, mutation_AF > 0.3)
)
vcfs <- lapply(mutations, function(data){
  fix <- data %>% 
    mutate(QUAL = NA, INFO=NA) %>% 
    dplyr::rename(ID = SNV, FILTER = Filter) %>% 
    mutate(FITLER1 = str_match_all(FILTER, "[a-z]+"), FITLER2 = str_match_all(FILTER, "[0-9]+")) %>% 
    unnest() %>% 
    group_by(CHROM, POS, ID, REF, ALT, QUAL, INFO) %>% 
    summarise(FITLER2 = max(FITLER2),
              FITLER1 = ifelse("evs" %in% FITLER1, "evs", "base")) %>% 
    ungroup() %>% 
    mutate(FILTER = paste0(FITLER1, FITLER2)) %>% 
    dplyr::select(-FITLER1, FITLER2) %>% 
    dplyr::select(CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO) %>% 
    arrange(CHROM, POS) %>% 
    mutate(POS = as.character(POS))
  # head(as.matrix(fix))
  gt <- dplyr::select(data, SNV) %>% 
    mutate(tumor = list(unique(data$tumor))) %>% 
    unnest() %>% 
    left_join(select(data, SNV, tumor, Filter)) %>% 
    mutate(GT = ifelse(is.na(Filter), "0/0", "1/1")) %>% 
    mutate(FORMAT = "GT") %>% 
    dplyr::select(SNV, FORMAT, GT, tumor) %>% 
    unique() %>% 
    reshape2::dcast(SNV + FORMAT ~ tumor, value.var = "GT") %>% 
    dplyr::rename(ID = SNV)
  gt <- left_join(dplyr::select(fix, ID), gt) %>% 
    dplyr::select(-ID)
  # head(as.matrix(gt))
  mutation_vcf <- ref_mut_vcf
  mutation_vcf@meta <- header
  mutation_vcf@fix <- as.matrix(fix)
  mutation_vcf@gt <- as.matrix(gt)
  return(mutation_vcf)
})
invisible(lapply(names(vcfs), function(v) write.vcf(vcfs[[v]], paste0("data/mutations/sixto/trees/tips/", v, ".vcf.gz"))))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/sixto/trees/tips
wget https://raw.githubusercontent.com/edgardomortiz/vcf2phylip/master/vcf2phylip.py
for file in $(ls *.vcf.gz); do python vcf2phylip.py -i $file -n ; done
```

```{r, eval=F, echo=T}
files <- list.files(path = "data/mutations/sixto/trees/tips/", pattern = ".nexus")
invisible(lapply(files, function(f){
  mutations <- read.phyDat(file.path("data/mutations/sixto/trees/tips", f), format = "nexus")
  mutations <- as.DNAbin(mutations)
  ips::write.fas(mutations, file.path("data/mutations/sixto/trees/tips", gsub("nexus", "fa", f)))
}))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/sixto/trees/tips
for file in $(ls *.min4.fa); do iqtree -s $file -redo ; done
```

```{r sixtotipsphylo, fig.width=12}
files <- list.files(path = "data/mutations/sixto/trees/tips/", pattern = ".min4.fa.treefile")
phylos <- lapply(files, function(f) read.tree(file.path("data/mutations/sixto/trees/tips", f)))
names(phylos) <- gsub(".min4.fa.treefile", "", files)
cowplot::plot_grid(plotlist = lapply(phylos, function(p) 
  ggtree(phylobase::phylo4d(p)) + geom_tippoint(aes(col = substr(label, 1, 1)), size = 4) + geom_tiplab()
), labels = names(phylos))
```

<!-- ### Comparison -->

<!-- **Only for the code as the tree are not good, using iqtree.** -->

<!-- ```{r } -->
<!-- library(dendextend) -->
<!-- archi <-  -->
<!--   "((((AS:3.83,AL:6.04)A:6.44,(BS:3.34,BL:10.64)B:1.6)AB:6.5,((CS:3.8,CL:5.13)C:14.45,(DS:3.5,DL:7.76)D:6.2)CD:1.7,(ES:0,EL:2.2)E:3.7)XE:26.5)T:1.3;" -->
<!-- archi <- ape::read.tree(text = archi) -->
<!-- archi.dendro <- phylogram::as.dendrogram(archi) -->
<!-- phylo <- ape::read.tree("data/mutations/angela/trees/iqtree/filtered_tip_mutations.min4.fa.treefile") -->
<!-- phylo <- ape::chronos(phylo) -->
<!-- phylo <- phytools::midpoint.root(phylo) -->
<!-- phylo.dendro <- phylogram::as.dendrogram(phylo) -->
<!-- dndlist <- dendlist(archi.dendro, phylo.dendro) %>%  -->
<!--   untangle(method = "ladderize") -->
<!-- dndlist %>%  -->
<!--   tanglegram(sort = T, common_subtrees_color_branches = TRUE, -->
<!--              main_right = "Phylogeny", main_left = "Architecture")    -->
<!-- ``` -->

### Leaves

```{r leavesvcfsSixto, eval=F}
library(vcfR)
leaf_mutations <- vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")
# vcf header
ref_mut_vcf <- read.vcfR("data/mutations/sixto/raw/B1L1_vs_T1.nonhz.vcf")
header <- ref_mut_vcf@meta
header[45] <- "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">"
header[46] <- "##FILTER=<ID=base,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), and sequencing depth (5th-95th).\">"
header[47] <- "##FILTER=<ID=evs,Description=\"Filter based on minimum allele count for mutation (>5) and normal sample (=0), sequencing depth (5th-95th), and empiric variant score of Strelka2.\">"
header <- header[-c(48:69)]
# filters
mutations <- list(
  base = filter(leaf_mutations, Filter %in% c("base",  "evs")),
  evs = filter(leaf_mutations, Filter %in% c("evs")),
  af1 = filter(leaf_mutations, mutation_AF > 0.1),
  af2 = filter(leaf_mutations, mutation_AF > 0.2)
)
vcfs <- lapply(mutations, function(data){
  fix <- data %>% 
    mutate(QUAL = NA, INFO=NA) %>% 
    dplyr::rename(ID = SNV, FILTER = Filter) %>% 
    group_by(CHROM, POS, ID, REF, ALT, QUAL, INFO) %>% 
    dplyr::select(CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO) %>% 
    arrange(CHROM, POS) %>% 
    mutate(POS = as.character(POS)) %>% 
    ungroup()
  # head(as.matrix(fix))
  gt <- dplyr::select(data, SNV) %>% 
    mutate(tumor = list(unique(data$tumor))) %>% 
    unnest() %>% 
    left_join(select(data, SNV, tumor, Filter)) %>% 
    mutate(GT = ifelse(is.na(Filter), "0/0", "1/1")) %>% 
    mutate(FORMAT = "GT") %>% 
    dplyr::select(SNV, FORMAT, GT, tumor) %>% 
    unique() %>% 
    reshape2::dcast(SNV + FORMAT ~ tumor, value.var = "GT") %>% 
    dplyr::rename(ID = SNV)
  gt <- left_join(dplyr::select(fix, ID), gt) %>% 
    dplyr::select(-ID)
  # head(as.matrix(gt))
  mutation_vcf <- ref_mut_vcf
  mutation_vcf@meta <- header
  mutation_vcf@fix <- as.matrix(fix)
  mutation_vcf@gt <- as.matrix(gt)
  return(mutation_vcf)
})
invisible(lapply(names(vcfs), function(v) write.vcf(vcfs[[v]], paste0("data/mutations/sixto/trees/leaves/", v, ".vcf.gz"))))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/sixto/trees/leaves
wget https://raw.githubusercontent.com/edgardomortiz/vcf2phylip/master/vcf2phylip.py
for file in $(ls *.vcf.gz); do python vcf2phylip.py -i $file -n ; done
```

```{r, eval=F, echo=T}
files <- list.files(path = "data/mutations/sixto/trees/leaves/", pattern = ".nexus")
invisible(lapply(files, function(f){
  mutations <- read.phyDat(file.path("data/mutations/sixto/trees/leaves", f), format = "nexus")
  mutations <- as.DNAbin(mutations)
  ips::write.fas(mutations, file.path("data/mutations/sixto/trees/leaves", gsub("nexus", "fa", f)))
}))
```

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/sixto/trees/leaves
for file in $(ls *.min4.fa); do iqtree -s $file -redo ; done
```

```{r sixtoleavessphylo, fig.width=12}
files <- list.files(path = "data/mutations/sixto/trees/leaves/", pattern = ".min4.fa.treefile")
phylos <- lapply(files, function(f) read.tree(file.path("data/mutations/sixto/trees/leaves", f)))
names(phylos) <- gsub(".min4.fa.treefile", "", files)
cowplot::plot_grid(plotlist = lapply(phylos, function(p) 
  ggtree(phylobase::phylo4d(p)) + geom_tippoint(aes(col = substr(label, 1, 1)), size = 4) + geom_tiplab()
  ), labels = names(phylos))
```
