```{r setuphistory, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(420)
library(knitr)
library(kableExtra)
if(knitr:::is_html_output()) options(knitr.table.format = "html") 
if(knitr:::is_latex_output()) options(knitr.table.format = "latex") 
library(tidyverse)
library(ape)
library(phangorn)
library(ggtree)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r insilicomut}
n_branches <- 5
tips <- c("L", "O")
p_null <- 3
phylo <- rtree(n_branches, tip.label = paste0("B", 1:n_branches))
branches <- phylo$tip.label
for(b in branches){
  phylo <- add.tips(phylo, paste0(b, "_L"), b, edge.length = abs(rnorm(1, mean(phylo$edge.length)/4, sd(phylo$edge.length)/2)))
  phylo <- add.tips(phylo, paste0(b, "_O"), b, edge.length = abs(rnorm(1, mean(phylo$edge.length)/4, sd(phylo$edge.length)/2)))
  phylo <- drop.tip(phylo, b)
}
branches_list <- lapply(branches, function(b) c(paste0(b, "_L"), paste0(b, "_O")))
names(branches_list) <- branches
phylo <- makeNodeLabel(phylo, "u", prefix = "#", nodeList = branches_list)
rm(branches_list, b)
# orr et al. 2019 0.06 subs/site = 2.0 M
archi <- phylo
archi$edge.length <- (phylo$edge.length + 
                        abs(rnorm(length(phylo$edge.length), 0, sd(phylo$edge.length)))
                      ) * (1/0.06) # subs/site to meters 
nulltrees <- lapply(ape::rmtree(10^p_null-1, length(archi$tip.label), tip.label = archi$tip.label), 
                    treedist, archi) %>% 
  bind_rows() %>% 
  mutate(null = 1) %>% 
  bind_rows(treedist(phylo, archi)) %>% 
  mutate(null = ifelse(is.na(null), 0, null)) %>% 
  reshape2::melt("null") %>% 
  mutate(variable = gsub(".", " ", variable, fixed = T)) %>% 
  filter(variable %in% c("branch score difference", "path difference")) %>%
  arrange(value) %>% 
  group_by(variable) %>% 
  mutate(rank = 1:n()) %>% 
  mutate(pvalue = rank/n()) %>% 
  mutate(pvalue = ifelse(pvalue == min(pvalue), paste("P <", pvalue), paste("P =", pvalue))) 
```

```{r shuffle_tree, eval=F, echo=F}
shuffle_tree = function(tree){
  max_rf = 2*(length(tree$tip.label) - 3) # max RF dist
  rf = 0
  while(rf < max_rf){
    null_tree = tree
    null_tree$tip.label = sample(null_tree$tip.label)
    rf = RF.dist(null_tree, tree)
  }
  return(new.tree)
}
plot(shuffle_tree(phylo))
```

# Mutations history

> Phylogenetic history of intraindividual mutations (T4.3): Once the confirmed mutations are available, we will use phylogenomic methods to establish the intra-tree evolutionary history of mutations and assess its fit with with the history expected from tree architecture (WP2) as illustrated by (Orr et al., 2019).

## *in silico*

Let's generate a tree from 10 branch tips belonging to 5 branches with varying genetic distances due to mutations.
We can simulate a corresponding architecture with physical distance in meters instead of substitutions per site, and we will use a normal low to introduce noise in physical branches length compared to phylogeny length.
We thus obtained two similar trees (Fig. \@ref(fig:treeplot)), significantly more similar than random trees with the same number of tips (Fig. \@ref(fig:nulltreesinsilico), **the model null is too weak to me**).

```{r treeplot, fig.cap="Simulated phylogeny and archtiecture of sampled branches and tips."}
cowplot::plot_grid(
  ggtree(phylo) +
    geom_nodelab(nudge_x = 0.06) +
    geom_tiplab() +
    geom_treescale() +
    xlim(0,2) +
    ggtitle("Phylogeny", "subtitutions per site"),
  ggtree(archi) +
    geom_nodelab(nudge_x = 2) +
    geom_tiplab() +
    geom_treescale() +
    xlim(0,50) +
    ggtitle("Architecture", "meters"),
  nrow = 1
) 
```

```{r nulltreesinsilico, fig.cap="Observed differences between phylogeny and architecture compared to null trees with the same number of tips."}
ggplot(nulltrees, aes(value)) +
  geom_histogram(data = filter(nulltrees, null == 1), fill = "lightgrey") +
  geom_vline(aes(xintercept = value), col = "red", size = 1.2, data = filter(nulltrees, null == 0)) +
  geom_label(aes(x = value, label = pvalue),
             y = c(100, 100), x = c(5, 58.5),
             col = "red",
            data = filter(nulltrees, null == 0)) +
  facet_wrap(~ variable, scales = "free") 
```

## Angela

### File formats

We have vcfs that we can convert in phylip and nexus using a python script and then in fasta using R:

```{bash, eval=F, echo=T}
cd treemutation/data/mutations/angela/trees/mutations
wget https://raw.githubusercontent.com/edgardomortiz/vcf2phylip/master/vcf2phylip.py
python vcf2phylip.py -i filtered_tip_mutations.vcf.gz -n
```

```{r, eval=F, echo=T}
mutations <- read.phyDat("data/mutations/angela/trees/mutations/filtered_tip_mutations.min4.nexus", format = "nexus")
mutations <- as.DNAbin(mutations)
write.fas(mutations, "data/mutations/angela/trees/mutations/filtered_tip_mutations.min4.fa")
```

### `MrBayes`

We can use the nexus file in `MrBayes` manually inputting command as follow (following manual basic example with increased ngen):

```{bash, eval=F, echo=T}
mb
execute filtered_tip_mutations.min4.nexus
lset nst=6 rates=invgamma
mcmc ngen=100000 samplefreq=100 printfreq=100 diagnfreq=1000
no
sump
sumt
quit
```

```{r}
phylo <- read.nexus("data/mutations/angela/trees/mrbayes/filtered_tip_mutations.min4.nexus.con.tre")
plot(phylo)
```

### `iqtree`

We can use the fasta file with `iqtree`:

```{bash, eval=F, echo=T}
iqtree -s filtered_tip_mutations.min4.fa
```

```{r }
phylo <- read.tree("data/mutations/angela/trees/iqtree/filtered_tip_mutations.min4.fa.treefile")
plot(phylo)
```

### `RAxML`

We can deposit online deposit the fasta file on https://raxml-ng.vital-it.ch/#/ for RaXML:

```{r}
phylo <- read.tree("data/mutations/angela/trees/raxml/result.raxml.bestTree")
plot(phylo)
```

### `phangorn`

We can use locally the nexus file with phangorn:

```{r, echo=T}
library(phangorn)
mutations <- read.phyDat("data/mutations/angela/trees/mutations/filtered_tip_mutations.min4.nexus", format = "nexus")
dm <- dist.hamming(mutations)
tree <- NJ(dm)
parsimony(tree, mutations)
treeRA <- random.addition(mutations)
treeNNI <- optim.parsimony(tree, mutations)
treeRatchet <- pratchet(mutations, start=tree, maxit=100, k=5)
treeRatchet <- acctran(treeRatchet, mutations)
plot(midpoint(treeRatchet))
```

### Comparison

**Only for the code as the tree are not good, using iqtree.**

```{r }
library(dendextend)
phylo <- read.tree("data/mutations/angela/trees/iqtree/filtered_tip_mutations.min4.fa.treefile")
archi <- 
  "((((AS:3.83,AL:6.04)A:6.44,(BS:3.34,BL:10.64)B:1.6)AB:6.5,((CS:3.8,CL:5.13)C:14.45,(DS:3.5,DL:7.76)D:6.2)CD:1.7,(ES:0,EL:2.2)E:3.7)XE:26.5)T:1.3;"
archi <- read.tree(text = archi)
phylo <- chronos(phylo)
phylo <- phytools::midpoint.root(phylo)
phylo.dendro <- phylogram::as.dendrogram(phylo)
archi.dendro <- phylogram::as.dendrogram(archi)
dndlist <- dendlist(phylo.dendro, archi.dendro) %>% 
  untangle(method = "ladderize")
dndlist %>% 
  tanglegram(sort = T, common_subtrees_color_branches = TRUE,
             main_right = "Architecture", main_left = "Phylogeny")
```

```{r, eval=F}
lengths <- lapply(c(archi = angela, phylo = stree), 
                  function(x)
                    data.frame(
                      length = adephylo::distRoot(x, method = "patristic")) %>% 
                    rownames_to_column("tip")) %>% 
  bind_rows(.id = "tree") %>% 
  reshape2::dcast(tip ~ tree) %>% 
  mutate(branch = str_sub(tip, 1, 1), tip = str_sub(tip, 2, 2))

ggplot(lengths, aes(archi, phylo, col = tip)) +
  geom_point() +
  geom_smooth(method = "lm") +
  xlab("Architecture distance") +
  ylab("Phylogeny distance") +
  viridis::scale_color_viridis(direction = -1, discrete = T, option = "B")
```
