```{r setupcomp2, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(tidyverse)
library(ggtree)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r datacomp2}
# genomes
sextonia_genome <- read_tsv("data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
dicorynia_genome <- read_tsv("data/mutations/sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
# busco
sextonia_busco <- read_tsv("data/mutations/sixto/genome/short_summary.specific.viridiplantae_odb10.HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD_busco.txt",
                  skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1) %>% 
  mutate(Type = factor(Type, levels= c("Complete and single-copy BUSCOs (S)",
                                       'Complete and duplicated BUSCOs (D)',
                                       "Fragmented BUSCOs (F)",
                                       "Missing BUSCOs (M)")))
dicorynia_busco <- read_tsv("data/mutations/angela/genome/short_summary.specific.viridiplantae_odb10.Dgu_HS1_HYBRID_SCAFFOLD_busco.txt",
                  skip = 10, n_max = 4, col_names = c("X1", "N", "Type")) %>% 
  dplyr::select(-X1) %>% 
  mutate(Type = factor(Type, levels= c("Complete and single-copy BUSCOs (S)",
                                       'Complete and duplicated BUSCOs (D)',
                                       "Fragmented BUSCOs (F)",
                                       "Missing BUSCOs (M)")))
# coverages
covs_files <- list.files("data/mutations/angela/genome/dists/", full.names = T)
names(covs_files) <- list.files("data/mutations/angela/genome/dists/")
covs_angela <- lapply(covs_files, read_tsv, col_names = c("chromosome", "coverage", "proportion"), cols(
  chromosome = col_character(),
  coverage = col_double(),
  proportion = col_double()
)) %>% 
  bind_rows(.id = "file") %>% 
  mutate(library = gsub(".mosdepth.global.dist.txt", "", file)) %>% 
  filter(chromosome == "total") %>% 
  dplyr::select(library, coverage, proportion)

covs_files <- list.files("data/mutations/sixto/genome/dists/", full.names = T)
names(covs_files) <- list.files("data/mutations/sixto/genome/dists/")
covs_sixto <- lapply(covs_files, read_tsv, col_names = c("chromosome", "coverage", "proportion"), cols(
  chromosome = col_character(),
  coverage = col_double(),
  proportion = col_double()
)) %>%
  bind_rows(.id = "file") %>%
  mutate(library = gsub(".mosdepth.global.dist.txt", "", file)) %>%
  filter(chromosome == "total") %>% 
  dplyr::select(library, coverage, proportion)
coverages <- list(
  "Sixto" = covs_sixto, 
  "Angela" = covs_angela) %>% 
  bind_rows(.id = "tree") %>% 
  vroom::vroom_write("save/coverages_comparisons2.tsv")
rm(covs_files, covs_angela, covs_sixto)
cov_lims <- coverages %>% 
  mutate(proportion = round(proportion, 2)) %>% 
  mutate(proportion = ifelse(tree == "Verzy" & proportion == 0.94, 0.95, proportion)) %>% 
  filter(proportion == 0.05 | proportion == 0.95) %>% 
  group_by(tree, library, proportion) %>% 
  summarise(coverage = mean(coverage, na.omit = T)) %>% 
  mutate(proportion = recode(as.character(proportion), "0.05" = "high_cov", "0.95" = "low_cov")) %>% 
  reshape2::dcast(tree + library ~ proportion)
# archi
angela <- 
  "((((AS:3.83,AL:6.04)A:6.44,(BS:3.34,BL:10.64)B:1.6)AB:6.5,((CS:3.8,CL:5.13)C:14.45,(DS:3.5,DL:7.76)D:6.2)CD:1.7,(ES:0,EL:2.2)E:3.7)X:26.5)T:1.3;"
angela <- phylobase::phylo4d(ape::read.tree(text = angela))
sixto <- 
  "(((((B1L:5.2,(C3L:4.47, C1S:0.75)C:0.4)XC:3.08,(E2L:3.8, E4S:3.5)E:1.6)XE:0.6,F2S:4.37)XF:1.6,(I1L:3.3, I2S:1.15)I:1.1)XI:28.64)T:1.3;"
sixto <- phylobase::phylo4d(ape::read.tree(text = sixto))
# mutations
archi_mutations_angela_evs <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
  filter(Filter == "evs") %>% 
  mutate(tip = paste0(branch, tip)) %>% 
  group_by(SNV, CHROM, POS, REF, ALT, branch, tip) %>% 
  summarise(mutation_AF = mean(mutation_AF)) %>% 
  group_by(SNV, CHROM, POS) %>% 
  summarise(tip = paste(tip, collapse = " "), mutation_AF = mean(mutation_AF), n_tips = n(),
             n_branches = length(unique(branch)), branch = paste(sort(unique(branch)), collapse = " "),
            REF = first(REF), ALT = first(ALT)) %>% 
  ungroup() %>% 
  arrange(desc(n_tips), desc(mutation_AF)) %>% 
  mutate(origin1 = ifelse(n_tips == 1, tip, "branch")) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & n_branches == 1, branch, origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & branch == "A B", "AB", origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & branch == "C D", "CD", origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch", "X", origin1)) %>% 
  mutate(origin2 = ifelse(n_tips == 1, "tip", "branch")) %>% 
  mutate(origin2 = ifelse(origin1 %in% c("CD", "AB"), "carpenter", origin2)) %>% 
  mutate(origin2 = ifelse(origin1 == "X", "crown", origin2))
archi_mutations_angela_base <- vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
  mutate(tip = paste0(branch, tip)) %>% 
  group_by(SNV, CHROM, POS, REF, ALT, branch, tip) %>% 
  summarise(mutation_AF = mean(mutation_AF)) %>% 
  group_by(SNV, CHROM, POS) %>% 
  summarise(tip = paste(tip, collapse = " "), mutation_AF = mean(mutation_AF), n_tips = n(),
             n_branches = length(unique(branch)), branch = paste(sort(unique(branch)), collapse = " "),
            REF = first(REF), ALT = first(ALT)) %>% 
  ungroup() %>% 
  arrange(desc(n_tips), desc(mutation_AF)) %>% 
  mutate(origin1 = ifelse(n_tips == 1, tip, "branch")) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & n_branches == 1, branch, origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & branch == "A B", "AB", origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & branch == "C D", "CD", origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch", "X", origin1)) %>% 
  mutate(origin2 = ifelse(n_tips == 1, "tip", "branch")) %>% 
  mutate(origin2 = ifelse(origin1 %in% c("CD", "AB"), "carpenter", origin2)) %>% 
  mutate(origin2 = ifelse(origin1 == "X", "crown", origin2))
archi_mutations_sixto_evs <- vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
  filter(Filter == "evs") %>% 
  mutate(tip = paste0(branch, tip)) %>%
  mutate(branch = str_sub(branch, 1, 1)) %>%
  group_by(SNV, CHROM, POS, REF, ALT, branch, tip) %>% 
  summarise(mutation_AF = mean(mutation_AF)) %>% 
  group_by(SNV, CHROM, POS) %>% 
  summarise(tip = paste(tip, collapse = " "), mutation_AF = mean(mutation_AF), n_tips = n(),
             n_branches = length(unique(branch)), branch = paste(sort(unique(branch)), collapse = " "),
            REF = first(REF), ALT = first(ALT)) %>% 
  ungroup() %>% 
  arrange(desc(n_tips), desc(mutation_AF)) %>% 
  mutate(origin1 = ifelse(n_tips == 1, tip, "branch")) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & n_branches == 1, branch, origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch", paste0("X", str_sub(branch, -1, -1)), origin1)) %>% 
    mutate(origin2 = ifelse(n_tips == 1, "tip", "x")) %>% 
  mutate(origin2 = ifelse(origin2 == "x" & n_branches == 1, "branch", origin2)) %>% 
  mutate(origin2 = ifelse(origin2 == "x", "carpenter", origin2)) %>% 
  mutate(origin2 = ifelse(origin1 == "XI", "crown", origin2))
archi_mutations_sixto_base <- vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
  mutate(tip = paste0(branch, tip)) %>%
  mutate(branch = str_sub(branch, 1, 1)) %>%
  group_by(SNV, CHROM, POS, REF, ALT, branch, tip) %>% 
  summarise(mutation_AF = mean(mutation_AF)) %>% 
  group_by(SNV, CHROM, POS) %>% 
  summarise(tip = paste(tip, collapse = " "), mutation_AF = mean(mutation_AF), n_tips = n(),
             n_branches = length(unique(branch)), branch = paste(sort(unique(branch)), collapse = " "),
            REF = first(REF), ALT = first(ALT)) %>% 
  ungroup() %>% 
  arrange(desc(n_tips), desc(mutation_AF)) %>% 
  mutate(origin1 = ifelse(n_tips == 1, tip, "branch")) %>% 
  mutate(origin1 = ifelse(origin1 == "branch" & n_branches == 1, branch, origin1)) %>% 
  mutate(origin1 = ifelse(origin1 == "branch", paste0("X", str_sub(branch, -1, -1)), origin1)) %>% 
    mutate(origin2 = ifelse(n_tips == 1, "tip", "x")) %>% 
  mutate(origin2 = ifelse(origin2 == "x" & n_branches == 1, "branch", origin2)) %>% 
  mutate(origin2 = ifelse(origin2 == "x", "carpenter", origin2)) %>% 
  mutate(origin2 = ifelse(origin1 == "XI", "crown", origin2))
# lapply(list(archi_mutations_angela_evs, archi_mutations_angela_base, archi_mutations_sixto_evs, archi_mutations_sixto_base),
#        function(x) nrow(x) == length(unique(x$SNV)))
# phylo4d
angela@data <- data.frame(label = angela@label) %>% 
  left_join(archi_mutations_angela_evs %>% 
              group_by(origin1) %>% 
              summarise(mutations_evs = n()) %>% 
              dplyr::rename(label = origin1)) %>% 
  mutate(mutations_evs = ifelse(is.na(mutations_evs), 0, mutations_evs)) %>% 
  left_join(archi_mutations_angela_base %>% 
              group_by(origin1) %>% 
              summarise(mutations_base = n()) %>% 
              dplyr::rename(label = origin1)) %>% 
  mutate(mutations_base = ifelse(is.na(mutations_base), 0, mutations_base)) %>% 
  dplyr::select(-label)
sixto@data <- data.frame(label = sixto@label) %>% 
  left_join(archi_mutations_sixto_evs %>% 
              group_by(origin1) %>% 
              summarise(mutations_evs = n()) %>% 
              dplyr::rename(label = origin1)) %>% 
  mutate(mutations_evs = ifelse(is.na(mutations_evs), 0, mutations_evs)) %>% 
  left_join(archi_mutations_sixto_base %>% 
              group_by(origin1) %>% 
              summarise(mutations_base = n()) %>% 
              dplyr::rename(label = origin1)) %>% 
  mutate(mutations_base = ifelse(is.na(mutations_base), 0, mutations_base)) %>% 
  dplyr::select(-label)
# phylogenies
angela_phylo <- ape::read.tree("data/mutations/angela/trees/evs.min4.fa.treefile")
sixto_phylo <- ape::read.tree("data/mutations/sixto/trees/evs.min4.fa.treefile")

```

# Comparisons 2

This chapter compares the mutations of Sixto and Angela on the basis of the closest possible filtering. 

## Genomes

The total genome size of *Sextonia rubra* is larger than that of *Dicoryinia guyanensis*, but the *Dicoryinia guyanensis* genome was better assembled in fewer scaffolds (Fig. \@ref(fig:compgenomes2)).

```{r compgenomes2, fig.cap="Caption."}
list("Sextonia rubra" = sextonia_genome, 
     "Dicorynia guyanensis" = dicorynia_genome) %>% 
  bind_rows(.id = "Species") %>% 
  ggplot(aes(reorder(as.character(scf), stop), stop)) +
  geom_col() +
  coord_flip() +
  ylab("Length (mb)") + 
  xlab("Anchored scaffolds") +
  facet_wrap(~ Species, scales = "free_y") +
  theme(strip.text.x = element_text(face = "italic"), axis.text.y = element_text(size = 4))
```

```{r compgenomes2busco, fig.cap="Caption."}
list("Sextonia rubra" = sextonia_busco, 
     "Dicorynia guianensis" = dicorynia_busco) %>% 
  bind_rows(.id = "Species") %>% 
  ggplot(aes(Species, N, fill = Type)) +
  geom_col() +
  geom_text(aes(y = N, label = N), col = "white", position = position_stack(vjust = .5)) +
  scale_y_sqrt() +
  scale_fill_discrete("", guide = guide_legend(nrow = 4)) +
  theme(axis.title.x = element_blank(), axis.text.x = element_text(face = "italic"), 
        axis.line.x = element_blank(), axis.ticks.x = element_blank())
```


## Libraries

As expected, Angela's sequencing depth is twice as high as Sixto's, averaging about 150X versus 80X (Fig. \@ref(fig:compcovs2)), which translates into greater accepted sequencing depth in Angela (note that as a result, Sixto's libraries are less filtered in terms of coverage, Fig. \@ref(fig:compvolims2)). 

```{r compcovs2, fig.cap="Caption."}
ggplot(coverages, aes(coverage, proportion*100, col = library)) +
  geom_line() +
  xlim(0, 400) +
  geom_hline(yintercept = 5, col = "red", linetype = "dashed") +
  geom_hline(yintercept = 95, col = "red", linetype = "dashed") +
  scale_color_discrete(guide = "none") + 
  xlab("Sequencing depth (X)") +
  ylab("% of base in the genome with at least X reads") +
  facet_wrap(~ tree, nrow = 4) 
```

```{r compvolims2, fig.cap="Caption."}
cov_lims %>% 
  ggplot(aes(x = paste(tree, library), xend = paste(tree, library), y = low_cov, yend = high_cov, col = tree)) + 
  geom_segment() +
  coord_flip() +
  scale_color_discrete("") +
  ylab("Filtered sequencing depth (X, 5-95%)") + xlab("") +
  theme(axis.text.y = element_text(size = 4))
```

## Mutations

The origins of the mutations are distributed in the crown, but most of the mutations with basic filtering come from the base of the crown, followed by the carpenter, the branches and finally the tips (Fig. \@ref(fig:mutcomp2)). Interestingly, stronger filtering favoured mutations in the tips as well as at the base of the crown.

```{r mutcomp2, fig.width=10, fig.height=10, fig.cap="Caption."}
g1 <- ggtree(angela, right = T) +
  geom_point2(aes(size = mutations_evs, col = mutations_evs)) +
  geom_tiplab(offset = 1) +
  geom_nodelab(nudge_x = 0.8) +
  geom_label(aes(label = mutations_evs, fill = mutations_evs), vjust=1.5, col = "white") +
  theme_tree() +
  theme(legend.position = "right") +
  xlim(20, 50) +
  viridis::scale_color_viridis(guide = "none", limits = c(1, 260), trans = scales::log10_trans()) +
  viridis::scale_fill_viridis(guide = "none", limits = c(1, 260), trans = scales::log10_trans()) +
  scale_size_continuous(guide = "none", limits = c(1, 260)) +
  coord_flip() +
  ggtitle("Mutations origin in the crown", "At least in 2-leaves with evs filtering.")
g2 <- ggtree(angela, right = T) +
  geom_point2(aes(size = mutations_base, col = mutations_base)) +
  geom_tiplab(offset = 1) +
  geom_nodelab(nudge_x = 0.8) +
  geom_label(aes(label = mutations_base, fill = mutations_base), vjust=1.5, col = "white") +
  theme_tree() +
  theme(legend.position = "right") +
  xlim(20, 50) +
  viridis::scale_color_viridis(guide = "none", limits = c(1, 10849), trans = scales::log10_trans()) +
  viridis::scale_fill_viridis(guide = "none", limits = c(1, 10849), trans = scales::log10_trans()) +
  scale_size_continuous(guide = "none", limits = c(1, 10849)) +
  coord_flip() +
  ggtitle("Mutations origin in the crown", "At least in 2-leaves with base filtering.")
g3 <- ggtree(sixto, right = T) +
  geom_point2(aes(size = mutations_evs, col = mutations_evs)) +
  geom_tiplab(offset = 1) +
  geom_nodelab(nudge_x = 0.8) +
  geom_label(aes(label = mutations_evs, fill = mutations_evs), vjust=1.5, col = "white") +
  theme_tree() +
  theme(legend.position = "right") +
  xlim(20, 45) +
  viridis::scale_color_viridis(guide = "none", limits = c(1, 260), trans = scales::log10_trans()) +
  viridis::scale_fill_viridis(guide = "none", limits = c(1, 260), trans = scales::log10_trans()) +
  scale_size_continuous(guide = "none", limits = c(1, 260)) +
  coord_flip() +
  ggtitle("Mutations origin in the crown", "At least in 2-leaves with evs filtering.")
g4 <- ggtree(sixto, right = T) +
  geom_point2(aes(size = mutations_base, col = mutations_base)) +
  geom_tiplab(offset = 1) +
  geom_nodelab(nudge_x = 0.8) +
  geom_label(aes(label = mutations_base, fill = mutations_base), vjust=1.5, col = "white") +
  theme_tree() +
  theme(legend.position = "right") +
  xlim(20, 45) +
  viridis::scale_color_viridis(guide = "none", limits = c(1, 10849), trans = scales::log10_trans()) +
  viridis::scale_fill_viridis(guide = "none", limits = c(1, 10849), trans = scales::log10_trans()) +
  scale_size_continuous(guide = "none", limits = c(1, 10849)) +
  coord_flip() +
  ggtitle("Mutations origin in the crown", "At least in 2-leaves with base filtering.")
cowplot::plot_grid(g1, g2, g3, g4)
rm(g1, g2, g3, g4)
```

```{r}
lapply(list(angela = "data/mutations/angela/filters/leaf_mutations.tsv",
            sixto = "data/mutations/sixto/filters/leaf_mutations.tsv"), vroom::vroom) %>% 
  bind_rows(.id = "tree") %>% 
  # filter(mutation_AF > 0.15) %>% 
  # filter(Filter == "evs") %>% 
  mutate(tip = paste0(branch, tip)) %>% 
  dplyr::select(tree, tip, SNV) %>% 
  unique() %>% 
  group_by(tree, tip) %>% 
  summarise(N = n()) %>% 
  left_join(
    list(angela = data.frame(length = adephylo::distRoot(angela)), 
         sixto = data.frame(length = adephylo::distRoot(sixto))) %>% 
      bind_rows(.id = "tree") %>% 
      rownames_to_column("tip")) %>% 
  # split(.$tree) %>% lapply(function(x) lm(N ~ length, x)) %>% lapply(summary)
  ggplot(aes(length, N, col = tree)) +
  geom_smooth(method = "lm", se = F) +
  geom_point() +
  ggrepel::geom_text_repel(aes(label = tip)) +
  xlab("Branch length from trunk") + ylab("Number of detected mutations")
```

## Frequencies

Surprisingly, the allelic frequencies of mutations occurring at the base of the crown are not significantly higher than those at the tips (Fig. \@ref(fig:afcomp2)).

```{r afcomp2, fig.cap="Caption."}
list("Sixto" = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_sixto_base, SNV, origin2)),
     "Angela" = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_angela_base, SNV, origin2))) %>% 
  bind_rows(.id = "tree") %>% 
  filter(!is.na(origin2)) %>% 
  ggplot(aes(x = mutation_AF, color = origin2)) +
  geom_density() + scale_x_log10() +
  xlab("Allelic frequencies (AF)") +
  facet_grid(tree ~ Filter)
```


```{r, eval=F}
list("Sixto" = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_sixto_base, SNV, origin2)),
     "Angela" = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_angela_base, SNV, origin2))) %>% 
  bind_rows(.id = "tree") %>% 
  filter(!is.na(origin2)) %>% 
    filter(Filter == "base") %>% 
  ggplot(aes(x = mutation_AF, color = tree)) +
  geom_density() +
  xlab("Allelic fraction (AF)")
```


```{r deltaafcomp2, fig.cap="Caption."}
list("Sixto" = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_sixto_base, SNV, origin2)),
     "Angela" = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_angela_base, SNV, origin2))) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  mutate(DeltaAF = max(mutation_AF) - min(mutation_AF)) %>% 
  filter(!is.na(origin2)) %>% 
  ggplot(aes(x = DeltaAF, color = origin2)) +
  geom_density() + scale_x_log10() +
  xlab("Variation in allelic frequencies per SNV (DeltaAF)") +
  facet_grid(tree ~ Filter)
```

```{r deltaafcomp3, fig.cap="Caption."}
t <- list("Sixto" = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_sixto_base, SNV, origin2)),
     "Angela" = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv") %>% 
       left_join(dplyr::select(archi_mutations_angela_base, SNV, origin2))) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(DeltaAF =max(mutation_AF) - min(mutation_AF), repetitions = n())
t2 <- t %>% group_by(repetitions) %>% summarise(DeltaAF = median(DeltaAF))
coefs <- lm(log(DeltaAF) ~ repetitions, t2)$coefficients
ggplot(t, aes(repetitions, DeltaAF, group = NA)) +
  geom_violin(aes(group = cut_width(repetitions, 1)), scale = "width") +
  stat_function(fun = function(x){exp(coefs[1] +  coefs[2]*x)}, col = "blue") +
  geom_point(data = t2, col = "red") +
  annotate("text", 20, 0.32, label = expression(y==e^{alpha+beta*x})) +
  annotate("text", 20, 0.3, label = expression(beta==0.09285)) +
  annotate("text", 20, 0.28, label = expression(p==5.25e-07)) +
  annotate("text", 20, 0.26, label = expression(R^2==0.7682)) +
  annotate("text", 20, 0.24, label = expression(N==19)) +
  xlab("Number of libraries with the muation") + ylab(expression(Delta[AF]==max[AF]-min[AF]))
rm(t, t2, coefs)
```

## Phylogeny

Therefore, mutations are strongly shared across the crown and do not always follow the architecture. As a result, the phylogeny of the mutations does not match the architecture of the tree, with the exception of branch I, which is conserved in Sixto (Fig. \@ref(fig:angelaphyloarchi) & Fig. \@ref(fig:sixtophyloarchi)). 

```{r angelaphyloarchi, fig.cap="Caption."}
dendextend::dendlist(phylogram::as.dendrogram(ape::read.tree(text = "((((AS:3.83,AL:6.04)A:6.44,(BS:3.34,BL:10.64)B:1.6)AB:6.5,((CS:3.8,CL:5.13)C:14.45,(DS:3.5,DL:7.76)D:6.2)CD:1.7,(ES:0,EL:2.2)E:3.7)X:26.5)T:1.3;")), 
                     phylogram::as.dendrogram(phytools::midpoint.root(ape::chronos(angela_phylo)))) %>%
  dendextend::untangle(method = "ladderize") %>%
  dendextend::tanglegram(sort = T, common_subtrees_color_branches = TRUE,
                         main_right = "Angela - Phylogeny - EVS", main_left = " Angela - Architecture")
```

```{r sixtophyloarchi, fig.cap="Caption."}
dendextend::dendlist(phylogram::as.dendrogram(ape::read.tree(text =  "(((((B1L:5.2,(C3L:4.47, C1S:0.75)C:0.4)XC:3.08,(E2L:3.8, E4S:3.5)E:1.6)XE:0.6,F2S:4.37)XF:1.6,(I1L:3.3, I2S:1.15)I:1.1)XI:28.64)T:1.3;")), 
                     phylogram::as.dendrogram(phytools::midpoint.root(ape::chronos(sixto_phylo)))) %>%
  dendextend::untangle(method = "ladderize") %>%
  dendextend::tanglegram(sort = T, common_subtrees_color_branches = TRUE,
                         main_right = "Sixto - Phylogeny - EVS", main_left = " Sixto - Architecture")
```

## Light

The light condition of the sampled tips at the time of sampling did not affect the number of mutations observed per library (Fig. \@ref(fig:lightComp2)). 

```{r lightComp2, fig.cap="Caption."}
list(Angela = vroom::vroom("data/mutations/angela//filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  dplyr::select(tree, tip, tumor, Filter, SNV) %>% 
  unique() %>% 
  group_by(tree, tip, tumor, Filter) %>% 
  summarise(mutations = n()) %>% 
  mutate(condition = recode(tip, "L" = "Light", "S" = "Shadow")) %>% 
  ggplot(aes(tree, mutations, fill = condition)) +
  geom_boxplot() +
  facet_wrap(~Filter, scales = "free") +
  ggpubr::stat_compare_means(label = "p.signif", method = "t.test") +
  xlab("") + ylab("Number of mutations")
```

```{r, eval=F}
list(Angela = vroom::vroom("data/mutations/angela//filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  dplyr::select(tree, tip, tumor, Filter, SNV) %>% 
  unique() %>% 
  group_by(tree, tip, tumor, Filter) %>% 
  summarise(mutations = n()) %>% 
  mutate(condition = recode(tip, "L" = "Light", "S" = "Shadow")) %>% 
  filter(Filter == "base") %>% 
  ggplot(aes(tree, mutations, fill = condition)) +
  geom_boxplot() +
  ggpubr::stat_compare_means(label = "p.signif", method = "t.test") +
  xlab("") + ylab("Number of mutations")
```

## Type

Mutation types are similar between Angela and Sixto, with the exception of an increase in C->A and C->T but a decrease in T->A in Sixto compared to Angela, which are probably due to the sampling effect (Fig. \@ref(fig:typeComp2), to be further investigated).

```{r typeComp2, fig.cap="Caption."}
list(Angela = vroom::vroom("data/mutations/angela/filters/filtered_tip_mutations_spectra.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/filtered_tip_mutations_spectra.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(type, tumor, tree) %>% 
  summarise(N = n()) %>% 
  group_by(tree, tumor) %>% 
  mutate(pct = N/sum(N)*100) %>% 
  ggplot(aes(type, pct, fill = tree)) +
  geom_boxplot(position = "dodge") + 
  theme(axis.text.x = element_blank(), axis.title.x = element_blank(),
        axis.line.x = element_blank(), axis.ticks.x = element_blank(),
        legend.position = "bottom") +
  facet_wrap(~ type, nrow = 1, scales = "free_x") +
  xlab("Percentage of mutations (%) per sample") +
  ggpubr::stat_compare_means(label = "p.signif", method = "t.test") +
  ylab("Percentage of mutations") +
  ggtitle("Mutation type per library", "At least in 2-leaves with evs filtering.")
```

## Spectra

The mutation spectra are similar between Angela and Sixto, with a few exceptions that are probably due to a sampling effect (Fig. \@ref(fig:spectraComp2), to be explored further). 

```{r spectraComp2, fig.cap="Caption."}
list(Angela = vroom::vroom("data/mutations/angela/filters/filtered_tip_mutations_spectra.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/filtered_tip_mutations_spectra.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(type, spectra, tumor, tree) %>% 
  summarise(N = n()) %>% 
  group_by(tree, tumor) %>% 
  mutate(pct = N/sum(N)*100) %>% 
  ggplot(aes(spectra, pct, fill = tree)) +
  geom_boxplot(position = "dodge") + 
  facet_wrap(~ type, scales = "free_x") +
  theme(axis.text.x = element_text(angle = 90), legend.position = "bottom", axis.title.x = element_blank()) +
  theme(panel.spacing = unit(0, "lines"), panel.border = element_rect(size = 0.25, color = "black", fill = NA)) +
  scale_y_log10() +
  # ggpubr::stat_compare_means(label = "p.signif") +
  ylab("Percentage of mutations per library") + ggtitle("Mutation spectra", "At least in 2-leaves with evs filtering.")
```

## Rates

Angela and Sixto show ten to ten thousand mutations depending on the filtering and the minimum accepted allelic frequency (Fig. \@ref(fig:ratesComp2)). Using base filtering, Angela has a higher number of mutations than Sixto, due to the twofold sequencing depth. Nevertheless, ev filtering gave a similar number of mutations close to thousands in both trees.

```{r ratesComp2, fig.cap="Caption."}
list(Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>% 
  bind_rows(.id = "tree") %>%
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  group_by(tree, Filter) %>% 
  arrange(desc(mutation_AF)) %>% 
  mutate(mutations = 1) %>% 
  mutate(cummulated_mutations = cumsum(mutations)) %>% 
  ggplot(aes(mutation_AF, cummulated_mutations, col = tree, linetype = Filter)) +
  geom_line() +
  scale_y_log10() + scale_x_reverse() +
  xlab("Allelic frequency") + ylab("Cummulated mutations with decreasing AF")
```

## Annotations

*In progress.*

## Genes

*In progress.*

```{r}
list(
  angela = data.frame(type = c("all", "transcripts with genes", "within CDS", "non-synonymous"),
           count = c(15468, 2069, 567,314)),
  sixto = data.frame(type = c("all", "transcripts with genes", "within CDS", "non-synonymous"),
           count = c(3223, 210, 31, 9))
) %>% bind_rows(.id = "tree") %>% 
  mutate(type = factor(type, levels = c("all", "transcripts with genes", "within CDS", "non-synonymous"))) %>% 
  ggplot(aes(type, count, fill = tree)) +
  geom_col(position = "dodge") +
  coord_flip() +
  # scale_y_log10() +
  geom_text(aes(y = count+100, label = format(count, big.mark = " "), group = tree), 
            position = position_dodge(width = .9), col = "black") +
  xlab("")
  # ggtitle("Mutations")
```

```{r}
data <- list(angela = read_tsv("save/mutations_angela_sixto.tsv"), 
     sixto = read_tsv("save/mutations_annotated_sixto.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  filter(CDS == 1) %>% 
  filter(mutation_AF > 0.01) %>% 
  dplyr::select(tree, SNV, mutation_AF, nonsynonymous) %>% 
  mutate(synonymy = ifelse(is.na(nonsynonymous), "S", "NS")) %>% 
  dplyr::select(tree, SNV, synonymy, mutation_AF) 
aov(lm(log(mutation_AF) ~ tree + synonymy, data)) %>% summary()
ggplot(data, aes(tree, mutation_AF, fill = synonymy)) +
  geom_violin() +
  ggtitle("Anova: log(AF) ~ tree + synonymy", "tree p < 2e-16, synonymy p = 1.47e-14") +
   ylab("Allelic fraction") +
  scale_fill_manual(values = c("firebrick", "darkgreen"))
```


## Fruits

### From SSRseq Olivier

#### Align genotypes

```{r, eval=F, echo=T}
library(Biostrings)
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
alleles <- readxl::read_xlsx("data/mutations/fruits/SSRseq_TreeMutation_DataAnalysis.xlsx", "AlleleInformation") %>% 
  dplyr::rename(locus = Locus, genotype = AlleleSeqCode, sequence = AlleleSequence) %>% 
  dplyr::select(locus, genotype, sequence) %>% 
  mutate(locus2 = gsub("TreeMut_SNP-", "", locus)) %>% 
  separate(locus2, c("mutation", "pos"), "_Super-Scaffold_", convert = T) %>% 
  separate_rows(mutation, convert = T) %>% 
  separate(pos, "CHROM", convert = T) %>% 
  left_join(candidates) %>% 
  mutate(name = paste0("SNV", sprintf("%03d", mutation), "_A", genotype))
angela_alleles <- DNAStringSet(filter(alleles, tree == "Angela")$sequence)
names(angela_alleles) <- filter(alleles, tree == "Angela")$name
writeXStringSet(angela_alleles, "data/mutations/fruits/angela_alleles.fa")
sixto_alleles <- DNAStringSet(filter(alleles, tree == "Sixto")$sequence)
names(sixto_alleles) <- filter(alleles, tree == "Sixto")$name
writeXStringSet(sixto_alleles, "data/mutations/fruits/sixto_alleles.fa")
refs <- bind_rows(readxl::read_xlsx("data/mutations/fruits/treemutation_fruits.xlsx", "Sixto") %>% 
            mutate(tree = "Sixto"),
          readxl::read_xlsx("data/mutations/fruits/treemutation_fruits.xlsx", "Angela") %>% 
            mutate(tree = "Angela")) %>% 
  mutate(library = paste0(Code_Espece, Tube), tissue = Tissu) %>%
  dplyr::select(library, tree, tissue) %>% 
  bind_rows(readxl::read_xlsx("data/mutations/fruits/Angela.xlsx", "libraries") %>% 
              mutate(tissue = paste("branch", branch), library = idOld, tree = "Angela") %>% 
              dplyr::select(library, tree, tissue)) %>% 
  bind_rows(readxl::read_xlsx("data/mutations/fruits/Sixto.xlsx", "samples") %>% 
              mutate(tissue = paste("branch", Branch), library = id, tree = "Sixto") %>% 
              dplyr::select(library, tree, tissue))
```

```{bash, eval=F, echo=T}
bwa mem -t 2 ../angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa angela_alleles.fa | samtools sort > angela_alleles_aligned.bam
samtools index angela_alleles_aligned.bam
bwa mem -t 2 ../sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa sixto_alleles.fa | samtools sort > sixto_alleles_aligned.bam
samtools index sixto_alleles_aligned.bam
```

#### Automate IGV

```{bash installigvreports, eval=F, echo=T}
conda create -n igvreports python=3.7.1
conda activate igvreports
pip install igv-reports
conda deactivate
```

```{bash igvreportsangela2, eval=F, echo=T}
conda activate igvreports
create_report data/mutations/fruits/angela_fruits_candidate_mutations.tsv \
data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa \
--sequence 2 --begin 3 --end 3 \
--flanking 1000 \
--info-columns SNV CHROM POS REF ALT af replicate branch rank \
--tracks data/mutations/fruits/angela_alleles_aligned.bam data/mutations/angela/annotation/Dgu_HS1_HYBRID_SCAFFOLD.fa.out.gff \
--output data/mutations/fruits/angela_fruits_aligned.html
conda deactivate
```

```{bash igvreportssixto2, eval=F, echo=T}
conda activate igvreports
create_report data/mutations/fruits/sixto_fruits_candidate_mutations.tsv \
data/mutations/sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa \
--sequence 2 --begin 3 --end 3 \
--flanking 1000 \
--info-columns SNV CHROM POS REF ALT af replicate branch rank \
--tracks data/mutations/fruits/sixto_alleles_aligned.bam data/mutations/sixto/annotation/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa.out.gff \
--output data/mutations/fruits/sixto_fruits_aligned.html
conda deactivate
```

#### Results

```{r}
readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(Tree = ifelse(as.numeric(gsub("SNV", "", SNV)) > 124, "Sixto", "Angela")) %>% 
  group_by(Tree, Type) %>%
  summarise(N = n()) %>% 
  reshape2::dcast(Type ~ Tree) %>% 
  mutate(Total = Angela + Sixto) %>% 
  mutate(Percentage = round(Total/160*100))
```


```{r datafruitsolivier}
refs <- bind_rows(readxl::read_xlsx("data/mutations/fruits/treemutation_fruits.xlsx", "Sixto") %>% 
            mutate(tree = "Sixto"),
          readxl::read_xlsx("data/mutations/fruits/treemutation_fruits.xlsx", "Angela") %>% 
            mutate(tree = "Angela")) %>% 
  mutate(library = paste0(Code_Espece, Tube), tissue = Tissu) %>%
  dplyr::select(library, tree, tissue, Echantillon) %>% 
  bind_rows(readxl::read_xlsx("data/mutations/fruits/Angela.xlsx", "libraries") %>% 
              mutate(tissue = paste("branch", branch), library = idOld, tree = "Angela") %>% 
              dplyr::select(library, tree, tissue)) %>% 
  bind_rows(readxl::read_xlsx("data/mutations/fruits/Sixto.xlsx", "samples") %>% 
              mutate(tissue = paste("branch", Branch), library = id, tree = "Sixto") %>% 
              dplyr::select(library, tree, tissue))
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
mutations <- readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "mutations")
geno <- readxl::read_xlsx("data/mutations/fruits/SSRseq_TreeMutation_DataAnalysis.xlsx", "Genotypes", skip = 10)
loci_names <- names(geno)[-1][seq(1, ncol(geno)-2, 2)]
names(geno)[1+seq(1, ncol(geno)-2, 2)] <- paste0(loci_names, "_Locus1")
names(geno)[2+seq(1, ncol(geno)-2, 2)] <- paste0(loci_names, "_Locus2")
```

```{r allfruitsolivier}
all <- geno %>% 
  reshape2::melt(id.vars = "Indiv", variable.name = "locus", value.name = "genotype") %>% 
  mutate(genotype = as.numeric(ifelse(genotype == "NA", NA, genotype))) %>%
  filter(!is.na(genotype)) %>% 
  mutate(locus = gsub("TreeMut_SNP-", "", locus)) %>% 
  separate(locus, "SNV", "_Super-Scaffold_") %>% 
  separate_rows(SNV, convert = T) %>% 
  mutate(SNV =  paste0("SNV", sprintf("%03d", SNV))) %>% 
  dplyr::rename(Genotype = genotype) %>% 
  left_join(mutations) %>% 
  filter(!is.na(Base)) %>% 
  separate(Indiv, c("library"), "_[[:alpha:]]$", remove = F) %>% 
  left_join(refs) %>% 
  filter(!is.na(tissue)) %>% 
  left_join(mutate(candidates, SNV =  paste0("SNV", sprintf("%03d", mutation))))
stats <- all %>% 
  group_by(tree, SNV, library, tissue, af, branch, REF, ALT) %>% 
  summarise(type = paste0(unique(sort(Base)), collapse = ", ")) %>% 
  group_by(tree, SNV, af, branch, REF, ALT, tissue) %>% 
  mutate(Ntot = n()) %>% 
  group_by(tree, SNV, af, branch, REF, ALT, tissue, Ntot, type) %>% 
  summarise(N = n()) %>% 
  mutate(pct = N/Ntot*100) %>% 
  mutate(title = paste0(SNV, " ", REF, ">", ALT, " AF=", round(af, 2), " branches=", branch))
```

```{r graphfruitsolivier}
graph <- stats %>% 
  ggplot(aes(paste(tree, tissue), y = pct, fill = type)) +
  geom_col() + 
  geom_text(aes(y = 5, label = Ntot), col = "white") +
  ggforce::facet_wrap_paginate(~ title, scales = "free", nrow = 1, ncol = 1) +
  coord_flip() +
  ylab("Percentage") + xlab("")
```

```{r saveallgraphfruitsolivier, eval=F, echo=T}
pdf( "data/mutations/fruits/fruits_mutations.pdf")
for(i in 1:ggforce::n_pages(graph))
  print(graph + ggforce::facet_wrap_paginate(~title, scales = "free",
                                         ncol = 1, nrow = 1, page = i))
dev.off()
```

```{r exgraphfruitsolivier}
graph + ggforce::facet_wrap_paginate(~title, scales = "free",
                                     ncol = 1, nrow = 1, page = 1)
```

```{r snv045, eval=F}
samples <- all %>% 
  filter(SNV == "SNV045") %>% 
  filter(!is.na(Echantillon)) %>% 
  group_by(tree, Echantillon) %>% 
  summarise(type = paste0(unique(sort(Base)), collapse = ", ")) %>% 
  filter(type == "T") %>% 
  separate(Echantillon, c("species", "branch_fruit", "sample_fruit", "tissue_short")) %>% 
  mutate(fruit = paste0(branch_fruit, sample_fruit))
  # only branch B !
all %>% 
  separate(Echantillon, c("species", "branch_fruit", "sample_fruit", "tissue_short")) %>% 
  mutate(fruit = paste0(branch_fruit, sample_fruit)) %>% 
  filter(fruit %in% samples$fruit) %>% 
  group_by(fruit, tissue, SNV) %>% 
  summarise(type = ifelse(length(unique(Base)) == 1, "homozygote", "heterozygote")) %>% 
  group_by(fruit, tissue, type) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(fruit + tissue ~ type) %>% 
  filter(tissue != "endocarpe") %>% 
  mutate(heterozygote = ifelse(is.na(heterozygote), 0, heterozygote)) %>% 
  mutate(homozygosity = homozygote / (heterozygote + homozygote) * 100) %>% 
  ggplot(aes(homozygosity)) +
  geom_density() +
  xlim(0, 100) +
  ggtitle('Fruits with SNV045 homozygosity across mutations',
          '5/16 non 100% homozygote')
```
#### Tissues

```{r}
fruits <- all %>% 
  separate(Echantillon, c("species", "branch_fruit", "sample_fruit", "tissue_short")) %>% 
  mutate(fruit = paste0(branch_fruit, sample_fruit)) %>% 
  filter(!is.na(branch_fruit)) %>% 
  filter(tissue != "endocarpe") %>% 
  group_by(SNV, af, branch, branch_fruit, fruit, tissue) %>% 
  summarise(genotype = paste0(unique(sort(Base)), collapse = "")) %>% 
  mutate(type = ifelse(nchar(genotype) == 2, "heterozygous", "homozygous"))
```

```{r}
# branch coherence
fruits %>% 
  filter(type == "heterozygous") %>% 
  rowwise() %>% 
  mutate(branches = list(str_split(branch, ", ")[[1]])) %>% 
  filter(!(branch_fruit %in% branches)) %>% 
  filter(SNV != "SNV128", SNV != "SNV132", SNV != "SNV151", SNV != "SNV160",) %>% 
  group_by(SNV, branch) %>% 
  summarise(branch_fruit = paste(unique(branch_fruit, collapse = ",")),
            fruits = length(unique(fruit))) %>% 
  left_join(fruits %>% 
              dplyr::select(SNV, fruit) %>% 
              unique() %>% 
              group_by(SNV) %>% 
              summarise(total = n())) %>% 
  mutate(percentage = round(fruits / total * 100))
```

```{r}
# tissue coherence
agreement <- fruits %>%
  filter(tissue %in% c("embryo sac", "cotyledon")) %>% 
  reshape2::dcast(SNV + fruit ~ tissue, value.var = "type") %>% 
  mutate(agreement = !(cotyledon != `embryo sac`)) %>% 
  mutate(agreement = ifelse(is.na(agreement), T, agreement)) %>% 
  group_by(SNV, agreement) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(SNV ~ agreement) %>% 
  mutate(`FALSE` = ifelse(is.na(`FALSE`), 0, `FALSE`)) %>% 
  mutate(agreement = round(`TRUE` / (`TRUE` + `FALSE`) * 100))
```

```{r}
ggplot(agreement, aes(agreement)) + 
  geom_histogram() +
  xlab("Agreement cotyledon & embryon (%)")
```

```{r}
filter(agreement, agreement < 100)
```

#### Transmission

```{r}
transmission <- fruits %>% 
  filter(tissue == "embryo sac") %>% 
  group_by(SNV, af, type) %>% 
  summarise(N = n()) %>% 
  reshape2::dcast(SNV + af ~ type) %>% 
  mutate(heterozygous = ifelse(is.na(heterozygous), 0, heterozygous)) %>% 
  mutate(transmission = heterozygous / (heterozygous+homozygous))
transmission %>% 
  filter(transmission > 0) %>% 
  ggplot(aes(af, transmission)) +
  geom_smooth(method = "lm") +
  geom_point(size = 2) +
  ggrepel::geom_text_repel(aes(label = SNV)) +
  scale_x_log10() +
  scale_y_log10() +
  xlab("Mean alleleic fraction in branches") +
  ylab("Transmission rate across embryons")
```

```{r}
readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(Tree = ifelse(as.numeric(gsub("SNV", "", SNV)) > 124, "Sixto", "Angela")) %>% 
  left_join(candidates %>% 
              mutate(SNV = paste0("SNV", sprintf("%03d", mutation))) %>% 
              dplyr::select(SNV, af)) %>% 
  filter(Type %in% c("reference only", "mutation")) %>% 
  left_join(transmission %>% 
              filter(transmission > 0) %>% 
              mutate(type = "transmitted") %>% 
              dplyr::select(SNV, type)) %>% 
  mutate(type = ifelse(is.na(type), "untransmitted", type)) %>% 
  ggplot(aes(type, af, fill = type)) +
  geom_boxplot() +
  scale_y_log10() +
  coord_flip() +
  ggtitle("Wilcoxon test, p = 0.2") +
  xlab("") +
  ylab("Mean alleleic fraction in branches") +
  scale_fill_discrete(guide = "none")
```

### Old

```{r}
list(
  angela = data.frame(type = c("all", "transcripts with genes", "within CDS", "non-synonymous", "GO-enriched"),
           count = c(152, 41, 19, 17, 0)),
  sixto = data.frame(type = c("all", "transcripts with genes", "within CDS", "non-synonymous", "GO-enriched"),
           count = c(36, 0, 0, 0, 0))
) %>% bind_rows(.id = "tree") %>% 
  mutate(type = factor(type, levels = c("all", "transcripts with genes", "within CDS", "non-synonymous", "GO-enriched"))) %>% 
  ggplot(aes(type, count, fill = tree)) +
  geom_col(position = "dodge") +
  coord_flip() +
  geom_text(aes(y = 2, label = format(count, big.mark = " "), group = tree), 
            position = position_dodge(width = .9), col = "white") +
  xlab("") +
  ggtitle("Fruit mutations")
```




```{r, eval=F}
library(tidyverse)
library(googlesheets4)
theme_set(bayesplot::theme_default())
nano <- list(
  Sixto = read_sheet(ss = "https://docs.google.com/spreadsheets/d/1w9KauZ_XFGkEZoZcLbVywD1UK2FwJMtmzoNymsLEc4k/edit?usp=sharing", sheet = "Sixto_Nano"),
  Angela = read_sheet(ss = "https://docs.google.com/spreadsheets/d/1w9KauZ_XFGkEZoZcLbVywD1UK2FwJMtmzoNymsLEc4k/edit?usp=sharing", sheet = "Angela_Nano")
) %>% bind_rows(.id = "Tree")
nano %>% 
  mutate(A260_A280_ratio = "ok") %>% 
  mutate(A260_A280_ratio = ifelse(A260_A280 < 1.8, "low", A260_A280_ratio)) %>% 
  mutate(A260_A280_ratio = ifelse(A260_A280 > 2, "high", A260_A280_ratio)) %>% 
  mutate(ADN_ng_uL = ifelse(ADN_ng_uL < 0, 0, ADN_ng_uL)) %>% 
  ggplot(aes(ADN_ng_uL, fill = A260_A280_ratio)) +
  geom_histogram() +
  facet_wrap(~ Tree, scales = "free") +
  scale_fill_manual(expression(frac(A260,A280)), values = c("darkorange", "brown2", "chartreuse3")) +
  xlab(expression(ADN~ng.µL^-1))
```

