```{r setupcomp5, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
# library(tidyverse)
library(readr)
library(dplyr)
library(tidyr)
library(ggplot2)
library(patchwork)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

```{r coveragecomp5, eval=F}
covs_files <- list.files("data/mutations/bordeaux/", pattern = "10k.cov", full.names = T)
names(covs_files) <- list.files("data/mutations/bordeaux/", pattern = "10k.cov", full.names = F)
covs_3p <- lapply(covs_files, read_tsv, col_names = c("chromosome", "start", "stop", "coverage"), cols(
  chromosome = col_character(),
  start = col_double(),
  stop = col_double(),
  coverage = col_double()
)) %>%
  bind_rows(.id = "file") %>%
  mutate(library = gsub(".10k.cov", "", file)) %>% 
  mutate(length = stop - start)
covs_3p <- lapply(split(covs_3p, covs_3p$library), function(l)
       lapply(0:500, function(c) 
         data.frame(coverage = c,
                    proportion = sum(filter(l, coverage >= c)$length)/sum(l$length))
         ) %>% bind_rows()) %>% 
  bind_rows(.id = "library")

covs_files <- list.files("data/mutations/hetre/", pattern = "Fagus_sylvatica_v3.regions.bed", full.names = T)
names(covs_files) <- list.files("data/mutations/hetre/", pattern = "Fagus_sylvatica_v3.regions.bed", full.names = F)
covs_verzy <- lapply(covs_files, read_tsv, col_names = c("chromosome", "start", "stop", "coverage"), cols(
  chromosome = col_character(),
  start = col_double(),
  stop = col_double(),
  coverage = col_double()
)) %>%
  bind_rows(.id = "file") %>%
  mutate(library = gsub("_on_Fagus_sylvatica_v3.regions.bed", "", file)) %>% 
  mutate(length = stop - start)
covs_verzy <- lapply(split(covs_verzy, covs_verzy$library), function(l)
       lapply(0:500, function(c) 
         data.frame(coverage = c,
                    proportion = sum(filter(l, coverage >= c)$length)/sum(l$length))
         ) %>% bind_rows()) %>% 
  bind_rows(.id = "library")

covs_files <- list.files("data/mutations/angela/genome/dists/", full.names = T)
names(covs_files) <- list.files("data/mutations/angela/genome/dists/")
covs_angela <- lapply(covs_files, read_tsv, col_names = c("chromosome", "coverage", "proportion"), cols(
  chromosome = col_character(),
  coverage = col_double(),
  proportion = col_double()
)) %>% 
  bind_rows(.id = "file") %>% 
  mutate(library = gsub(".mosdepth.global.dist.txt", "", file)) %>% 
  filter(chromosome == "total") %>% 
  dplyr::select(library, coverage, proportion)

covs_files <- list.files("data/mutations/sixto/genome/dists/", full.names = T)
names(covs_files) <- list.files("data/mutations/sixto/genome/dists/")
covs_sixto <- lapply(covs_files, read_tsv, col_names = c("chromosome", "coverage", "proportion"), cols(
  chromosome = col_character(),
  coverage = col_double(),
  proportion = col_double()
)) %>%
  bind_rows(.id = "file") %>%
  mutate(library = gsub(".mosdepth.global.dist.txt", "", file)) %>%
  filter(chromosome == "total") %>% 
  dplyr::select(library, coverage, proportion)

list(
  # "Napoleon" = NA,
  "3P" = covs_3p,
  "Verzy" = covs_verzy,
  "Sixto" = covs_sixto, 
  "Angela" = covs_angela) %>% 
  bind_rows(.id = "tree") %>% 
  vroom::vroom_write("save/coverages_comparisons.tsv")
```

```{r filteringComp5, eval=F}
# covlims
coverages <- vroom::vroom("save/coverages_comparisons.tsv")
cov_lims <- coverages %>% 
  mutate(proportion = round(proportion, 2)) %>% 
  mutate(proportion = ifelse(tree == "Verzy" & proportion == 0.94, 0.95, proportion)) %>% 
  filter(proportion == 0.05 | proportion == 0.95) %>% 
  group_by(tree, library, proportion) %>% 
  summarise(coverage = mean(coverage, na.omit = T)) %>% 
  mutate(proportion = recode(as.character(proportion), "0.05" = "high_cov", "0.95" = "low_cov")) %>% 
  reshape2::dcast(tree + library ~ proportion) %>% 
  bind_rows(data.frame(tree = "Napoleon", library = c("upper", "lower"), high_cov = 120, low_cov = 2)) %>% 
  mutate(low_cov = ifelse(low_cov < 10, 10, low_cov))
# napo
napo <- src_sqlite("data/mutations/swiss/strelka2_raw.sql") %>% tbl("mutations")
napo_mutations <- napo %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  filter(normal_altCountT1 == 0, normal_altCountT2 == 0) %>% 
  filter(mutation_altCountT1 >= 5, mutation_altCountT2 >= 5) %>% 
  collect() %>% 
  lazy_dt() %>% 
  left_join(rename(cov_lims, tumor = library, mutation_high_cov = high_cov, mutation_low_cov = low_cov)) %>% 
  left_join(rename(cov_lims, normal = library, normal_high_cov = high_cov, normal_low_cov = low_cov)) %>% 
  filter(normal_DP <= normal_high_cov & normal_DP >= normal_low_cov & mutation_DP <= mutation_high_cov & mutation_DP >= mutation_low_cov) %>%
  mutate(Filter = ifelse(FILTER == "PASS", "evs", "base")) %>% 
  dplyr::select(-ChromKey, -NT, -ID, -QUAL, -FILTER, -SOMATIC, -PNOISE, -PNOISE2, -normal, -caller, 
                -mutation_high_cov, -mutation_low_cov, -normal_high_cov, -normal_low_cov) %>% 
  group_by(tumor, SNV, Filter, CHROM, POS, REF, ALT, SGT) %>% 
  summarise_all(median)
vroom::vroom_write(as_tibble(napo_mutations), "data/mutations/swiss/leaf_mutations.tsv")
# 3p
bdx <- src_sqlite("data/mutations/bordeaux/strelka2_raw.sql") %>% tbl("mutations")
bdx_mutations <- bdx %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  filter(normal_altCountT1 == 0, normal_altCountT2 == 0) %>% 
  filter(mutation_altCountT1 >= 5, mutation_altCountT2 >= 5) %>% 
  collect() %>% 
  lazy_dt() %>% 
  left_join(rename(cov_lims, tumor = library, mutation_high_cov = high_cov, mutation_low_cov = low_cov)) %>% 
  left_join(rename(cov_lims, normal = library, normal_high_cov = high_cov, normal_low_cov = low_cov)) %>% 
  filter(normal_DP <= normal_high_cov & normal_DP >= normal_low_cov & mutation_DP <= mutation_high_cov & mutation_DP >= mutation_low_cov) %>%
  mutate(Filter = ifelse(FILTER == "PASS", "evs", "base")) %>% 
  dplyr::select(-ChromKey, -NT, -ID, -QUAL, -FILTER, -SOMATIC, -PNOISE, -PNOISE2, -normal, -caller, 
                -mutation_high_cov, -mutation_low_cov, -normal_high_cov, -normal_low_cov) %>%
  as_tibble() %>% 
  group_by(tree, tumor, SNV, Filter, CHROM, POS, REF, ALT, SGT) %>% 
  summarise_all(median)
vroom::vroom_write(as_tibble(bdx_mutations), "data/mutations/bordeaux/leaf_mutations.tsv")
# Verzy
verzy <- src_sqlite("data/mutations/hetre/mutations.raw.sql") %>% tbl("mutations")
verzy_mutations <- verzy %>% 
  filter(reference == "Fagus_sylvatica_v3") %>% 
  mutate(SNV = paste0(CHROM, "#", as.integer(POS))) %>% 
  filter(normal_altCountT1 == 0, normal_altCountT2 == 0) %>% 
  filter(mutation_altCountT1 >= 5, mutation_altCountT2 >= 5) %>% 
  collect() %>% 
  lazy_dt() %>% 
  left_join(rename(cov_lims, tumor = library, mutation_high_cov = high_cov, mutation_low_cov = low_cov)) %>% 
  left_join(rename(cov_lims, normal = library, normal_high_cov = high_cov, normal_low_cov = low_cov)) %>% 
  filter(normal_DP <= normal_high_cov & normal_DP >= normal_low_cov & mutation_DP <= mutation_high_cov & mutation_DP >= mutation_low_cov) %>%
  mutate(Filter = ifelse(FILTER == "PASS", "evs", "base")) %>% 
  dplyr::select(-ChromKey, -NT, -ID, -QUAL, -FILTER, -SOMATIC, -PNOISE, -PNOISE2, -normal, -caller, 
                -mutation_high_cov, -mutation_low_cov, -normal_high_cov, -normal_low_cov) %>%
  as_tibble() %>% 
  group_by(tree, tumor, SNV, Filter, CHROM, POS, REF, ALT, SGT) %>% 
  summarise_all(median)
vroom::vroom_write(as_tibble(verzy_mutations), "data/mutations/hetre/leaf_mutations.tsv")
rm(list = ls()) ; invisible(gc())
```

```{r spectraComp5data, eval=F}
# libs
library(Biostrings)
library(genomeIntervals)
# oak
genome <- readDNAStringSet("data/mutations/bordeaux/Qrob_PM1N.fa")
genome_index <- read_tsv("data/mutations/bordeaux/Qrob_PM1N.fa.fai",
                         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length)
# napo
mutations <- vroom::vroom("data/mutations/swiss/leaf_mutations.tsv") %>% 
  filter(Filter == "evs") %>% 
  group_by(SNV, CHROM, POS, REF, ALT) %>% 
  unique() %>% 
  ungroup()
seqs <- lapply(1:nrow(mutations), function(i)
  as.character(unlist(extractAt(genome[mutations$CHROM[i]], IRanges(mutations$POS[i]-1, width=3)))))
t <- mutations %>% 
  mutate(spectra = unlist(seqs)) %>% 
  mutate(REF5 = str_sub(spectra, 1, 1), REFspectra = str_sub(spectra, 2, 2), REF3 = str_sub(spectra, 3, 3)) %>% 
  dplyr::select(-REFspectra) %>% 
  mutate(type = paste0(REF, "->", ALT)) %>% 
  mutate(strand = recode(type,
                         "G->T" = "-",
                         "C->A" = "+",
                         "G->C" = "-",
                         "C->G" = "+",
                         "G->A" = "-",
                         "C->T" = "+",
                         "A->T" = "-",
                         "T->A" = "+",
                         "A->G" = "-",
                         "T->C" = "+",
                         "A->C" = "-",
                         "T->G" = "+"
                       )) %>%
  mutate(type = recode(type,
                       "G->T" = "C->A",
                       "G->C" = "C->G",
                       "G->A" = "C->T",
                       "A->T" = "T->A",
                       "A->G" = "T->C",
                       "A->C" = "T->G"
  )) %>%
  mutate(POS5 = REF5, POS3 = REF3) %>% 
  mutate(POS5r = recode(POS5, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(REFr = recode(REF, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS3r = recode(POS3, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS5f = ifelse(strand == "+", POS5, POS3r)) %>%
  mutate(REFf = ifelse(strand == "+", REF, REFr)) %>%
  mutate(POS3f = ifelse(strand == "+", POS3, POS5r)) %>% 
  mutate(spectra = paste0(POS5f, REFf, POS3f)) %>% 
  dplyr::select(CHROM, POS, SNV, REF, ALT, tumor, mutation_AF, Filter, type, spectra)
write_tsv(t, "data/mutations/swiss/leaf_mutations_spectra.tsv") 
# 3p
mutations <- vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv") %>% 
  filter(Filter == "evs") %>% 
  group_by(SNV, CHROM, POS, REF, ALT) %>% 
  unique() %>% 
  ungroup()
seqs <- lapply(1:nrow(mutations), function(i)
  as.character(unlist(extractAt(genome[mutations$CHROM[i]], IRanges(mutations$POS[i]-1, width=3)))))
t <- mutations %>% 
  mutate(spectra = unlist(seqs)) %>% 
  mutate(REF5 = str_sub(spectra, 1, 1), REFspectra = str_sub(spectra, 2, 2), REF3 = str_sub(spectra, 3, 3)) %>% 
  dplyr::select(-REFspectra) %>% 
  mutate(type = paste0(REF, "->", ALT)) %>% 
  mutate(strand = recode(type,
                         "G->T" = "-",
                         "C->A" = "+",
                         "G->C" = "-",
                         "C->G" = "+",
                         "G->A" = "-",
                         "C->T" = "+",
                         "A->T" = "-",
                         "T->A" = "+",
                         "A->G" = "-",
                         "T->C" = "+",
                         "A->C" = "-",
                         "T->G" = "+"
                       )) %>%
  mutate(type = recode(type,
                       "G->T" = "C->A",
                       "G->C" = "C->G",
                       "G->A" = "C->T",
                       "A->T" = "T->A",
                       "A->G" = "T->C",
                       "A->C" = "T->G"
  )) %>%
  mutate(POS5 = REF5, POS3 = REF3) %>% 
  mutate(POS5r = recode(POS5, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(REFr = recode(REF, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS3r = recode(POS3, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS5f = ifelse(strand == "+", POS5, POS3r)) %>%
  mutate(REFf = ifelse(strand == "+", REF, REFr)) %>%
  mutate(POS3f = ifelse(strand == "+", POS3, POS5r)) %>% 
  mutate(spectra = paste0(POS5f, REFf, POS3f)) %>% 
  dplyr::select(CHROM, POS, SNV, REF, ALT, tumor, mutation_AF, Filter, type, spectra)
write_tsv(t, "data/mutations/bordeaux/leaf_mutations_spectra.tsv") 
# verzy
genome <- readDNAStringSet("data/mutations/hetre/Fagus_sylvatica_v3.fa")
genome_index <- read_tsv("data/mutations/hetre/Fagus_sylvatica_v3.fa.fai",
                         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, end = length) %>% 
  dplyr::select(-length)
mutations <- vroom::vroom("data/mutations/hetre/leaf_mutations.tsv") %>% 
  filter(Filter == "evs") %>% 
  group_by(SNV, CHROM, POS, REF, ALT) %>% 
  unique() %>% 
  ungroup()
seqs <- lapply(1:nrow(mutations), function(i)
  as.character(unlist(extractAt(genome[mutations$CHROM[i]], IRanges(mutations$POS[i]-1, width=3)))))
t <- mutations %>% 
  mutate(spectra = unlist(seqs)) %>% 
  mutate(REF5 = str_sub(spectra, 1, 1), REFspectra = str_sub(spectra, 2, 2), REF3 = str_sub(spectra, 3, 3)) %>% 
  dplyr::select(-REFspectra) %>% 
  mutate(type = paste0(REF, "->", ALT)) %>% 
  mutate(strand = recode(type,
                         "G->T" = "-",
                         "C->A" = "+",
                         "G->C" = "-",
                         "C->G" = "+",
                         "G->A" = "-",
                         "C->T" = "+",
                         "A->T" = "-",
                         "T->A" = "+",
                         "A->G" = "-",
                         "T->C" = "+",
                         "A->C" = "-",
                         "T->G" = "+"
                       )) %>%
  mutate(type = recode(type,
                       "G->T" = "C->A",
                       "G->C" = "C->G",
                       "G->A" = "C->T",
                       "A->T" = "T->A",
                       "A->G" = "T->C",
                       "A->C" = "T->G"
  )) %>%
  mutate(POS5 = REF5, POS3 = REF3) %>% 
  mutate(POS5r = recode(POS5, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(REFr = recode(REF, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS3r = recode(POS3, "A" = "T", "T" = "A", "C" = "G", "G" = "C")) %>%
  mutate(POS5f = ifelse(strand == "+", POS5, POS3r)) %>%
  mutate(REFf = ifelse(strand == "+", REF, REFr)) %>%
  mutate(POS3f = ifelse(strand == "+", POS3, POS5r)) %>% 
  mutate(spectra = paste0(POS5f, REFf, POS3f)) %>% 
  dplyr::select(CHROM, POS, SNV, REF, ALT, tumor, mutation_AF, Filter, type, spectra)
write_tsv(t, "data/mutations/hetre/leaf_mutations_spectra.tsv") 
rm(list = ls()) ; invisible(gc())
```

```{r datacomp5}
# genomes
quercus_genome <- read_tsv("data/mutations/swiss/Qrob_PM1N.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Qrob_Chr", "", CHROM)))
fagus_genome <- read_tsv("data/mutations/hetre/Fagus_sylvatica_v3.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Fsylvatica_scaffold_", "", CHROM)))
sextonia_genome <- read_tsv("data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
dicorynia_genome <- read_tsv("data/mutations/sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa.fai", 
         col_names = c("CHROM", "length", "bytesindex", "basesperline", "bytesperline")) %>% 
  dplyr::select(CHROM, length) %>% 
  mutate(start = 0, stop = length/10^6) %>% 
  mutate(scf = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
# coverages
coverages <- vroom::vroom("save/coverages_comparisons.tsv")
cov_lims <- coverages %>% 
  mutate(proportion = round(proportion, 2)) %>% 
  mutate(proportion = ifelse(tree == "Verzy" & proportion == 0.94, 0.95, proportion)) %>% 
  filter(proportion == 0.05 | proportion == 0.95) %>% 
  group_by(tree, library, proportion) %>% 
  summarise(coverage = mean(coverage, na.omit = T)) %>% 
  mutate(proportion = recode(as.character(proportion), "0.05" = "high_cov", "0.95" = "low_cov")) %>% 
  reshape2::dcast(tree + library ~ proportion) %>% 
  bind_rows(data.frame(tree = "Napoleon", library = c("upper", "lower"), high_cov = 120, low_cov = 2)) %>% 
  mutate(low_cov = ifelse(low_cov < 10, 10, low_cov))
```

# Comparisons 5

This chapter compares the mutations of Napoleon, 3P, the dwarf beech, Sixto and Angela based on the closest possible filtering.

## Genomes

```{r compgenomes5}
list("Quercus robur" = quercus_genome, 
     "Fagus sylvatica" = fagus_genome, 
     "Sextonia rubra" = sextonia_genome, 
     "Dicorynia guyanensis" = dicorynia_genome) %>% 
  bind_rows(.id = "Species") %>% 
  filter(stop >= 1) %>% 
  ggplot(aes(reorder(as.character(scf), stop), stop)) +
  geom_col() +
  coord_flip() +
  ylab("Length (mb)") + 
  xlab("Chromosome / Scaffold (>1Mb)") +
  facet_wrap(~ Species, scales = "free_y") +
  theme(strip.text.x = element_text(face = "italic"), axis.text.y = element_text(size = 4))
```

## Libraries

```{r compvos5}
ggplot(coverages, aes(coverage, proportion*100, col = library)) +
  geom_line() +
  xlim(0, 400) +
  geom_hline(yintercept = 5, col = "red", linetype = "dashed") +
  geom_hline(yintercept = 95, col = "red", linetype = "dashed") +
  scale_color_discrete(guide = "none") + 
  xlab("Sequencing depth (X)") +
  ylab("% of base in the genome with at least X reads") +
  facet_wrap(~ tree, nrow = 4) 
```

```{r compvolims5}
cov_lims %>% 
  ggplot(aes(x = paste(tree, library), xend = paste(tree, library), y = low_cov, yend = high_cov, col = tree)) + 
  geom_segment() +
  coord_flip() +
  scale_color_discrete("") +
  ylab("Kept sequencing depth") + xlab("") +
  theme(axis.text.y = element_blank())
```

## Mutations

```{r nmutComp5, fig.cap="Caption."}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  dplyr::select(tree, tumor, Filter, SNV) %>% 
  unique() %>% 
  group_by(tree, tumor, Filter) %>% 
  summarise(mutations = n()) %>% 
  ggplot(aes(Filter, mutations, col = tree)) +
  geom_boxplot() +
  facet_wrap(~Filter, scales = "free") +
  ggpubr::stat_compare_means(label = "p.signif") +
  xlab("") + ylab("Number of mutations")
```

## Frequencies

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  ggplot(aes(mutation_AF, col = tree)) +
  geom_density() +
  xlim(0, 0.5) +
  facet_wrap(~ Filter) +
  scale_x_log10() +
  xlab("Allelic frequencies (AF)")
```

## Types

```{r typeComp5, fig.cap="Caption."}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations_spectra.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations_spectra.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations_spectra.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/filtered_tip_mutations_spectra.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/filtered_tip_mutations_spectra.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(type, tumor, tree) %>% 
  summarise(N = n()) %>% 
  group_by(tree, tumor) %>% 
  mutate(pct = N/sum(N)*100) %>% 
  ggplot(aes(type, pct, col = tree)) +
  geom_boxplot(position = "dodge") + 
  theme(axis.text.x = element_blank(), axis.title.x = element_blank(),
        axis.line.x = element_blank(), axis.ticks.x = element_blank(),
        legend.position = "bottom") +
  facet_wrap(~ type, nrow = 1, scales = "free_x") +
  xlab("Percentage of mutations (%) per sample") +
  ggpubr::stat_compare_means(label = "p.signif") +
  ylab("Percentage of mutations")
```

## Spectra

```{r spectraComp5, fig.cap="Caption."}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations_spectra.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations_spectra.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations_spectra.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/filtered_tip_mutations_spectra.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/filtered_tip_mutations_spectra.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  group_by(type, spectra, tumor, tree) %>% 
  summarise(N = n()) %>% 
  group_by(tree, tumor) %>% 
  mutate(pct = N/sum(N)*100) %>% 
  ggplot(aes(spectra, pct, col = tree)) +
  geom_boxplot(position = "dodge") + 
  facet_wrap(~ type, scales = "free_x") +
  theme(axis.text.x = element_text(angle = 90), legend.position = "bottom", axis.title.x = element_blank()) +
  theme(panel.spacing = unit(0, "lines"), panel.border = element_rect(size = 0.25, color = "black", fill = NA)) +
  # scale_y_log10() +
  # ggpubr::stat_compare_means(label = "p.signif") +
  ylab("Percentage of mutations per library")
```

## Rates

### Cummulated distributions

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  select(tree, SNV) %>% 
  unique() %>% 
  group_by(tree) %>% 
  summarise(N = n())
```


```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  ggplot(aes(x = mutation_AF)) +
  geom_histogram(aes(y = ..count..+1, fill = tree), position = "dodge", binwidth = 0.01) +
  scale_y_log10()
```

```{r }
library(dplyr)
library(ggplot2)
theme_set(bayesplot::theme_default())
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
g1 <- ggplot(filter(mutations_hist, tree == "Angela"),
       aes(mutation_AF_class, N)) +
  geom_vline(xintercept = 0.25, linetype = "dashed") +
  geom_col(fill = "#ffc000", col = NA) +
  geom_text(aes(x = 0.4, y = 200, label = N), col = "black", fontface = "bold",
            data = filter(mutations_hist, tree == "Angela", mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  xlab("Allelic fraction (bin of 0.001)") + ylab("Count")
g2 <- ggplot(filter(mutations_hist, tree == "Sixto"),
       aes(mutation_AF_class, N)) +
  geom_vline(xintercept = 0.25, linetype = "dashed") +
  geom_col(fill = "#ffc000", col = NA) +
  geom_text(aes(x = 0.3, y = 50, label = N), col = "black", fontface = "bold",
            data = filter(mutations_hist, tree == "Sixto", mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  xlab("Allelic fraction (bin of 0.001)") + ylab("Count")
ggsave(g1, filename = "~/Téléchargements/AngelaAF.png", width = 4, height = 2, dpi = 800)
ggsave(g2, filename = "~/Téléchargements/SixtoAF.png", width = 4, height = 2, dpi = 800)
```


```{r }
library(readr)
library(dplyr)
library(ggplot2)
theme_set(bayesplot::theme_default())
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
mutations_fruits <- readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(mutation = as.numeric(gsub("SNV", "", SNV))) %>% 
  dplyr::select(mutation, Type) %>% 
  left_join(candidates) %>% 
  mutate(transmitted = ifelse(mutation %in% c(6, 13, 57, 94, 107, 128, 132, 133, 151, 160), "yes", "no")) %>% 
  mutate(SNV = paste0("Super-Scaffold_", CHROM, "#", POS)) %>% 
  select(SNV, transmitted)
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF)) %>% 
  left_join(mutations_fruits) %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "unknown", transmitted)) %>% 
  mutate(transmitted = factor(transmitted, levels = c("yes", "no", "unknown")))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
g1 <- ggplot(filter(mutations_hist, tree == "Angela"),
       aes(mutation_AF_class, N)) +
  geom_col(fill = "#ffc000", col = NA) +
  ylab("Count") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(),
        axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "cm")) +
  xlim(0, 0.52)
g2 <- ggplot() +
  geom_vline(aes(xintercept = mutation_AF), alpha = 1, col = "#ffc000",
             data = filter(mutations, tree == "Angela", transmitted == "no") %>% 
               select(-SNV) %>% mutate(mutation_AF = round(mutation_AF, 3)) %>% unique()) +
  geom_vline(aes(xintercept = mutation_AF), alpha = 1, size = 1, col = "#cc0000",
             data = filter(mutations, tree == "Angela", transmitted == "yes")) +
  xlab("Allelic fraction (bin of 0.001)") +
  theme(axis.line.y = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "cm")) +
  xlim(0, 0.52)
g3 <- cowplot::plot_grid(g1, g2, 
                   nrow = 2, align = "v", 
                   rel_heights = c(5,1))
cowplot::save_plot("~/Téléchargements/AngelaAF.png", g3, dpi = 800)
```

```{r }
library(readr)
library(dplyr)
library(ggplot2)
theme_set(bayesplot::theme_default())
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
mutations_fruits <- readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(mutation = as.numeric(gsub("SNV", "", SNV))) %>% 
  dplyr::select(mutation, Type) %>% 
  left_join(candidates) %>% 
  mutate(transmitted = ifelse(mutation %in% c(6, 13, 57, 94, 107, 128, 132, 133, 151, 160), "yes", "no")) %>% 
  mutate(SNV = paste0("Super-Scaffold_", CHROM, "#", POS)) %>% 
  select(SNV, transmitted)
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF)) %>% 
  left_join(mutations_fruits) %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "unknown", transmitted)) %>% 
  mutate(transmitted = factor(transmitted, levels = c("yes", "no", "unknown")))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
g1 <- ggplot(filter(mutations_hist, tree == "Sixto"),
       aes(mutation_AF_class, N)) +
  geom_col(fill = "#ffc000", col = NA) +
  ylab("Count") +
  theme(axis.title.x = element_blank(), axis.line.x = element_blank(),
        axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "cm")) +
  xlim(0, 0.52)
g2 <- ggplot() +
  geom_vline(aes(xintercept = mutation_AF), alpha = 1, col = "#ffc000",
             data = filter(mutations, tree == "Sixto", transmitted == "no") %>% 
               select(-SNV) %>% mutate(mutation_AF = round(mutation_AF, 3)) %>% unique()) +
  geom_vline(aes(xintercept = mutation_AF), alpha = 1, size = 1, col = "#cc0000",
             data = filter(mutations, tree == "Sixto", transmitted == "yes")) +
  xlab("Allelic fraction (bin of 0.001)") +
  theme(axis.line.y = element_blank(),
        plot.margin = unit(c(0, 0, 0, 0), "cm")) +
  xlim(0, 0.52)
g3 <- cowplot::plot_grid(g1, g2, 
                   nrow = 2, align = "v", 
                   rel_heights = c(5,1))
cowplot::save_plot("~/Téléchargements/SixtoAF.png", g3, dpi = 800)
```


```{r }
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
g <- ggplot(filter(mutations_hist, !(tree %in% c("Sixto", "Angela"))),
       aes(mutation_AF_class, N)) +
  geom_vline(xintercept = 0.25, linetype = "dashed") +
  geom_col(fill = "black", col = NA) +
  facet_wrap(~ tree, nrow = 3, scales = "free_y") +
  geom_text(aes(x = 0.4, y = 25, label = N), col = "black", fontface = "bold",
            data = filter(mutations_hist, !(tree %in% c("Sixto", "Angela")),
                          mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  xlab("Allelic fraction") + ylab("Count")
ggsave(g1, filename = "~/Téléchargements/test.png", width = 2, height = 2, dpi = 800, bg = "white")
g
```

```{r }
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
g <- ggplot(mutations_hist,
       aes(mutation_AF_class, N)) +
  geom_vline(xintercept = 0.25, linetype = "dashed") +
  geom_col(fill = "black", col = NA) +
  facet_wrap(~ tree, ncol = 1, scales = "free_y") +
  geom_text(aes(x = 0.4, y = 25, label = N), col = "black", fontface = "bold",
            data = filter(mutations_hist,
                          mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  xlab("Allelic fraction") + ylab("Count")
ggsave(g, filename = "~/Téléchargements/test.png", width = 8, height = 4*2, dpi = 800, bg = "white")
g
```

```{r }
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
mutations_fruits <- readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(mutation = as.numeric(gsub("SNV", "", SNV))) %>% 
  dplyr::select(mutation, Type) %>% 
  left_join(candidates) %>% 
  mutate(transmitted = ifelse(mutation %in% c(6, 57, 94, 107, 128, 132, 133, 151, 160), "yes", "no")) %>% 
  mutate(SNV = paste0("Super-Scaffold_", CHROM, "#", POS)) %>% 
  select(SNV, transmitted)
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF)) %>% 
  left_join(mutations_fruits) %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "unknown", transmitted)) %>% 
  mutate(transmitted = factor(transmitted, levels = c("yes", "no", "unknown")))
br <- 0.025
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, transmitted, mutation_AF_class) %>% 
  summarise(N = n())
g <- ggplot(filter(mutations_hist, tree %in% c("Sixto", "Angela")),
       aes(mutation_AF_class, N+1)) +
  geom_area(aes(col = "AF>0.25"), fill = "lightblue", alpha  = 0.5,
            data = data.frame(N = c(10000, 10000), mutation_AF_class = c(0.25, 0.52))) +
  geom_col(aes(fill = "unknown"), alpha = 1, position = "identity",
           data = filter(mutations_hist, tree %in% c("Sixto", "Angela"), transmitted == "unknown")) +
  geom_col(aes(fill = "no"), alpha = 1, position = "identity",
           data = filter(mutations_hist, tree %in% c("Sixto", "Angela"), transmitted == "no")) +
  geom_col(aes(fill = "yes"), alpha = 1, position = "identity",
           data = filter(mutations_hist, tree %in% c("Sixto", "Angela"), transmitted == "yes")) +
  scale_y_log10(labels = scales::label_comma()) +
  facet_wrap(~ tree, nrow = 2) +
  geom_text(aes(x = 0.4, y = 100, label = N), col = "blue", fontface = "bold",
            data = filter(mutations_hist, tree %in% c("Sixto", "Angela"), mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  scale_fill_manual("Transmitted", values = c("red", "lightgrey", "chartreuse3")) +
  xlab("Allelic fraction") + ylab("Count") +
  scale_color_manual("", values = "white", labels = expression(AF>=0.25))
# ggsave(g, filename = "~/Téléchargements/figure_test.png", width = 8, height = 4, dpi = 800, bg = "white")
g
```


```{r }
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM)))
mutations_fruits <- readxl::read_xlsx("data/mutations/fruits/fruit_mutations_igv.xlsx", "classification") %>% 
  mutate(mutation = as.numeric(gsub("SNV", "", SNV))) %>% 
  dplyr::select(mutation, Type) %>% 
  left_join(candidates) %>% 
  mutate(transmitted = ifelse(mutation %in% c(6, 57, 94, 107, 128, 132, 133, 151, 160), "yes", "no")) %>% 
  mutate(SNV = paste0("Super-Scaffold_", CHROM, "#", POS)) %>% 
  select(SNV, transmitted)
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF)) %>% 
  left_join(mutations_fruits) %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "unknown", transmitted)) %>% 
  mutate(transmitted = factor(transmitted, levels = c("yes", "no", "unknown")))
br <- 0.025
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, transmitted, mutation_AF_class) %>% 
  summarise(N = n())
g <- ggplot(mutations_hist, aes(mutation_AF_class, N+1)) +
  geom_area(aes(col = "AF>0.25"), fill = "lightblue", alpha  = 0.5,
            data = data.frame(N = c(10000, 10000), mutation_AF_class = c(0.25, 1))) +
  geom_col(aes(fill = "unknown"), alpha = 1, position = "identity",
           data = filter(mutations_hist, transmitted == "unknown")) +
  geom_col(aes(fill = "no"), alpha = 1, position = "identity",
           data = filter(mutations_hist, transmitted == "no")) +
  geom_col(aes(fill = "yes"), alpha = 1, position = "identity",
           data = filter(mutations_hist, transmitted == "yes")) +
  scale_y_log10(labels = scales::label_comma()) +
  facet_wrap(~ tree, nrow = 2) +
  geom_text(aes(x = 0.4, y = 100, label = N), col = "blue", fontface = "bold",
            data = filter(mutations_hist, mutation_AF_class >= 0.25) %>%
              group_by(tree) %>%
              summarise(N = sum(N))) +
  scale_fill_manual("Transmitted", values = c("red", "lightgrey", "chartreuse3")) +
  xlab("Allelic fraction") + ylab("Count") +
  scale_color_manual("", values = "white", labels = expression(AF>=0.25)) +
  theme(legend.position = c(0.8, 0.2))
# ggsave(g, filename = "~/Téléchargements/figure_SI_test.png", width = 8, height = 4, dpi = 800, bg = "white")
g
```


```{r ratesComp5, fig.cap="Caption."}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  group_by(tree, Filter) %>% 
  arrange(desc(mutation_AF)) %>% 
  mutate(mutations = 1) %>% 
  mutate(cummulated_mutations = cumsum(mutations)) %>% 
  ggplot(aes(mutation_AF, cummulated_mutations, col = tree)) +
  geom_line() +
  scale_y_log10() + scale_x_reverse() +
  xlab("Allelic frequency") + ylab("Cummulated mutations with decreasing AF") +
  facet_wrap(~ Filter, nrow = 2)
```

```{r, eval = F}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  group_by(tree, Filter) %>% 
  arrange(desc(mutation_AF)) %>% 
  mutate(mutations = 1) %>% 
  mutate(cummulated_mutations = cumsum(mutations)) %>% 
  filter(Filter == "base", mutation_AF > 0.23, mutation_AF < 0.27) %>% 
  group_by(tree) %>% 
  summarise(cummulated_mutations = median(cummulated_mutations))
```


```{r, eval=F}
g <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  group_by(tree, Filter) %>% 
  arrange(desc(mutation_AF)) %>% 
  mutate(mutations = 1) %>% 
  mutate(cummulated_mutations = cumsum(mutations)) %>% 
  filter(Filter == "base") %>% 
  ggplot(aes(mutation_AF, cummulated_mutations, col = tree)) +
  geom_line() +
  scale_y_log10() + scale_x_reverse() +
  xlab("Allelic frequency") + ylab("Cummulated mutations with decreasing AF")
```


```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  # filter(tree %in% c("Angela", "Sixto")) %>% 
  rename(library = tumor) %>% 
  group_by(tree, library) %>% 
  summarise(N = n()) %>% 
  left_join(coverages %>% 
              group_by(tree, library) %>% 
              filter(proportion > 0.49, proportion < 0.51) %>% 
              summarise(coverage = mean(coverage))) %>% 
  mutate(coverage = ifelse(tree == "Napoleon", 60, coverage)) %>% 
  # lm(N ~ coverage, data = .) %>% sjPlot::tab_model()
  ggplot(aes(coverage, N, col = tree, group = NA))+
  geom_point() +
  geom_smooth(method = "lm") +
  annotate("text", 100, 4000, label = expression(beta==15.20)) +
  annotate("text", 100, 3500, label = expression(p==0.026)) +
  annotate("text", 100, 3000, label = expression(R^2==0.144)) +
  annotate("text", 100, 2500, label = expression(N==28)) +
  xlab("Mean sequencing depth per library (X)") + ylab("Number of mutations detected per library")
```

### Lognormal distributions

```{r , fig.cap="Caption."}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, tumor) %>% 
  mutate(mutation_AF_ints = cut(mutation_AF, breaks = seq(0, 1, length.out = 100))) %>% 
  group_by(tree, tumor, mutation_AF_ints) %>% 
  summarise(N = n()) %>% 
  separate(mutation_AF_ints, c("startAF", "stopAF"), ",", remove = F) %>% 
  mutate(startAF = as.numeric(gsub("(", "", startAF, fixed = T)), stopAF = as.numeric(gsub("]", "", stopAF))) %>% 
  mutate(AF = (startAF+stopAF)/2) %>%
  ggplot(aes(AF, N, col = tree, group = tumor)) +
  geom_line() +
  geom_point() +
  ylim(1, NA) + scale_y_log10() +
  xlab("Allelic frequency") + ylab("Numner of mutations")
```

```{r}
t <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree")
fit <- MASS::fitdistr(t$mutation_AF, "lognormal") 
ggplot(t, aes(mutation_AF)) +
  geom_histogram(aes(y = ..density..), bins = 100, fill = "lightgrey") +
  stat_function(fun = function(x) dlnorm(x, meanlog = fit$estimate[1], sdlog = fit$estimate[2]), col = "red") +
  xlim(0, 0.5) +
  xlab("Allelic frequency") + ylab("Observed mutations density")
```

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, tumor) %>% 
  summarise(mu = mean(log(mutation_AF)),  sigma = sd(log(mutation_AF))) %>% 
  rowwise() %>% 
  mutate(AF = list(seq(0, 1, length.out = 1000))) %>% 
  unnest(AF) %>% 
  mutate(N = dlnorm(AF, meanlog = mu, sdlog = sigma)) %>% 
  group_by(tree, AF) %>% 
  summarise(ll = min(N), m = median(N), hh = max(N)) %>% 
  ggplot(aes(x = AF, fill = tree, col = tree)) +
  geom_ribbon(aes(ymin = ll, ymax = hh), alpha = 0.2, col = NA) +
  geom_line(aes(y = m), size = 2) +
  xlim(0, 0.5) +
  xlab("Allelic frequency") + ylab("Observed mutations density")
```

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, tumor) %>% 
  summarise(`mu[log]~p<2e-16~R^2==0.89` = mean(log(mutation_AF)),  `sigma[log]~p==0.094~R^2==0.03` = sd(log(mutation_AF))) %>% 
  reshape2::melt(c("tree", "tumor")) %>% 
  mutate(name = paste0(tree, ":", tumor)) %>% 
  left_join(coverages %>% 
              group_by(tree, library) %>% 
              filter(proportion > 0.4, proportion < 0.5) %>% 
              summarise(coverage = mean(coverage)) %>% 
              mutate(tumor = library)) %>% 
    mutate(coverage = ifelse(tree == "Napoleon", 60, coverage)) %>% 
  # split(.$variable) %>% lapply(function(x) lm(value ~ coverage, data = x)) %>% lapply(summary)
  ggplot(aes(coverage, value, col = tree, group = NA)) +
  facet_wrap(~variable, scales = "free", labeller = label_parsed) +
  geom_smooth(method = "lm", col = c(rep("blue", 80), rep("grey", 80))) +
  geom_point() +
  xlab("Mean sequencing depth per library (X)") +
  ylab("Parameters of the log-distribution of AF")
```

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  group_by(tree, tumor) %>% 
  summarise(sigma = sd(log(mutation_AF))) %>% 
  group_by(tree) %>% 
  summarise(ll = min(sigma), l = quantile(sigma, 0.25), m = median(sigma),
            h = quantile(sigma, 0.75), hh = max(sigma)) %>% 
    ggplot(aes(x = tree, xend = tree,
             color = tree, fill = tree)) +
  geom_point(aes(y = m), shape = 21, size = 3, alpha = 0.5) +
  geom_segment(aes(y = ll, yend = hh),
               size = 1, show.legend = F, alpha = 0.5) +
  geom_segment(aes(y = l, yend = h), size = 2, alpha = 0.5) +
  coord_flip() +
  ylab(expression(sigma[log])) + xlab("")
```

### Exponential distrbutions

```{r, eval=F}
t <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree")
mu <- quantile(t$mutation_AF, 0.1)
dexptrunc <- function(x, rate){
  if(rate > 0)
    return(TruncatedDistributions::dtexp(x, a=mu, b=Inf, rate=rate))
  else
    return(rep(NaN, length(x)))
} 
pexptrunc <- function(q, rate){
  if(length(x) > 0)
    return(TruncatedDistributions::ptexp(q, a=mu, b=Inf, rate=rate))
  else
    return(x)
} 
fit <- fitdistrplus::fitdist(filter(t, mutation_AF > mu)$mutation_AF, "exptrunc", start = list(rate=50))
r <- fit$estimate[1]
c <- r*exp(r*mu)
ggplot(t, aes(mutation_AF)) +
  geom_histogram(aes(y = ..density..), bins = 100, fill = "lightgrey") +
  stat_function(fun = function(x) dexptrunc(x, rate = r), col = "red") +
  stat_function(fun = function(x) c*exp(-r*x), col = "blue") +
  geom_vline(xintercept = mu, col = "black", linetype = "dashed") +
  xlim(0, 0.5) +
  xlab("Allelic frequency") + ylab("Observed mutations density")
```

```{r, eval=F}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(treetumor = paste0(tree, "_", tumor)) %>% 
  dplyr::select(treetumor, mutation_AF) %>% 
  split(.$treetumor) %>% 
  lapply(function(x){
    mu <- quantile(t$mutation_AF, 0.1)
    dexptrunc <- function(x, rate){
      if(rate > 0)
        return(TruncatedDistributions::dtexp(x, a=mu, b=Inf, rate=rate))
      else
        return(rep(NaN, length(x)))
    } 
    pexptrunc <- function(q, rate){
      if(length(x) > 0)
        return(TruncatedDistributions::ptexp(q, a=mu, b=Inf, rate=rate))
      else
        return(x)
    } 
    fit <- try(fitdistrplus::fitdist(filter(x, mutation_AF > mu)$mutation_AF, "exptrunc", start = list(rate=15)))
    if(inherits(fit, "try-error")) data.frame(rate = NA, mu = mu, c = NA)
    if(!inherits(fit, "try-error")) data.frame(r = fit$estimate[1], mu = mu, c = fit$estimate[1]*exp(fit$estimate[1]*mu))
  }) %>% bind_rows(.id = "treetumor") %>% 
  separate(treetumor, c("tree", "tumor")) %>% 
  rowwise() %>%
  mutate(AF = list(seq(0, 1, length.out = 1000))) %>%
  unnest(AF) %>%
  mutate(N = c*exp(-r*AF)) %>% 
  group_by(tree, AF) %>%
  summarise(ll = min(N), m = median(N), hh = max(N)) %>%
  ggplot(aes(x = AF, fill = tree, col = tree)) +
  geom_ribbon(aes(ymin = ll, ymax = hh), alpha = 0.2, col = NA) +
  geom_line(aes(y = m), size = 2) +
  xlim(0, 0.5) +
  xlab("Allelic frequency") + ylab("Observed mutations density")
```

```{r, eval=F}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(treetumor = paste0(tree, "_", tumor)) %>% 
  dplyr::select(treetumor, mutation_AF) %>% 
  split(.$treetumor) %>% 
  lapply(function(x){
    mu <- quantile(t$mutation_AF, 0.1)
    dexptrunc <- function(x, rate){
      if(rate > 0)
        return(TruncatedDistributions::dtexp(x, a=mu, b=Inf, rate=rate))
      else
        return(rep(NaN, length(x)))
    } 
    pexptrunc <- function(q, rate){
      if(length(x) > 0)
        return(TruncatedDistributions::ptexp(q, a=mu, b=Inf, rate=rate))
      else
        return(x)
    } 
    fit <- try(fitdistrplus::fitdist(filter(x, mutation_AF > mu)$mutation_AF, "exptrunc", start = list(rate=15)))
    if(inherits(fit, "try-error")) data.frame(rate = NA, mu = mu, c = NA)
    if(!inherits(fit, "try-error")) data.frame(r = fit$estimate[1],  c = fit$estimate[1]*exp(fit$estimate[1]*mu))
  })  %>% bind_rows(.id = "treetumor") %>% 
  separate(treetumor, c("tree", "tumor")) %>% 
    reshape2::melt(c("tree", "tumor")) %>% 
  mutate(name = paste0(tree, ":", tumor)) %>% 
  left_join(coverages %>% 
              group_by(tree, library) %>% 
              filter(proportion > 0.4, proportion < 0.5) %>% 
              summarise(coverage = mean(coverage)) %>% 
              mutate(tumor = library)) %>% 
    mutate(coverage = ifelse(tree == "Napoleon", 60, coverage)) %>% 
  ggplot(aes(coverage, value, col = tree, group = NA)) +
  facet_wrap(~variable, scales = "free", labeller = label_parsed) +
  geom_smooth(col = "blue", method = "lm") +
  geom_point() +
  xlab("Mean sequencing depth per library (X)") +
  ylab("Parameters of the power-distribution of AF")
```

### Power law distribution

```{r}
t <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree")
mu <- quantile(t$mutation_AF, 0.1)
fit <- igraph::fit_power_law(t$mutation_AF, xmin = mu)
a <- fit$alpha
b <- fit$xmin
c <- (a-1)*b^(a-1)
ggplot(t, aes(mutation_AF)) +
  geom_histogram(aes(y = ..density..), bins = 100, fill = "lightgrey") +
   stat_function(fun = function(x) c*x^(-a), col = "red") +
  xlab("Allelic frequency") + ylab("Observed mutations density") +
  geom_vline(xintercept = b, linetype = 'dashed', col = "red") +
  xlim(0, 0.5) + ylim(0, 50)
```

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(treetumor = paste0(tree, "_", tumor)) %>% 
  dplyr::select(treetumor, mutation_AF) %>% 
  split(.$treetumor) %>% 
  lapply(function(x){
    mu <- quantile(t$mutation_AF, 0.1)
    fit <- igraph::fit_power_law(x$mutation_AF, xmin = mu)
    a <- fit$alpha
    b <- fit$xmin
    c <- (a-1)*b^(a-1)
    return(data.frame(a = a, b = b, c = c))
  }) %>% bind_rows(.id = "treetumor") %>% 
  separate(treetumor, c("tree", "tumor")) %>% 
  rowwise() %>%
  mutate(AF = list(seq(0, 1, length.out = 1000))) %>%
  unnest(AF) %>%
  mutate(N = c*AF^(-a)) %>% 
  group_by(tree, AF) %>%
  summarise(ll = min(N), m = median(N), hh = max(N)) %>%
  ggplot(aes(x = AF, fill = tree, col = tree)) +
  geom_ribbon(aes(ymin = ll, ymax = hh), alpha = 0.2, col = NA) +
  geom_line(aes(y = m), size = 2) +
  xlim(0, 0.5) +  ylim(0, 500) +
  xlab("Allelic frequency") + ylab("Observed mutations density")
```

```{r}
list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(treetumor = paste0(tree, "_", tumor)) %>% 
  dplyr::select(treetumor, mutation_AF) %>% 
  split(.$treetumor) %>% 
  lapply(function(x){
    mu <- quantile(t$mutation_AF, 0.1)
    fit <- igraph::fit_power_law(x$mutation_AF, xmin = mu)
    a <- fit$alpha
    b <- fit$xmin
    c <- (a-1)*b^(a-1)
    return(data.frame(a = a, b = b, c = c))
  }) %>% bind_rows(.id = "treetumor") %>% 
  separate(treetumor, c("tree", "tumor")) %>% 
    reshape2::melt(c("tree", "tumor")) %>% 
  mutate(name = paste0(tree, ":", tumor)) %>% 
  filter(variable != "b") %>% 
  left_join(coverages %>% 
              group_by(tree, library) %>% 
              filter(proportion > 0.4, proportion < 0.5) %>% 
              summarise(coverage = mean(coverage)) %>% 
              mutate(tumor = library)) %>% 
    mutate(coverage = ifelse(tree == "Napoleon", 60, coverage)) %>% 
  ggplot(aes(coverage, value, col = tree, group = NA)) +
  facet_wrap(~variable, scales = "free", labeller = label_parsed) +
  geom_smooth(method = "lm", col = "blue") +
  geom_point() +
  xlab("Mean sequencing depth per library (X)") +
  ylab("Parameters of the power-distribution of AF")
```

## Annotations

*Angela and Sixto in progress.*

## Genes

*Angela and Sixto in progress.*
