```{r setupfigurefinalevs, include=FALSE}
rm(list = ls()) ; invisible(gc()) ; set.seed(42)
library(knitr)
library(tidyverse)
theme_set(bayesplot::theme_default())
opts_chunk$set(
  echo = F, message = F, warning = F, fig.height = 6, fig.width = 8,
  cache = T, cache.lazy = F)
```

# Manuscript figures EVS

This chapter develop the code for figure preparation for the final manuscript.

## Figure 2 EVS

```{r figlightnevs}
make_plot_light <- function(name = "angela"){
  t1 <- vroom::vroom(paste0("data/mutations/", name, "/filters/leaf_mutations_spectra.tsv"), 
                     show_col_types = FALSE) %>%   
    filter(Filter == "evs") %>% 
    dplyr::select(tip, tumor, Filter, SNV) %>% 
    unique() %>% 
    group_by(tip, tumor, Filter) %>% 
    summarise(mutations = n()) %>% 
    mutate(type = "All") %>% 
    dplyr::select(-Filter)
  t2 <- vroom::vroom(paste0("data/mutations/", name, "/filters/leaf_mutations_spectra.tsv"), 
                     show_col_types = FALSE) %>%   
    filter(Filter == "evs") %>% 
    dplyr::select(type, tip, tumor, SNV) %>% 
    unique() %>% 
    group_by(type, tip, tumor) %>% 
    summarise(mutations = n()) %>% 
    mutate(type = gsub("-", "", type))
  m <- 10^(log10(max(t1$mutations))*1.1)
  bind_rows(t1, t2) %>% 
    mutate(condition = recode(tip, "L" = "Light", "S" = "Shade")) %>% 
    ggplot(aes(condition, mutations, fill = condition)) +
    geom_boxplot() +
    facet_wrap(~type, nrow = 1) +
    ggpubr::stat_compare_means(label = "p.signif", method = "t.test",
                               bracket.size	= 1.2, tip.length	= 0,
                               label.y = 0.925*log10(m), vjust = -0.5,
                               comparisons = list(c("Light", "Shade"))) +
    xlab("") + ylab("Number of mutations") +
    scale_fill_manual("", values = c("#FFDF00", "#8B8680")) +
    theme(axis.text.x = element_blank(), axis.ticks = element_blank(), axis.title.x = element_blank(),
          panel.spacing = unit(0, "lines"),
          # legend.position = c(0.7, -0.15)) +
          legend.position = "bottom", 
          legend.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
          legend.spacing = margin(0, unit = "pt"),
          legend.box.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
          legend.box.spacing = margin(0, unit = "pt")) +
    scale_y_log10(limits = c(NA, m)) +
    guides(fill = guide_legend(nrow = 1))
  # cowplot::save_plot(paste0("~/Téléchargements/", name, "LightN.png"), g, dpi = 800, base_width = 2, base_height = 4)
}
ge <- make_plot_light("angela")
gf <- make_plot_light("sixto")
```

```{r fig2finalevs}
cowplot::plot_grid(ge, gf, nrow = 1, labels = c("a", "b"))
```

## Figure 3 EVS

```{r histafevs}
library(readr)
library(dplyr)
library(ggplot2)
theme_set(bayesplot::theme_default())
candidates <- bind_rows(
  read_tsv("data/mutations/fruits/angela_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 1:n(), tree = "Angela"),
  read_tsv("data/mutations/fruits/sixto_fruits_candidate_mutations.tsv") %>% 
    mutate(mutation = 125:(124+n()), tree = "Sixto")
) %>% 
  dplyr::select(CHROM, POS, REF, ALT, af, branch, mutation, tree) %>% 
  mutate(CHROM = as.numeric(gsub("Super-Scaffold_", "", CHROM))) %>% 
  mutate(SNV = paste0("Super-Scaffold_", as.numeric(CHROM), "#", as.numeric(POS))) %>% 
  dplyr::select(SNV)
mutations_fruits <- read_tsv("data/mutations/fruits2/fruit_mutations.tsv") %>% 
  filter(allelic_fraction > 0, heterozygous_call_gatk == 1) %>%
  mutate(transmitted = "yes") %>% 
  mutate(SNV = paste0("Super-Scaffold_", as.numeric(scaffold), "#", as.numeric(position))) %>% 
  dplyr::select(SNV, transmitted) %>% 
  full_join(candidates) %>%
  unique() %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "no", transmitted))
mutations <- list(Napoleon = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
     "3P" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
     Verzy = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
     Angela = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
     Sixto = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  filter(Filter == "evs") %>% 
  group_by(tree, SNV) %>% 
  summarise(mutation_AF = median(mutation_AF)) %>% 
  left_join(mutations_fruits) %>% 
  mutate(transmitted = ifelse(is.na(transmitted), "unknown", transmitted)) %>% 
  mutate(transmitted = factor(transmitted, levels = c("yes", "no", "unknown")))
br <- 0.001
mutations_hist <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
br_log <- 0.02
mutations_hist_log <- mutations %>%
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br_log), labels = seq(0,1-br_log,br_log)+br_log/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
mutations_hist_log_tested <- mutations %>%
  filter(transmitted != "unknown") %>% 
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br_log), labels = seq(0,1-br_log,br_log)+br_log/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
mutations_hist_log_embryo <- mutations %>%
  filter(transmitted == "yes") %>% 
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br_log), labels = seq(0,1-br_log,br_log)+br_log/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(N = n())
make_plot_af <- function(name = "Angela") {
  g1.B <- ggplot(filter(mutations_hist, tree == name),
                 aes(mutation_AF_class, N+max(filter(mutations_hist, tree == name)$N)/100*5)) +
    geom_col(fill = "#ffd659ff", col = NA, alpha = 0.8) +
    xlim(0, 0.52) +
    xlab("Allelic fraction") +
    theme(axis.line = element_line(color="#57504d"),
          axis.ticks = element_line(color="#57504d"),
          axis.title = element_text(color="#57504d"),
          axis.text = element_text(color="#57504d"))
  g1 <- ggplot(filter(mutations_hist_log, tree == name),
                 aes(mutation_AF_class, N+1)) +
    geom_col(aes(fill = "mutation"), col = NA) +
    geom_col(aes(fill = "tested for transmission"), col = NA, 
           data = filter(mutations_hist_log_tested, tree == name)) +
    geom_col(aes(fill = "found transmitted"), col = NA, 
           data = filter(mutations_hist_log_embryo, tree == name)) +
    ylab("Count") + xlab("Allelic fraction") +
    xlim(0, 0.52) +
    scale_fill_manual("", values = c("#cc0000", "#ffc000", "grey")) +
    scale_y_log10(breaks = c(2, 11, 101, 1001),
                  labels = c("1", "10", "100", "1,000"), 
                  limits = c(1, 7000)) +
    theme(legend.position = c(0.8, 0.8),
          legend.text = element_text(size = 6),
          legend.key.size = unit(1, "lines"),
          legend.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
          legend.spacing.y = unit(5, "pt"),
          legend.box.margin = margin(t = 0, r = 0, b = 0, l = 0, unit = "pt"),
          # legend.box.spacing = margin(0, unit = "pt")
          ) +
    annotation_custom(ggplotGrob(
      g1.B +
        theme(axis.title.y = element_blank(), text = element_text(size = 10))
    )
    , xmin = 0.2, xmax = 0.52, ymin = log(2), ymax = log(50))
  return(g1)
}
ga <- make_plot_af("Angela")
gb <- make_plot_af("Sixto")
```

```{r cumafevs}
library(tidyr)
cols <- c("#3e40b7", "#0264c5", "#01a99c", "#df2e36", "#f38d04")
br <- 0.05
gc <- list("Q. robur - Lausanne (60X)" = vroom::vroom("data/mutations/swiss/leaf_mutations.tsv"),
           "Q. robur - Bordeaux (160X)" = vroom::vroom("data/mutations/bordeaux/leaf_mutations.tsv"),
           "F. sylvatica (64X)" = vroom::vroom("data/mutations/hetre/leaf_mutations.tsv"),
           "D. guianensis (138X)" = vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
           "S. rubra (73X)" = vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>%
  bind_rows(.id = "tree") %>% 
  mutate(Filter = recode(Filter, "evs" = "evs base")) %>% 
  separate_rows(Filter) %>% 
  group_by(tree, Filter) %>% 
  arrange(desc(mutation_AF)) %>% 
  mutate(mutations = 1) %>% 
  mutate(cummulated_mutations = cumsum(mutations)) %>% 
  filter(Filter == "evs") %>% 
  mutate(mutation_AF_class = as.numeric(as.character(cut(mutation_AF, breaks = seq(0,1,br), labels = seq(0,1-br,br)+br/2)))) %>% 
  group_by(tree, mutation_AF_class) %>% 
  summarise(l = min(cummulated_mutations), m = median(cummulated_mutations), h = max(cummulated_mutations)) %>% 
  mutate(tree = factor(tree, levels = c("F. sylvatica (64X)", 
                                        "Q. robur - Lausanne (60X)", 
                                        "Q. robur - Bordeaux (160X)", 
                                        "S. rubra (73X)",
                                        "D. guianensis (138X)"))) %>% 
  ggplot(aes(x = mutation_AF_class, m, col = tree, fill = tree)) +
  geom_ribbon(aes(ymin = l, ymax = h), col = NA, alpha = 0.5) +
  geom_line(aes(y = m)) +
  scale_y_log10(labels = scales::label_comma()) + 
  scale_color_manual("", values = cols) +
  xlab("Allelic fraction") + ylab("Cumulative number\nof mutations") +
  theme(legend.position = c(0.7, 0.8), 
        legend.text = element_text(size = 8, face = "italic")) +
  scale_fill_manual(guide = "none", values = cols)
```

```{r synafevs}
evs <- unique(
  (bind_rows(vroom::vroom("data/mutations/angela/filters/leaf_mutations.tsv"),
            vroom::vroom("data/mutations/sixto/filters/leaf_mutations.tsv")) %>% 
    filter(Filter == "evs"))$SNV)

data_inter <- list(Angela = read_tsv("save/mutations_angela_sixto.tsv"), 
             Sixto = read_tsv("save/mutations_annotated_sixto.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  filter(SNV %in% evs) %>% 
  filter(mutation_AF > 0.01) %>% 
  dplyr::select(tree, SNV, mutation_AF, transcript) %>% 
  mutate(type = "All") %>% 
  mutate(category = ifelse(is.na(transcript), "intergenic   ", "gene")) %>% 
  mutate(comparison = ifelse(is.na(transcript), "yes", "no")) %>% 
  dplyr::select(tree, SNV, type, category, comparison, mutation_AF)
data_intron <- list(Angela = read_tsv("save/mutations_angela_sixto.tsv"), 
             Sixto = read_tsv("save/mutations_annotated_sixto.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  filter(SNV %in% evs) %>% 
  filter(transcript == 1) %>% 
  filter(mutation_AF > 0.01) %>% 
  dplyr::select(tree, SNV, mutation_AF, CDS) %>% 
  mutate(type = "Genic only") %>% 
  mutate(category = ifelse(is.na(CDS), "intron   ", "exon")) %>% 
  mutate(comparison = ifelse(is.na(CDS), "yes", "no")) %>% 
  dplyr::select(tree, SNV, type, category, comparison, mutation_AF)
data_syn <- list(Angela = read_tsv("save/mutations_angela_sixto.tsv"), 
             Sixto = read_tsv("save/mutations_annotated_sixto.tsv")) %>% 
  bind_rows(.id = "tree") %>% 
  filter(SNV %in% evs) %>% 
  filter(CDS == 1) %>% 
  filter(mutation_AF > 0.01) %>% 
  dplyr::select(tree, SNV, mutation_AF, nonsynonymous) %>% 
  mutate(type = "CDS only") %>% 
  mutate(category = ifelse(is.na(nonsynonymous), "synonymous      ", "non-synonymous")) %>% 
  mutate(comparison = ifelse(is.na(nonsynonymous), "yes", "no")) %>% 
  dplyr::select(tree, SNV, type, category, comparison, mutation_AF)
data <- bind_rows(data_inter, data_intron, data_syn) %>% 
  filter(category %in% c("intergenic   ", "intron   ", "synonymous      ", "non-synonymous")) %>% 
  mutate(comparison = factor(comparison, levels = c("yes", "no"))) %>% 
  mutate(category = factor(category, levels = c("intergenic   ", "intron   ",
                                                "synonymous      ", "non-synonymous"))) %>%
    mutate(type = factor(type, levels = c("All", "Genic only", "CDS only")))  %>% 
  mutate(tree = recode(tree, "Angela" = "D. guianensis", "Sixto" = "S. rubra"))
data_m <- group_by(data, tree, type, category, comparison) %>% 
  summarise(mutation_AF = median(mutation_AF))
gd <- ggplot(data, aes(category, mutation_AF, fill = category)) +
  geom_violin(col = NA) +
  geom_boxplot(width = 0.1, fill = 'white', alpha = 0.5, outlier.alpha = 0) +
  facet_wrap(~ tree, nrow = 1) +
  ylab("Allelic fraction") +
  ggpubr::stat_compare_means(method = "t.test",  bracket.size	= 1.2, size = 3,
                             label.y = c(log10(0.2), log10(0.3), log10(0.5)),
                             comparisons = list(c("non-synonymous", "synonymous      "), 
                                                c("non-synonymous", "intron   "),
                                                c("intergenic   ", "non-synonymous"))) +
  scale_fill_manual("", values  = c("#3581D8", "#28CC2D", "#FFE135", "#D82E3F")) +
  # guides(fill=guide_legend(nrow=2,byrow=TRUE)) +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.line.x = element_blank(), panel.spacing = unit(0.5, "lines")) +
  scale_y_log10(limits = c(NA, 0.65)) +
  theme(legend.position = c(0.5, 0), legend.direction = "horizontal", 
        legend.text = element_text(size = 7), legend.key.size = unit(1, "lines"), 
        legend.spacing.y = unit(8, "points"), strip.text = element_text(face = "italic"))
```

```{r, eval=FALSE}
g <- cowplot::plot_grid(ga, gb, gc, gd, nrow = 2, labels = c("a", "b", "c", "d"),
                        rel_heights = c(1, 1))
cowplot::save_plot(paste0("figs/Fig3_evs.png"), g, dpi = 1000, base_width = 8, base_height = 6, bg = "white")
```

```{r fig3finalevs, fig.cap="Fig. 3 | Allelic fractions of somatic mutations among trees and among genomic elements. Histogram of allelic fractions of mutations detected in the crown of the two tropical trees: the Dicorynia guianensis tree named Angela (a), and the Sextonia rubra tree named Sixto (b). The main histogram shows the allelic fractions of the somatic mutations using a bin of 0.02 and a log-transformed count with the mutations detected in the crown in yellow, the mutations tested for transmission in grey, and the mutations transmitted to the embryos in red. The inner histogram shows the allelic fractions of the somatic mutations using a bin of 0.001 and a natural count. (c) Cumulative number of somatic mutations per branch with decreasing allelic fraction for five trees reanalysed with the same pipeline. The five trees include the two tropical trees studied, the Dicorynia guianensis tree named Angela in orange and the Sextonia rubra tree named Sixto in red, and three temperate trees, two pedunculate oaks Quercus robur named 3P in green and Napoleon in blue and a tortuous beech Fagus sylvatica named Verzy in purple. All trees were analysed with the same pipeline (see methods) but were sequenced with a different depth indicated in brackets. The line represents the median value while the area represents the minimum and maximum values on the 2 to 10 branches per tree. (d) Comparisons of allelic fractions for non-synonymous mutations in red with synonymous mutations in yellow, intronic mutations in green and intergenic mutations in blue for the two tropical trees: the Dicorynia guianensis tree named Angela (left panel), and the Sextonia rubra tree named Sixto (right panel). The p-value above the bars indicates the significance of the Student's T-test for the pairs of groups."}
include_graphics("figs/Fig3_evs.png")
```
