<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Comparisons 2 | TreeMutation</title>
  <meta name="description" content="Mutation in the tropical tree canopy." />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Comparisons 2 | TreeMutation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Mutation in the tropical tree canopy." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Comparisons 2 | TreeMutation" />
  
  <meta name="twitter:description" content="Mutation in the tropical tree canopy." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2023-06-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations-sixto.html"/>
<link rel="next" href="comparisons-5.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TreeMutation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="species.html"><a href="species.html"><i class="fa fa-check"></i><b>1</b> Species</a>
<ul>
<li class="chapter" data-level="1.1" data-path="species.html"><a href="species.html#dicorynia-guianensis"><i class="fa fa-check"></i><b>1.1</b> <em>Dicorynia guianensis</em></a></li>
<li class="chapter" data-level="1.2" data-path="species.html"><a href="species.html#sextonia-rubra"><i class="fa fa-check"></i><b>1.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="1.3" data-path="species.html"><a href="species.html#eschweilera-coriacea"><i class="fa fa-check"></i><b>1.3</b> <em>Eschweilera coriacea</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="site.html"><a href="site.html"><i class="fa fa-check"></i><b>2</b> Site</a>
<ul>
<li class="chapter" data-level="2.1" data-path="site.html"><a href="site.html#paracou"><i class="fa fa-check"></i><b>2.1</b> Paracou</a></li>
<li class="chapter" data-level="2.2" data-path="site.html"><a href="site.html#régina"><i class="fa fa-check"></i><b>2.2</b> Régina</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="individuals.html"><a href="individuals.html"><i class="fa fa-check"></i><b>3</b> Individuals</a>
<ul>
<li class="chapter" data-level="3.1" data-path="individuals.html"><a href="individuals.html#angela---dicorynia-guyanensis"><i class="fa fa-check"></i><b>3.1</b> <strong>Angela</strong> - <em>Dicorynia guyanensis</em></a></li>
<li class="chapter" data-level="3.2" data-path="individuals.html"><a href="individuals.html#sextonia-rubra-1"><i class="fa fa-check"></i><b>3.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="3.3" data-path="individuals.html"><a href="individuals.html#eschweilera-sp."><i class="fa fa-check"></i><b>3.3</b> <em>Eschweilera sp.</em></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="individuals.html"><a href="individuals.html#mn5505---regina"><i class="fa fa-check"></i><b>3.3.1</b> MN5505 - Regina</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>4</b> Sampling</a></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#sequencing"><i class="fa fa-check"></i><b>5.1</b> Sequencing</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#assembly-workflow"><i class="fa fa-check"></i><b>5.2</b> Assembly workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-assembly.html"><a href="genome-assembly.html#v1"><i class="fa fa-check"></i><b>5.2.1</b> V1</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-assembly.html"><a href="genome-assembly.html#v2"><i class="fa fa-check"></i><b>5.2.2</b> V2</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#annotation-workflow"><i class="fa fa-check"></i><b>5.3</b> Annotation workflow</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#project-description"><i class="fa fa-check"></i><b>5.4</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mutations-detection.html"><a href="mutations-detection.html"><i class="fa fa-check"></i><b>6</b> Mutations detection</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mutations-detection.html"><a href="mutations-detection.html#in-silico-experiment"><i class="fa fa-check"></i><b>6.1</b> <em>In silico</em> experiment</a></li>
<li class="chapter" data-level="6.2" data-path="mutations-detection.html"><a href="mutations-detection.html#swiss-data---schmid-siegert2017"><i class="fa fa-check"></i><b>6.2</b> Swiss data - <span class="citation">Schmid-Siegert <em>et al.</em> (2017)</span></a></li>
<li class="chapter" data-level="6.3" data-path="mutations-detection.html"><a href="mutations-detection.html#bordeaux-data---plomion2018"><i class="fa fa-check"></i><b>6.3</b> Bordeaux data - <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="6.4" data-path="mutations-detection.html"><a href="mutations-detection.html#angela"><i class="fa fa-check"></i><b>6.4</b> Angela</a></li>
<li class="chapter" data-level="6.5" data-path="mutations-detection.html"><a href="mutations-detection.html#sextonia-rubra-2"><i class="fa fa-check"></i><b>6.5</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="6.6" data-path="mutations-detection.html"><a href="mutations-detection.html#hetre"><i class="fa fa-check"></i><b>6.6</b> Hetre</a></li>
<li class="chapter" data-level="6.7" data-path="mutations-detection.html"><a href="mutations-detection.html#project-description-1"><i class="fa fa-check"></i><b>6.7</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mutations-in-silico.html"><a href="mutations-in-silico.html"><i class="fa fa-check"></i><b>7</b> Mutations <em>in silico</em></a></li>
<li class="chapter" data-level="8" data-path="mutations-swiss.html"><a href="mutations-swiss.html"><i class="fa fa-check"></i><b>8</b> Mutations Swiss</a>
<ul>
<li class="chapter" data-level="8.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#mutations-from-schmid-siegert-on-3p"><i class="fa fa-check"></i><b>8.1</b> Mutations from Schmid-Siegert on 3P</a></li>
<li class="chapter" data-level="8.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#strelka2"><i class="fa fa-check"></i><b>8.2</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert"><i class="fa fa-check"></i><b>8.2.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.2.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering"><i class="fa fa-check"></i><b>8.2.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="mutations-swiss.html"><a href="mutations-swiss.html#gatk"><i class="fa fa-check"></i><b>8.3</b> <code>GATK</code></a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert-1"><i class="fa fa-check"></i><b>8.3.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.3.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering-1"><i class="fa fa-check"></i><b>8.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="mutations-swiss.html"><a href="mutations-swiss.html#conclusion"><i class="fa fa-check"></i><b>8.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html"><i class="fa fa-check"></i><b>9</b> Mutations Bordeaux</a>
<ul>
<li class="chapter" data-level="9.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#original-mutations-on-3p"><i class="fa fa-check"></i><b>9.1</b> Original mutations on 3P</a></li>
<li class="chapter" data-level="9.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#coverage"><i class="fa fa-check"></i><b>9.2</b> Coverage</a></li>
<li class="chapter" data-level="9.3" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#strelka2-1"><i class="fa fa-check"></i><b>9.3</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations"><i class="fa fa-check"></i><b>9.3.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.3.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-2"><i class="fa fa-check"></i><b>9.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#mutect2"><i class="fa fa-check"></i><b>9.4</b> <code>Mutect2</code></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations-1"><i class="fa fa-check"></i><b>9.4.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.4.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-3"><i class="fa fa-check"></i><b>9.4.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#conclusion-1"><i class="fa fa-check"></i><b>9.5</b> Conclusion</a></li>
<li class="chapter" data-level="9.6" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#swiss-bordeaux"><i class="fa fa-check"></i><b>9.6</b> Swiss &amp; Bordeaux</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutations-angela.html"><a href="mutations-angela.html"><i class="fa fa-check"></i><b>10</b> Mutations Angela</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mutations-angela.html"><a href="mutations-angela.html#quality-check"><i class="fa fa-check"></i><b>10.1</b> Quality check</a></li>
<li class="chapter" data-level="10.2" data-path="mutations-angela.html"><a href="mutations-angela.html#genome"><i class="fa fa-check"></i><b>10.2</b> Genome</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mutations-angela.html"><a href="mutations-angela.html#annotation"><i class="fa fa-check"></i><b>10.2.1</b> Annotation</a></li>
<li class="chapter" data-level="10.2.2" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations"><i class="fa fa-check"></i><b>10.2.2</b> Mutations</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mutations-angela.html"><a href="mutations-angela.html#heterozygosity"><i class="fa fa-check"></i><b>10.3</b> Heterozygosity</a></li>
<li class="chapter" data-level="10.4" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-filters"><i class="fa fa-check"></i><b>10.4</b> Mutations filters</a></li>
<li class="chapter" data-level="10.5" data-path="mutations-angela.html"><a href="mutations-angela.html#leaf-mutations"><i class="fa fa-check"></i><b>10.5</b> Leaf mutations</a></li>
<li class="chapter" data-level="10.6" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-architecture"><i class="fa fa-check"></i><b>10.6</b> Mutations architecture</a></li>
<li class="chapter" data-level="10.7" data-path="mutations-angela.html"><a href="mutations-angela.html#phylogeny"><i class="fa fa-check"></i><b>10.7</b> Phylogeny</a></li>
<li class="chapter" data-level="10.8" data-path="mutations-angela.html"><a href="mutations-angela.html#light-annotation"><i class="fa fa-check"></i><b>10.8</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="10.9" data-path="mutations-angela.html"><a href="mutations-angela.html#rate"><i class="fa fa-check"></i><b>10.9</b> Rate</a></li>
<li class="chapter" data-level="10.10" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-for-fruits"><i class="fa fa-check"></i><b>10.10</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="10.11" data-path="mutations-angela.html"><a href="mutations-angela.html#genes"><i class="fa fa-check"></i><b>10.11</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="mutations-sixto.html"><a href="mutations-sixto.html"><i class="fa fa-check"></i><b>11</b> Mutations Sixto</a>
<ul>
<li class="chapter" data-level="11.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#quality-check-1"><i class="fa fa-check"></i><b>11.1</b> Quality check</a></li>
<li class="chapter" data-level="11.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genome-1"><i class="fa fa-check"></i><b>11.2</b> Genome</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#annotation-1"><i class="fa fa-check"></i><b>11.2.1</b> Annotation</a></li>
<li class="chapter" data-level="11.2.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-1"><i class="fa fa-check"></i><b>11.2.2</b> Mutations</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rubrynolide-enzymes"><i class="fa fa-check"></i><b>11.3</b> Rubrynolide enzymes</a></li>
<li class="chapter" data-level="11.4" data-path="mutations-sixto.html"><a href="mutations-sixto.html#heterozygosity-1"><i class="fa fa-check"></i><b>11.4</b> Heterozygosity</a></li>
<li class="chapter" data-level="11.5" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-filters-1"><i class="fa fa-check"></i><b>11.5</b> Mutations filters</a></li>
<li class="chapter" data-level="11.6" data-path="mutations-sixto.html"><a href="mutations-sixto.html#leaf-mutations-1"><i class="fa fa-check"></i><b>11.6</b> Leaf mutations</a></li>
<li class="chapter" data-level="11.7" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-architecture-1"><i class="fa fa-check"></i><b>11.7</b> Mutations architecture</a></li>
<li class="chapter" data-level="11.8" data-path="mutations-sixto.html"><a href="mutations-sixto.html#phylogeny-1"><i class="fa fa-check"></i><b>11.8</b> Phylogeny</a></li>
<li class="chapter" data-level="11.9" data-path="mutations-sixto.html"><a href="mutations-sixto.html#light-annotation-1"><i class="fa fa-check"></i><b>11.9</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="11.10" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rate-1"><i class="fa fa-check"></i><b>11.10</b> Rate</a></li>
<li class="chapter" data-level="11.11" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-for-fruits-1"><i class="fa fa-check"></i><b>11.11</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="11.12" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genes-1"><i class="fa fa-check"></i><b>11.12</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="comparisons-2.html"><a href="comparisons-2.html"><i class="fa fa-check"></i><b>12</b> Comparisons 2</a>
<ul>
<li class="chapter" data-level="12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#genomes"><i class="fa fa-check"></i><b>12.1</b> Genomes</a></li>
<li class="chapter" data-level="12.2" data-path="comparisons-2.html"><a href="comparisons-2.html#libraries"><i class="fa fa-check"></i><b>12.2</b> Libraries</a></li>
<li class="chapter" data-level="12.3" data-path="comparisons-2.html"><a href="comparisons-2.html#mutations-2"><i class="fa fa-check"></i><b>12.3</b> Mutations</a></li>
<li class="chapter" data-level="12.4" data-path="comparisons-2.html"><a href="comparisons-2.html#frequencies"><i class="fa fa-check"></i><b>12.4</b> Frequencies</a></li>
<li class="chapter" data-level="12.5" data-path="comparisons-2.html"><a href="comparisons-2.html#phylogeny-2"><i class="fa fa-check"></i><b>12.5</b> Phylogeny</a></li>
<li class="chapter" data-level="12.6" data-path="comparisons-2.html"><a href="comparisons-2.html#light"><i class="fa fa-check"></i><b>12.6</b> Light</a></li>
<li class="chapter" data-level="12.7" data-path="comparisons-2.html"><a href="comparisons-2.html#type"><i class="fa fa-check"></i><b>12.7</b> Type</a></li>
<li class="chapter" data-level="12.8" data-path="comparisons-2.html"><a href="comparisons-2.html#spectra"><i class="fa fa-check"></i><b>12.8</b> Spectra</a></li>
<li class="chapter" data-level="12.9" data-path="comparisons-2.html"><a href="comparisons-2.html#rates"><i class="fa fa-check"></i><b>12.9</b> Rates</a></li>
<li class="chapter" data-level="12.10" data-path="comparisons-2.html"><a href="comparisons-2.html#annotations"><i class="fa fa-check"></i><b>12.10</b> Annotations</a></li>
<li class="chapter" data-level="12.11" data-path="comparisons-2.html"><a href="comparisons-2.html#genes-2"><i class="fa fa-check"></i><b>12.11</b> Genes</a></li>
<li class="chapter" data-level="12.12" data-path="comparisons-2.html"><a href="comparisons-2.html#fruits"><i class="fa fa-check"></i><b>12.12</b> Fruits</a>
<ul>
<li class="chapter" data-level="12.12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#from-ssrseq-olivier"><i class="fa fa-check"></i><b>12.12.1</b> From SSRseq Olivier</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="comparisons-5.html"><a href="comparisons-5.html"><i class="fa fa-check"></i><b>13</b> Comparisons 5</a>
<ul>
<li class="chapter" data-level="13.1" data-path="comparisons-5.html"><a href="comparisons-5.html#genomes-1"><i class="fa fa-check"></i><b>13.1</b> Genomes</a></li>
<li class="chapter" data-level="13.2" data-path="comparisons-5.html"><a href="comparisons-5.html#libraries-1"><i class="fa fa-check"></i><b>13.2</b> Libraries</a></li>
<li class="chapter" data-level="13.3" data-path="comparisons-5.html"><a href="comparisons-5.html#mutations-3"><i class="fa fa-check"></i><b>13.3</b> Mutations</a></li>
<li class="chapter" data-level="13.4" data-path="comparisons-5.html"><a href="comparisons-5.html#frequencies-1"><i class="fa fa-check"></i><b>13.4</b> Frequencies</a></li>
<li class="chapter" data-level="13.5" data-path="comparisons-5.html"><a href="comparisons-5.html#types"><i class="fa fa-check"></i><b>13.5</b> Types</a></li>
<li class="chapter" data-level="13.6" data-path="comparisons-5.html"><a href="comparisons-5.html#spectra-1"><i class="fa fa-check"></i><b>13.6</b> Spectra</a></li>
<li class="chapter" data-level="13.7" data-path="comparisons-5.html"><a href="comparisons-5.html#rates-1"><i class="fa fa-check"></i><b>13.7</b> Rates</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="comparisons-5.html"><a href="comparisons-5.html#cummulated-distributions"><i class="fa fa-check"></i><b>13.7.1</b> Cummulated distributions</a></li>
<li class="chapter" data-level="13.7.2" data-path="comparisons-5.html"><a href="comparisons-5.html#lognormal-distributions"><i class="fa fa-check"></i><b>13.7.2</b> Lognormal distributions</a></li>
<li class="chapter" data-level="13.7.3" data-path="comparisons-5.html"><a href="comparisons-5.html#exponential-distrbutions"><i class="fa fa-check"></i><b>13.7.3</b> Exponential distrbutions</a></li>
<li class="chapter" data-level="13.7.4" data-path="comparisons-5.html"><a href="comparisons-5.html#power-law-distribution"><i class="fa fa-check"></i><b>13.7.4</b> Power law distribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="manuscript-figures.html"><a href="manuscript-figures.html"><i class="fa fa-check"></i><b>14</b> Manuscript figures</a>
<ul>
<li class="chapter" data-level="14.0.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-1"><i class="fa fa-check"></i><b>14.0.1</b> Figure 1</a></li>
<li class="chapter" data-level="14.0.2" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-2"><i class="fa fa-check"></i><b>14.0.2</b> Figure 2</a></li>
<li class="chapter" data-level="14.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-3"><i class="fa fa-check"></i><b>14.1</b> Figure 3</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html"><i class="fa fa-check"></i><b>A</b> Appendix 1: Meetings</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sylvains-kick-off"><i class="fa fa-check"></i><b>A.1</b> 11/02/2021 Sylvain’s Kick-off</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#funding"><i class="fa fa-check"></i><b>A.1.1</b> Funding</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#site-1"><i class="fa fa-check"></i><b>A.1.2</b> Site</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#species-1"><i class="fa fa-check"></i><b>A.1.3</b> Species</a></li>
<li class="chapter" data-level="A.1.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#field-work"><i class="fa fa-check"></i><b>A.1.4</b> Field work</a></li>
<li class="chapter" data-level="A.1.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#genomics"><i class="fa fa-check"></i><b>A.1.5</b> Genomics</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#william-marrande"><i class="fa fa-check"></i><b>A.2</b> 27/05/2021 William Marrande</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sampling-1"><i class="fa fa-check"></i><b>A.2.1</b> Sampling</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#assembly"><i class="fa fa-check"></i><b>A.2.2</b> Assembly</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutation-detection"><i class="fa fa-check"></i><b>A.3</b> 06/07/2021 Mutation detection</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#virtual-experiment"><i class="fa fa-check"></i><b>A.3.1</b> Virtual experiment</a></li>
<li class="chapter" data-level="A.3.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#swiss-data"><i class="fa fa-check"></i><b>A.3.2</b> Swiss data</a></li>
<li class="chapter" data-level="A.3.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#angela-1"><i class="fa fa-check"></i><b>A.3.3</b> Angela</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mid-project-meeting"><i class="fa fa-check"></i><b>A.4</b> 21/01/2022 Mid-project meeting</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#notes"><i class="fa fa-check"></i><b>A.4.1</b> Notes</a></li>
<li class="chapter" data-level="A.4.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#follow-up"><i class="fa fa-check"></i><b>A.4.2</b> Follow-up</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#carrotage"><i class="fa fa-check"></i><b>A.5</b> 10/02 Carrotage</a></li>
<li class="chapter" data-level="A.6" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sixtos-results"><i class="fa fa-check"></i><b>A.6</b> 05/05 Sixto’s results</a></li>
<li class="chapter" data-level="A.7" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutations-days"><i class="fa fa-check"></i><b>A.7</b> 20-21/03 Mutations days</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html"><i class="fa fa-check"></i><b>B</b> Appendix 2: Tree 3D</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation"><i class="fa fa-check"></i><b>B.1</b> Tree Segmentation</a></li>
<li class="chapter" data-level="B.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-architecture"><i class="fa fa-check"></i><b>B.2</b> Tree architecture</a></li>
<li class="chapter" data-level="B.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-light"><i class="fa fa-check"></i><b>B.3</b> Tree light</a></li>
<li class="chapter" data-level="B.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-1"><i class="fa fa-check"></i><b>B.4</b> Tree Segmentation</a>
<ul>
<li class="chapter" data-level="B.4.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeseg"><i class="fa fa-check"></i><b>B.4.1</b> <code>treeseg</code></a></li>
<li class="chapter" data-level="B.4.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#computree"><i class="fa fa-check"></i><b>B.4.2</b> <code>Computree</code></a></li>
<li class="chapter" data-level="B.4.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-with-lidr"><i class="fa fa-check"></i><b>B.4.3</b> Tree Segmentation with <code>lidr</code></a></li>
<li class="chapter" data-level="B.4.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#forest-metrics"><i class="fa fa-check"></i><b>B.4.4</b> Forest Metrics</a></li>
<li class="chapter" data-level="B.4.5" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#sp4e"><i class="fa fa-check"></i><b>B.4.5</b> sp4e</a></li>
<li class="chapter" data-level="B.4.6" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeqsm"><i class="fa fa-check"></i><b>B.4.6</b> TreeQSM</a></li>
<li class="chapter" data-level="B.4.7" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#google-search"><i class="fa fa-check"></i><b>B.4.7</b> Google Search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html"><i class="fa fa-check"></i><b>C</b> Appendix 3: Literature</a>
<ul>
<li class="chapter" data-level="C.0.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#bergeron2023"><i class="fa fa-check"></i><b>C.0.1</b> Bergeron2023</a></li>
<li class="chapter" data-level="C.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#genome-assembly-1"><i class="fa fa-check"></i><b>C.1</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#johnson2020"><i class="fa fa-check"></i><b>C.1.1</b> <span class="citation">Johnson <em>et al.</em> (2020)</span></a></li>
<li class="chapter" data-level="C.1.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#plomion2018"><i class="fa fa-check"></i><b>C.1.2</b> <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.1.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#msc"><i class="fa fa-check"></i><b>C.1.3</b> Msc</a></li>
<li class="chapter" data-level="C.1.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#si"><i class="fa fa-check"></i><b>C.1.4</b> SI</a></li>
<li class="chapter" data-level="C.1.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#httpswww.oakgenome.frpage_id587"><i class="fa fa-check"></i><b>C.1.5</b> https://www.oakgenome.fr/?page_id=587</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#mutation-detection-1"><i class="fa fa-check"></i><b>C.2</b> Mutation detection</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#orr2019"><i class="fa fa-check"></i><b>C.2.1</b> <span class="citation">Orr <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#ramu2013"><i class="fa fa-check"></i><b>C.2.2</b> <span class="citation">Ramu <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#hanlon2019"><i class="fa fa-check"></i><b>C.2.3</b> <span class="citation">Hanlon <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#cibulskis2013"><i class="fa fa-check"></i><b>C.2.4</b> <span class="citation">Cibulskis <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#nicholson2018"><i class="fa fa-check"></i><b>C.2.5</b> <span class="citation">Nicholson <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.2.6" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#alioto2015"><i class="fa fa-check"></i><b>C.2.6</b> <span class="citation">Alioto <em>et al.</em> (2015)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#unclassified"><i class="fa fa-check"></i><b>C.3</b> Unclassified</a>
<ul>
<li class="chapter" data-level="C.3.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#michael2020"><i class="fa fa-check"></i><b>C.3.1</b> <span class="citation">Michael &amp; VanBuren (2020)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#to-read"><i class="fa fa-check"></i><b>C.4</b> To read</a>
<ul>
<li class="chapter" data-level="C.4.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#belser2018"><i class="fa fa-check"></i><b>C.4.1</b> <span class="citation">Belser <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.4.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#edwards1990"><i class="fa fa-check"></i><b>C.4.2</b> <span class="citation">Edwards <em>et al.</em> (1990)</span></a></li>
<li class="chapter" data-level="C.4.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#leebens-mack2019"><i class="fa fa-check"></i><b>C.4.3</b> <span class="citation">Leebens-Mack <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.4.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#links"><i class="fa fa-check"></i><b>C.4.4</b> links</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TreeMutation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="comparisons-2" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Comparisons 2<a href="comparisons-2.html#comparisons-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter compares the mutations of Sixto and Angela on the basis of the closest possible filtering.</p>
<div id="genomes" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Genomes<a href="comparisons-2.html#genomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The total genome size of <em>Sextonia rubra</em> is larger than that of <em>Dicoryinia guyanensis</em>, but the <em>Dicoryinia guyanensis</em> genome was better assembled in fewer scaffolds (Fig. <a href="comparisons-2.html#fig:compgenomes2">12.1</a>).</p>
<div class="figure"><span style="display:block;" id="fig:compgenomes2"></span>
<img src="treemutation_files/figure-html/compgenomes2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.1: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:compgenomes2busco"></span>
<img src="treemutation_files/figure-html/compgenomes2busco-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.2: Caption.
</p>
</div>
</div>
<div id="libraries" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Libraries<a href="comparisons-2.html#libraries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As expected, Angela’s sequencing depth is twice as high as Sixto’s, averaging about 150X versus 80X (Fig. <a href="comparisons-2.html#fig:compcovs2">12.3</a>), which translates into greater accepted sequencing depth in Angela (note that as a result, Sixto’s libraries are less filtered in terms of coverage, Fig. <a href="comparisons-2.html#fig:compvolims2">12.4</a>).</p>
<div class="figure"><span style="display:block;" id="fig:compcovs2"></span>
<img src="treemutation_files/figure-html/compcovs2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.3: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:compvolims2"></span>
<img src="treemutation_files/figure-html/compvolims2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.4: Caption.
</p>
</div>
</div>
<div id="mutations-2" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Mutations<a href="comparisons-2.html#mutations-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The origins of the mutations are distributed in the crown, but most of the mutations with basic filtering come from the base of the crown, followed by the carpenter, the branches and finally the tips (Fig. <a href="comparisons-2.html#fig:mutcomp2">12.5</a>). Interestingly, stronger filtering favoured mutations in the tips as well as at the base of the crown.</p>
<div class="figure"><span style="display:block;" id="fig:mutcomp2"></span>
<img src="treemutation_files/figure-html/mutcomp2-1.png" alt="Caption." width="960" />
<p class="caption">
Figure 12.5: Caption.
</p>
</div>
<p><img src="treemutation_files/figure-html/unnamed-chunk-64-1.png" width="768" /></p>
</div>
<div id="frequencies" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> Frequencies<a href="comparisons-2.html#frequencies" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Surprisingly, the allelic frequencies of mutations occurring at the base of the crown are not significantly higher than those at the tips (Fig. <a href="comparisons-2.html#fig:afcomp2">12.6</a>).</p>
<div class="figure"><span style="display:block;" id="fig:afcomp2"></span>
<img src="treemutation_files/figure-html/afcomp2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.6: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:deltaafcomp2"></span>
<img src="treemutation_files/figure-html/deltaafcomp2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.7: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:deltaafcomp3"></span>
<img src="treemutation_files/figure-html/deltaafcomp3-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.8: Caption.
</p>
</div>
</div>
<div id="phylogeny-2" class="section level2 hasAnchor" number="12.5">
<h2><span class="header-section-number">12.5</span> Phylogeny<a href="comparisons-2.html#phylogeny-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Therefore, mutations are strongly shared across the crown and do not always follow the architecture. As a result, the phylogeny of the mutations does not match the architecture of the tree, with the exception of branch I, which is conserved in Sixto (Fig. <a href="comparisons-2.html#fig:angelaphyloarchi">12.9</a> &amp; Fig. <a href="#fig:sixtophyloarchi"><strong>??</strong></a>).</p>
<p><img src="treemutation_files/figure-html/unnamed-chunk-66-1.png" width="480" /></p>
<pre><code>## 
## Setting initial dates...
## Fitting in progress... get a first set of estimates
##          (Penalised) log-lik = -6.362752 
## Optimising rates... dates... -6.362752 
## Optimising rates... dates... -6.361982 
## 
## log-Lik = -6.306181 
## PHIIC = 66.66</code></pre>
<div class="figure"><span style="display:block;" id="fig:angelaphyloarchi"></span>
<img src="treemutation_files/figure-html/angelaphyloarchi-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.9: Caption.
</p>
</div>
</div>
<div id="light" class="section level2 hasAnchor" number="12.6">
<h2><span class="header-section-number">12.6</span> Light<a href="comparisons-2.html#light" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The light condition of the sampled tips at the time of sampling did not affect the number of mutations observed per library (Fig. <a href="#fig:lightComp2"><strong>??</strong></a>).</p>
<div class="figure"><span style="display:block;" id="fig:lightComp2base"></span>
<img src="treemutation_files/figure-html/lightComp2base-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.10: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:lightComp2baseKS"></span>
<img src="treemutation_files/figure-html/lightComp2baseKS-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.11: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:lightComp2type"></span>
<img src="treemutation_files/figure-html/lightComp2type-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.12: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:lightComp2spectraAngela"></span>
<img src="treemutation_files/figure-html/lightComp2spectraAngela-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.13: Caption.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:lightComp2spectraSixto"></span>
<img src="treemutation_files/figure-html/lightComp2spectraSixto-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.14: Caption.
</p>
</div>
</div>
<div id="type" class="section level2 hasAnchor" number="12.7">
<h2><span class="header-section-number">12.7</span> Type<a href="comparisons-2.html#type" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mutation types are similar between Angela and Sixto, with the exception of an increase in C-&gt;A and C-&gt;T but a decrease in T-&gt;A in Sixto compared to Angela, which are probably due to the sampling effect (Fig. <a href="comparisons-2.html#fig:typeComp2">12.15</a>, to be further investigated).</p>
<div class="figure"><span style="display:block;" id="fig:typeComp2"></span>
<img src="treemutation_files/figure-html/typeComp2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.15: Caption.
</p>
</div>
</div>
<div id="spectra" class="section level2 hasAnchor" number="12.8">
<h2><span class="header-section-number">12.8</span> Spectra<a href="comparisons-2.html#spectra" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The mutation spectra are similar between Angela and Sixto, with a few exceptions that are probably due to a sampling effect (Fig. <a href="comparisons-2.html#fig:spectraComp2">12.16</a>, to be explored further).</p>
<div class="figure"><span style="display:block;" id="fig:spectraComp2"></span>
<img src="treemutation_files/figure-html/spectraComp2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.16: Caption.
</p>
</div>
</div>
<div id="rates" class="section level2 hasAnchor" number="12.9">
<h2><span class="header-section-number">12.9</span> Rates<a href="comparisons-2.html#rates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Angela and Sixto show ten to ten thousand mutations depending on the filtering and the minimum accepted allelic frequency (Fig. <a href="comparisons-2.html#fig:ratesComp2">12.17</a>). Using base filtering, Angela has a higher number of mutations than Sixto, due to the twofold sequencing depth. Nevertheless, ev filtering gave a similar number of mutations close to thousands in both trees.</p>
<div class="figure"><span style="display:block;" id="fig:ratesComp2"></span>
<img src="treemutation_files/figure-html/ratesComp2-1.png" alt="Caption." width="768" />
<p class="caption">
Figure 12.17: Caption.
</p>
</div>
</div>
<div id="annotations" class="section level2 hasAnchor" number="12.10">
<h2><span class="header-section-number">12.10</span> Annotations<a href="comparisons-2.html#annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>In progress.</em></p>
</div>
<div id="genes-2" class="section level2 hasAnchor" number="12.11">
<h2><span class="header-section-number">12.11</span> Genes<a href="comparisons-2.html#genes-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>In progress.</em></p>
<p><img src="treemutation_files/figure-html/unnamed-chunk-67-1.png" width="768" /></p>
<pre><code>##               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## tree           1  15.52  15.522   87.89  &lt; 2e-16 ***
## synonymy       1  10.63  10.633   60.20 1.47e-14 ***
## Residuals   1698 299.90   0.177                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><img src="treemutation_files/figure-html/unnamed-chunk-68-1.png" width="768" /></p>
</div>
<div id="fruits" class="section level2 hasAnchor" number="12.12">
<h2><span class="header-section-number">12.12</span> Fruits<a href="comparisons-2.html#fruits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="from-ssrseq-olivier" class="section level3 hasAnchor" number="12.12.1">
<h3><span class="header-section-number">12.12.1</span> From SSRseq Olivier<a href="comparisons-2.html#from-ssrseq-olivier" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="align-genotypes" class="section level4 hasAnchor" number="12.12.1.1">
<h4><span class="header-section-number">12.12.1.1</span> Align genotypes<a href="comparisons-2.html#align-genotypes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="comparisons-2.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb7-2"><a href="comparisons-2.html#cb7-2" aria-hidden="true" tabindex="-1"></a>candidates <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb7-3"><a href="comparisons-2.html#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_tsv</span>(<span class="st">&quot;data/mutations/fruits/angela_fruits_candidate_mutations.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="comparisons-2.html#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mutation =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">tree =</span> <span class="st">&quot;Angela&quot;</span>),</span>
<span id="cb7-5"><a href="comparisons-2.html#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_tsv</span>(<span class="st">&quot;data/mutations/fruits/sixto_fruits_candidate_mutations.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="comparisons-2.html#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mutation =</span> <span class="dv">125</span><span class="sc">:</span>(<span class="dv">124</span><span class="sc">+</span><span class="fu">n</span>()), <span class="at">tree =</span> <span class="st">&quot;Sixto&quot;</span>)</span>
<span id="cb7-7"><a href="comparisons-2.html#cb7-7" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="comparisons-2.html#cb7-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(CHROM, POS, REF, ALT, af, branch, mutation, tree) <span class="sc">%&gt;%</span> </span>
<span id="cb7-9"><a href="comparisons-2.html#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CHROM =</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;Super-Scaffold_&quot;</span>, <span class="st">&quot;&quot;</span>, CHROM)))</span>
<span id="cb7-10"><a href="comparisons-2.html#cb7-10" aria-hidden="true" tabindex="-1"></a>alleles <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;data/mutations/fruits/SSRseq_TreeMutation_DataAnalysis.xlsx&quot;</span>, <span class="st">&quot;AlleleInformation&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-11"><a href="comparisons-2.html#cb7-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">locus =</span> Locus, <span class="at">genotype =</span> AlleleSeqCode, <span class="at">sequence =</span> AlleleSequence) <span class="sc">%&gt;%</span> </span>
<span id="cb7-12"><a href="comparisons-2.html#cb7-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(locus, genotype, sequence) <span class="sc">%&gt;%</span> </span>
<span id="cb7-13"><a href="comparisons-2.html#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">locus2 =</span> <span class="fu">gsub</span>(<span class="st">&quot;TreeMut_SNP-&quot;</span>, <span class="st">&quot;&quot;</span>, locus)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-14"><a href="comparisons-2.html#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(locus2, <span class="fu">c</span>(<span class="st">&quot;mutation&quot;</span>, <span class="st">&quot;pos&quot;</span>), <span class="st">&quot;_Super-Scaffold_&quot;</span>, <span class="at">convert =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb7-15"><a href="comparisons-2.html#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(mutation, <span class="at">convert =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb7-16"><a href="comparisons-2.html#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(pos, <span class="st">&quot;CHROM&quot;</span>, <span class="at">convert =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb7-17"><a href="comparisons-2.html#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(candidates) <span class="sc">%&gt;%</span> </span>
<span id="cb7-18"><a href="comparisons-2.html#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">paste0</span>(<span class="st">&quot;SNV&quot;</span>, <span class="fu">sprintf</span>(<span class="st">&quot;%03d&quot;</span>, mutation), <span class="st">&quot;_A&quot;</span>, genotype))</span>
<span id="cb7-19"><a href="comparisons-2.html#cb7-19" aria-hidden="true" tabindex="-1"></a>angela_alleles <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">filter</span>(alleles, tree <span class="sc">==</span> <span class="st">&quot;Angela&quot;</span>)<span class="sc">$</span>sequence)</span>
<span id="cb7-20"><a href="comparisons-2.html#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(angela_alleles) <span class="ot">&lt;-</span> <span class="fu">filter</span>(alleles, tree <span class="sc">==</span> <span class="st">&quot;Angela&quot;</span>)<span class="sc">$</span>name</span>
<span id="cb7-21"><a href="comparisons-2.html#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(angela_alleles, <span class="st">&quot;data/mutations/fruits/angela_alleles.fa&quot;</span>)</span>
<span id="cb7-22"><a href="comparisons-2.html#cb7-22" aria-hidden="true" tabindex="-1"></a>sixto_alleles <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(<span class="fu">filter</span>(alleles, tree <span class="sc">==</span> <span class="st">&quot;Sixto&quot;</span>)<span class="sc">$</span>sequence)</span>
<span id="cb7-23"><a href="comparisons-2.html#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sixto_alleles) <span class="ot">&lt;-</span> <span class="fu">filter</span>(alleles, tree <span class="sc">==</span> <span class="st">&quot;Sixto&quot;</span>)<span class="sc">$</span>name</span>
<span id="cb7-24"><a href="comparisons-2.html#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(sixto_alleles, <span class="st">&quot;data/mutations/fruits/sixto_alleles.fa&quot;</span>)</span>
<span id="cb7-25"><a href="comparisons-2.html#cb7-25" aria-hidden="true" tabindex="-1"></a>refs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;data/mutations/fruits/treemutation_fruits.xlsx&quot;</span>, <span class="st">&quot;Sixto&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-26"><a href="comparisons-2.html#cb7-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">tree =</span> <span class="st">&quot;Sixto&quot;</span>),</span>
<span id="cb7-27"><a href="comparisons-2.html#cb7-27" aria-hidden="true" tabindex="-1"></a>          readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;data/mutations/fruits/treemutation_fruits.xlsx&quot;</span>, <span class="st">&quot;Angela&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-28"><a href="comparisons-2.html#cb7-28" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">tree =</span> <span class="st">&quot;Angela&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-29"><a href="comparisons-2.html#cb7-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">library =</span> <span class="fu">paste0</span>(Code_Espece, Tube), <span class="at">tissue =</span> Tissu) <span class="sc">%&gt;%</span></span>
<span id="cb7-30"><a href="comparisons-2.html#cb7-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(library, tree, tissue) <span class="sc">%&gt;%</span> </span>
<span id="cb7-31"><a href="comparisons-2.html#cb7-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;data/mutations/fruits/Angela.xlsx&quot;</span>, <span class="st">&quot;libraries&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-32"><a href="comparisons-2.html#cb7-32" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">tissue =</span> <span class="fu">paste</span>(<span class="st">&quot;branch&quot;</span>, branch), <span class="at">library =</span> idOld, <span class="at">tree =</span> <span class="st">&quot;Angela&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-33"><a href="comparisons-2.html#cb7-33" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(library, tree, tissue)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-34"><a href="comparisons-2.html#cb7-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;data/mutations/fruits/Sixto.xlsx&quot;</span>, <span class="st">&quot;samples&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-35"><a href="comparisons-2.html#cb7-35" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">tissue =</span> <span class="fu">paste</span>(<span class="st">&quot;branch&quot;</span>, Branch), <span class="at">library =</span> id, <span class="at">tree =</span> <span class="st">&quot;Sixto&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-36"><a href="comparisons-2.html#cb7-36" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(library, tree, tissue))</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="comparisons-2.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="at">-t</span> 2 ../angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa angela_alleles.fa <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="op">&gt;</span> angela_alleles_aligned.bam</span>
<span id="cb8-2"><a href="comparisons-2.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index angela_alleles_aligned.bam</span>
<span id="cb8-3"><a href="comparisons-2.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="at">-t</span> 2 ../sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa sixto_alleles.fa <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="op">&gt;</span> sixto_alleles_aligned.bam</span>
<span id="cb8-4"><a href="comparisons-2.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index sixto_alleles_aligned.bam</span></code></pre></div>
</div>
<div id="automate-igv" class="section level4 hasAnchor" number="12.12.1.2">
<h4><span class="header-section-number">12.12.1.2</span> Automate IGV<a href="comparisons-2.html#automate-igv" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="comparisons-2.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> igvreports python=3.7.1</span>
<span id="cb9-2"><a href="comparisons-2.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate igvreports</span>
<span id="cb9-3"><a href="comparisons-2.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install igv-reports</span>
<span id="cb9-4"><a href="comparisons-2.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="comparisons-2.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate igvreports</span>
<span id="cb10-2"><a href="comparisons-2.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="ex">create_report</span> data/mutations/fruits/angela_fruits_candidate_mutations.tsv <span class="dt">\</span></span>
<span id="cb10-3"><a href="comparisons-2.html#cb10-3" aria-hidden="true" tabindex="-1"></a>data/mutations/angela/genome/Dgu_HS1_HYBRID_SCAFFOLD.fa <span class="dt">\</span></span>
<span id="cb10-4"><a href="comparisons-2.html#cb10-4" aria-hidden="true" tabindex="-1"></a>--sequence 2 <span class="at">--begin</span> 3 <span class="at">--end</span> 3 <span class="dt">\</span></span>
<span id="cb10-5"><a href="comparisons-2.html#cb10-5" aria-hidden="true" tabindex="-1"></a>--flanking 1000 <span class="dt">\</span></span>
<span id="cb10-6"><a href="comparisons-2.html#cb10-6" aria-hidden="true" tabindex="-1"></a>--info-columns SNV CHROM POS REF ALT af replicate branch rank <span class="dt">\</span></span>
<span id="cb10-7"><a href="comparisons-2.html#cb10-7" aria-hidden="true" tabindex="-1"></a>--tracks data/mutations/fruits/angela_alleles_aligned.bam data/mutations/angela/annotation/Dgu_HS1_HYBRID_SCAFFOLD.fa.out.gff <span class="dt">\</span></span>
<span id="cb10-8"><a href="comparisons-2.html#cb10-8" aria-hidden="true" tabindex="-1"></a>--output data/mutations/fruits/angela_fruits_aligned.html</span>
<span id="cb10-9"><a href="comparisons-2.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="comparisons-2.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate igvreports</span>
<span id="cb11-2"><a href="comparisons-2.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="ex">create_report</span> data/mutations/fruits/sixto_fruits_candidate_mutations.tsv <span class="dt">\</span></span>
<span id="cb11-3"><a href="comparisons-2.html#cb11-3" aria-hidden="true" tabindex="-1"></a>data/mutations/sixto/genome/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa <span class="dt">\</span></span>
<span id="cb11-4"><a href="comparisons-2.html#cb11-4" aria-hidden="true" tabindex="-1"></a>--sequence 2 <span class="at">--begin</span> 3 <span class="at">--end</span> 3 <span class="dt">\</span></span>
<span id="cb11-5"><a href="comparisons-2.html#cb11-5" aria-hidden="true" tabindex="-1"></a>--flanking 1000 <span class="dt">\</span></span>
<span id="cb11-6"><a href="comparisons-2.html#cb11-6" aria-hidden="true" tabindex="-1"></a>--info-columns SNV CHROM POS REF ALT af replicate branch rank <span class="dt">\</span></span>
<span id="cb11-7"><a href="comparisons-2.html#cb11-7" aria-hidden="true" tabindex="-1"></a>--tracks data/mutations/fruits/sixto_alleles_aligned.bam data/mutations/sixto/annotation/HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa.out.gff <span class="dt">\</span></span>
<span id="cb11-8"><a href="comparisons-2.html#cb11-8" aria-hidden="true" tabindex="-1"></a>--output data/mutations/fruits/sixto_fruits_aligned.html</span>
<span id="cb11-9"><a href="comparisons-2.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
</div>
<div id="results" class="section level4 hasAnchor" number="12.12.1.3">
<h4><span class="header-section-number">12.12.1.3</span> Results<a href="comparisons-2.html#results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>##             Type Angela Sixto Total Percentage
## 1       mutation     16     5    21         15
## 2 reference only     81    20   101         72
## 3        suspect     11     6    17         12
## 4      unaligned     16     5    21         15</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="comparisons-2.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>( <span class="st">&quot;data/mutations/fruits/fruits_mutations.pdf&quot;</span>)</span>
<span id="cb13-2"><a href="comparisons-2.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>ggforce<span class="sc">::</span><span class="fu">n_pages</span>(graph))</span>
<span id="cb13-3"><a href="comparisons-2.html#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(graph <span class="sc">+</span> ggforce<span class="sc">::</span><span class="fu">facet_wrap_paginate</span>(<span class="sc">~</span>title, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>,</span>
<span id="cb13-4"><a href="comparisons-2.html#cb13-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">page =</span> i))</span>
<span id="cb13-5"><a href="comparisons-2.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p><img src="treemutation_files/figure-html/exgraphfruitsolivier-1.png" width="768" /></p>
</div>
<div id="tissues" class="section level4 hasAnchor" number="12.12.1.4">
<h4><span class="header-section-number">12.12.1.4</span> Tissues<a href="comparisons-2.html#tissues" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="treemutation_files/figure-html/unnamed-chunk-75-1.png" width="768" /></p>
<table>
<thead>
<tr>
<th style="text-align:left;">
SNV
</th>
<th style="text-align:right;">
FALSE
</th>
<th style="text-align:right;">
TRUE
</th>
<th style="text-align:right;">
agreement
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SNV006
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV013
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV031
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV054
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
98
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV057
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
85
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV107
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
88
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV128
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV132
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV151
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV160
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
100
</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr>
<th style="text-align:left;">
fruit
</th>
<th style="text-align:left;">
cotyledon
</th>
<th style="text-align:left;">
embryo sac
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A10
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A11
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A12
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A13
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A14
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A15
</td>
<td style="text-align:left;">
heterozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A16
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A17
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A18
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A19
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A2
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A20
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A3
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A4
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A5
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A6
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A7
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A8
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
A9
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B10
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B11
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B12
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B13
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B14
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B15
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B16
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B17
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B18
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B19
</td>
<td style="text-align:left;">
heterozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B2
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B20
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B21
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B22
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B23
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
B24
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B25
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B3
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B4
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B5
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B6
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B7
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B8
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
B9
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
C1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
D1
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
D2
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
D3
</td>
<td style="text-align:left;">
homozygous
</td>
<td style="text-align:left;">
heterozygous
</td>
</tr>
<tr>
<td style="text-align:left;">
D5
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
homozygous
</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr>
<th style="text-align:left;">
SNV
</th>
<th style="text-align:right;">
cotyledon
</th>
<th style="text-align:right;">
embryo sac
</th>
<th style="text-align:right;">
endocarpe
</th>
<th style="text-align:right;">
pericarpe
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SNV006
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV013
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV031
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV054
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV057
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV107
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV128
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV132
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV151
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV160
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations-sixto.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="comparisons-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
