<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Mutations Swiss | TreeMutation</title>
  <meta name="description" content="Mutation in the tropical tree canopy." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Mutations Swiss | TreeMutation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Mutation in the tropical tree canopy." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Mutations Swiss | TreeMutation" />
  
  <meta name="twitter:description" content="Mutation in the tropical tree canopy." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2023-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations-in-silico.html"/>
<link rel="next" href="mutations-bordeaux.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.0/leaflet.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.29/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TreeMutation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="species.html"><a href="species.html"><i class="fa fa-check"></i><b>1</b> Species</a>
<ul>
<li class="chapter" data-level="1.1" data-path="species.html"><a href="species.html#dicorynia-guianensis"><i class="fa fa-check"></i><b>1.1</b> <em>Dicorynia guianensis</em></a></li>
<li class="chapter" data-level="1.2" data-path="species.html"><a href="species.html#sextonia-rubra"><i class="fa fa-check"></i><b>1.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="1.3" data-path="species.html"><a href="species.html#eschweilera-coriacea"><i class="fa fa-check"></i><b>1.3</b> <em>Eschweilera coriacea</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="site.html"><a href="site.html"><i class="fa fa-check"></i><b>2</b> Site</a>
<ul>
<li class="chapter" data-level="2.1" data-path="site.html"><a href="site.html#paracou"><i class="fa fa-check"></i><b>2.1</b> Paracou</a></li>
<li class="chapter" data-level="2.2" data-path="site.html"><a href="site.html#régina"><i class="fa fa-check"></i><b>2.2</b> Régina</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="individuals.html"><a href="individuals.html"><i class="fa fa-check"></i><b>3</b> Individuals</a>
<ul>
<li class="chapter" data-level="3.1" data-path="individuals.html"><a href="individuals.html#angela---dicorynia-guyanensis"><i class="fa fa-check"></i><b>3.1</b> <strong>Angela</strong> - <em>Dicorynia guyanensis</em></a></li>
<li class="chapter" data-level="3.2" data-path="individuals.html"><a href="individuals.html#sextonia-rubra-1"><i class="fa fa-check"></i><b>3.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="3.3" data-path="individuals.html"><a href="individuals.html#eschweilera-sp."><i class="fa fa-check"></i><b>3.3</b> <em>Eschweilera sp.</em></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="individuals.html"><a href="individuals.html#mn5505---regina"><i class="fa fa-check"></i><b>3.3.1</b> MN5505 - Regina</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>4</b> Sampling</a></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#sequencing"><i class="fa fa-check"></i><b>5.1</b> Sequencing</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#assembly-workflow"><i class="fa fa-check"></i><b>5.2</b> Assembly workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-assembly.html"><a href="genome-assembly.html#v1"><i class="fa fa-check"></i><b>5.2.1</b> V1</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-assembly.html"><a href="genome-assembly.html#v2"><i class="fa fa-check"></i><b>5.2.2</b> V2</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#annotation-workflow"><i class="fa fa-check"></i><b>5.3</b> Annotation workflow</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#project-description"><i class="fa fa-check"></i><b>5.4</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mutations-detection.html"><a href="mutations-detection.html"><i class="fa fa-check"></i><b>6</b> Mutations detection</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mutations-detection.html"><a href="mutations-detection.html#in-silico-experiment"><i class="fa fa-check"></i><b>6.1</b> <em>In silico</em> experiment</a></li>
<li class="chapter" data-level="6.2" data-path="mutations-detection.html"><a href="mutations-detection.html#swiss-data---schmid-siegert2017"><i class="fa fa-check"></i><b>6.2</b> Swiss data - <span class="citation">Schmid-Siegert <em>et al.</em> (2017)</span></a></li>
<li class="chapter" data-level="6.3" data-path="mutations-detection.html"><a href="mutations-detection.html#bordeaux-data---plomion2018"><i class="fa fa-check"></i><b>6.3</b> Bordeaux data - <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="6.4" data-path="mutations-detection.html"><a href="mutations-detection.html#angela"><i class="fa fa-check"></i><b>6.4</b> Angela</a></li>
<li class="chapter" data-level="6.5" data-path="mutations-detection.html"><a href="mutations-detection.html#sextonia-rubra-2"><i class="fa fa-check"></i><b>6.5</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="6.6" data-path="mutations-detection.html"><a href="mutations-detection.html#hetre"><i class="fa fa-check"></i><b>6.6</b> Hetre</a></li>
<li class="chapter" data-level="6.7" data-path="mutations-detection.html"><a href="mutations-detection.html#project-description-1"><i class="fa fa-check"></i><b>6.7</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mutations-in-silico.html"><a href="mutations-in-silico.html"><i class="fa fa-check"></i><b>7</b> Mutations <em>in silico</em></a></li>
<li class="chapter" data-level="8" data-path="mutations-swiss.html"><a href="mutations-swiss.html"><i class="fa fa-check"></i><b>8</b> Mutations Swiss</a>
<ul>
<li class="chapter" data-level="8.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#mutations-from-schmid-siegert-on-3p"><i class="fa fa-check"></i><b>8.1</b> Mutations from Schmid-Siegert on 3P</a></li>
<li class="chapter" data-level="8.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#strelka2"><i class="fa fa-check"></i><b>8.2</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert"><i class="fa fa-check"></i><b>8.2.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.2.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering"><i class="fa fa-check"></i><b>8.2.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="mutations-swiss.html"><a href="mutations-swiss.html#gatk"><i class="fa fa-check"></i><b>8.3</b> <code>GATK</code></a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert-1"><i class="fa fa-check"></i><b>8.3.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.3.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering-1"><i class="fa fa-check"></i><b>8.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="mutations-swiss.html"><a href="mutations-swiss.html#conclusion"><i class="fa fa-check"></i><b>8.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html"><i class="fa fa-check"></i><b>9</b> Mutations Bordeaux</a>
<ul>
<li class="chapter" data-level="9.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#original-mutations-on-3p"><i class="fa fa-check"></i><b>9.1</b> Original mutations on 3P</a></li>
<li class="chapter" data-level="9.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#coverage"><i class="fa fa-check"></i><b>9.2</b> Coverage</a></li>
<li class="chapter" data-level="9.3" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#strelka2-1"><i class="fa fa-check"></i><b>9.3</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations"><i class="fa fa-check"></i><b>9.3.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.3.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-2"><i class="fa fa-check"></i><b>9.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#mutect2"><i class="fa fa-check"></i><b>9.4</b> <code>Mutect2</code></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations-1"><i class="fa fa-check"></i><b>9.4.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.4.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-3"><i class="fa fa-check"></i><b>9.4.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#conclusion-1"><i class="fa fa-check"></i><b>9.5</b> Conclusion</a></li>
<li class="chapter" data-level="9.6" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#swiss-bordeaux"><i class="fa fa-check"></i><b>9.6</b> Swiss &amp; Bordeaux</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutations-angela.html"><a href="mutations-angela.html"><i class="fa fa-check"></i><b>10</b> Mutations Angela</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mutations-angela.html"><a href="mutations-angela.html#quality-check"><i class="fa fa-check"></i><b>10.1</b> Quality check</a></li>
<li class="chapter" data-level="10.2" data-path="mutations-angela.html"><a href="mutations-angela.html#genome"><i class="fa fa-check"></i><b>10.2</b> Genome</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mutations-angela.html"><a href="mutations-angela.html#annotation"><i class="fa fa-check"></i><b>10.2.1</b> Annotation</a></li>
<li class="chapter" data-level="10.2.2" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations"><i class="fa fa-check"></i><b>10.2.2</b> Mutations</a></li>
<li class="chapter" data-level="10.2.3" data-path="mutations-angela.html"><a href="mutations-angela.html#comp"><i class="fa fa-check"></i><b>10.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mutations-angela.html"><a href="mutations-angela.html#heterozygosity"><i class="fa fa-check"></i><b>10.3</b> Heterozygosity</a></li>
<li class="chapter" data-level="10.4" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-filters"><i class="fa fa-check"></i><b>10.4</b> Mutations filters</a></li>
<li class="chapter" data-level="10.5" data-path="mutations-angela.html"><a href="mutations-angela.html#leaf-mutations"><i class="fa fa-check"></i><b>10.5</b> Leaf mutations</a></li>
<li class="chapter" data-level="10.6" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-architecture"><i class="fa fa-check"></i><b>10.6</b> Mutations architecture</a></li>
<li class="chapter" data-level="10.7" data-path="mutations-angela.html"><a href="mutations-angela.html#phylogeny"><i class="fa fa-check"></i><b>10.7</b> Phylogeny</a></li>
<li class="chapter" data-level="10.8" data-path="mutations-angela.html"><a href="mutations-angela.html#light-annotation"><i class="fa fa-check"></i><b>10.8</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="10.9" data-path="mutations-angela.html"><a href="mutations-angela.html#rate"><i class="fa fa-check"></i><b>10.9</b> Rate</a></li>
<li class="chapter" data-level="10.10" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-for-fruits"><i class="fa fa-check"></i><b>10.10</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="10.11" data-path="mutations-angela.html"><a href="mutations-angela.html#genes"><i class="fa fa-check"></i><b>10.11</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="mutations-sixto.html"><a href="mutations-sixto.html"><i class="fa fa-check"></i><b>11</b> Mutations Sixto</a>
<ul>
<li class="chapter" data-level="11.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#quality-check-1"><i class="fa fa-check"></i><b>11.1</b> Quality check</a></li>
<li class="chapter" data-level="11.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genome-1"><i class="fa fa-check"></i><b>11.2</b> Genome</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#annotation-1"><i class="fa fa-check"></i><b>11.2.1</b> Annotation</a></li>
<li class="chapter" data-level="11.2.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-1"><i class="fa fa-check"></i><b>11.2.2</b> Mutations</a></li>
<li class="chapter" data-level="11.2.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#comp-1"><i class="fa fa-check"></i><b>11.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rubrynolide-enzymes"><i class="fa fa-check"></i><b>11.3</b> Rubrynolide enzymes</a></li>
<li class="chapter" data-level="11.4" data-path="mutations-sixto.html"><a href="mutations-sixto.html#heterozygosity-1"><i class="fa fa-check"></i><b>11.4</b> Heterozygosity</a></li>
<li class="chapter" data-level="11.5" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-filters-1"><i class="fa fa-check"></i><b>11.5</b> Mutations filters</a></li>
<li class="chapter" data-level="11.6" data-path="mutations-sixto.html"><a href="mutations-sixto.html#leaf-mutations-1"><i class="fa fa-check"></i><b>11.6</b> Leaf mutations</a></li>
<li class="chapter" data-level="11.7" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-architecture-1"><i class="fa fa-check"></i><b>11.7</b> Mutations architecture</a></li>
<li class="chapter" data-level="11.8" data-path="mutations-sixto.html"><a href="mutations-sixto.html#phylogeny-1"><i class="fa fa-check"></i><b>11.8</b> Phylogeny</a></li>
<li class="chapter" data-level="11.9" data-path="mutations-sixto.html"><a href="mutations-sixto.html#light-annotation-1"><i class="fa fa-check"></i><b>11.9</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="11.10" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rate-1"><i class="fa fa-check"></i><b>11.10</b> Rate</a></li>
<li class="chapter" data-level="11.11" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-for-fruits-1"><i class="fa fa-check"></i><b>11.11</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="11.12" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genes-1"><i class="fa fa-check"></i><b>11.12</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="comparisons-2.html"><a href="comparisons-2.html"><i class="fa fa-check"></i><b>12</b> Comparisons 2</a>
<ul>
<li class="chapter" data-level="12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#genomes"><i class="fa fa-check"></i><b>12.1</b> Genomes</a></li>
<li class="chapter" data-level="12.2" data-path="comparisons-2.html"><a href="comparisons-2.html#libraries"><i class="fa fa-check"></i><b>12.2</b> Libraries</a></li>
<li class="chapter" data-level="12.3" data-path="comparisons-2.html"><a href="comparisons-2.html#mutations-2"><i class="fa fa-check"></i><b>12.3</b> Mutations</a></li>
<li class="chapter" data-level="12.4" data-path="comparisons-2.html"><a href="comparisons-2.html#frequencies"><i class="fa fa-check"></i><b>12.4</b> Frequencies</a></li>
<li class="chapter" data-level="12.5" data-path="comparisons-2.html"><a href="comparisons-2.html#phylogeny-2"><i class="fa fa-check"></i><b>12.5</b> Phylogeny</a></li>
<li class="chapter" data-level="12.6" data-path="comparisons-2.html"><a href="comparisons-2.html#light"><i class="fa fa-check"></i><b>12.6</b> Light</a></li>
<li class="chapter" data-level="12.7" data-path="comparisons-2.html"><a href="comparisons-2.html#type"><i class="fa fa-check"></i><b>12.7</b> Type</a></li>
<li class="chapter" data-level="12.8" data-path="comparisons-2.html"><a href="comparisons-2.html#spectra"><i class="fa fa-check"></i><b>12.8</b> Spectra</a></li>
<li class="chapter" data-level="12.9" data-path="comparisons-2.html"><a href="comparisons-2.html#rates"><i class="fa fa-check"></i><b>12.9</b> Rates</a></li>
<li class="chapter" data-level="12.10" data-path="comparisons-2.html"><a href="comparisons-2.html#annotations"><i class="fa fa-check"></i><b>12.10</b> Annotations</a></li>
<li class="chapter" data-level="12.11" data-path="comparisons-2.html"><a href="comparisons-2.html#genes-2"><i class="fa fa-check"></i><b>12.11</b> Genes</a></li>
<li class="chapter" data-level="12.12" data-path="comparisons-2.html"><a href="comparisons-2.html#fruits"><i class="fa fa-check"></i><b>12.12</b> Fruits</a>
<ul>
<li class="chapter" data-level="12.12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#from-ssrseq-olivier"><i class="fa fa-check"></i><b>12.12.1</b> From SSRseq Olivier</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="comparisons-5.html"><a href="comparisons-5.html"><i class="fa fa-check"></i><b>13</b> Comparisons 5</a>
<ul>
<li class="chapter" data-level="13.1" data-path="comparisons-5.html"><a href="comparisons-5.html#genomes-1"><i class="fa fa-check"></i><b>13.1</b> Genomes</a></li>
<li class="chapter" data-level="13.2" data-path="comparisons-5.html"><a href="comparisons-5.html#libraries-1"><i class="fa fa-check"></i><b>13.2</b> Libraries</a></li>
<li class="chapter" data-level="13.3" data-path="comparisons-5.html"><a href="comparisons-5.html#mutations-3"><i class="fa fa-check"></i><b>13.3</b> Mutations</a></li>
<li class="chapter" data-level="13.4" data-path="comparisons-5.html"><a href="comparisons-5.html#frequencies-2"><i class="fa fa-check"></i><b>13.4</b> Frequencies</a></li>
<li class="chapter" data-level="13.5" data-path="comparisons-5.html"><a href="comparisons-5.html#types"><i class="fa fa-check"></i><b>13.5</b> Types</a></li>
<li class="chapter" data-level="13.6" data-path="comparisons-5.html"><a href="comparisons-5.html#spectra-1"><i class="fa fa-check"></i><b>13.6</b> Spectra</a></li>
<li class="chapter" data-level="13.7" data-path="comparisons-5.html"><a href="comparisons-5.html#rates-1"><i class="fa fa-check"></i><b>13.7</b> Rates</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="comparisons-5.html"><a href="comparisons-5.html#cummulated-distributions"><i class="fa fa-check"></i><b>13.7.1</b> Cummulated distributions</a></li>
<li class="chapter" data-level="13.7.2" data-path="comparisons-5.html"><a href="comparisons-5.html#lognormal-distributions"><i class="fa fa-check"></i><b>13.7.2</b> Lognormal distributions</a></li>
<li class="chapter" data-level="13.7.3" data-path="comparisons-5.html"><a href="comparisons-5.html#exponential-distrbutions"><i class="fa fa-check"></i><b>13.7.3</b> Exponential distrbutions</a></li>
<li class="chapter" data-level="13.7.4" data-path="comparisons-5.html"><a href="comparisons-5.html#power-law-distribution"><i class="fa fa-check"></i><b>13.7.4</b> Power law distribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="manuscript-figures.html"><a href="manuscript-figures.html"><i class="fa fa-check"></i><b>14</b> Manuscript figures</a>
<ul>
<li class="chapter" data-level="14.0.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-1"><i class="fa fa-check"></i><b>14.0.1</b> Figure 1</a></li>
<li class="chapter" data-level="14.0.2" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-2"><i class="fa fa-check"></i><b>14.0.2</b> Figure 2</a></li>
<li class="chapter" data-level="14.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-3"><i class="fa fa-check"></i><b>14.1</b> Figure 3</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html"><i class="fa fa-check"></i><b>15</b> Manuscript figures EVS</a>
<ul>
<li class="chapter" data-level="15.1" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-2-evs"><i class="fa fa-check"></i><b>15.1</b> Figure 2 EVS</a></li>
<li class="chapter" data-level="15.2" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-3-evs"><i class="fa fa-check"></i><b>15.2</b> Figure 3 EVS</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html"><i class="fa fa-check"></i><b>A</b> Appendix 1: Meetings</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sylvains-kick-off"><i class="fa fa-check"></i><b>A.1</b> 11/02/2021 Sylvain’s Kick-off</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#funding"><i class="fa fa-check"></i><b>A.1.1</b> Funding</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#site-1"><i class="fa fa-check"></i><b>A.1.2</b> Site</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#species-1"><i class="fa fa-check"></i><b>A.1.3</b> Species</a></li>
<li class="chapter" data-level="A.1.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#field-work"><i class="fa fa-check"></i><b>A.1.4</b> Field work</a></li>
<li class="chapter" data-level="A.1.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#genomics"><i class="fa fa-check"></i><b>A.1.5</b> Genomics</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#william-marrande"><i class="fa fa-check"></i><b>A.2</b> 27/05/2021 William Marrande</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sampling-1"><i class="fa fa-check"></i><b>A.2.1</b> Sampling</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#assembly"><i class="fa fa-check"></i><b>A.2.2</b> Assembly</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutation-detection"><i class="fa fa-check"></i><b>A.3</b> 06/07/2021 Mutation detection</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#virtual-experiment"><i class="fa fa-check"></i><b>A.3.1</b> Virtual experiment</a></li>
<li class="chapter" data-level="A.3.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#swiss-data"><i class="fa fa-check"></i><b>A.3.2</b> Swiss data</a></li>
<li class="chapter" data-level="A.3.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#angela-1"><i class="fa fa-check"></i><b>A.3.3</b> Angela</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mid-project-meeting"><i class="fa fa-check"></i><b>A.4</b> 21/01/2022 Mid-project meeting</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#notes"><i class="fa fa-check"></i><b>A.4.1</b> Notes</a></li>
<li class="chapter" data-level="A.4.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#follow-up"><i class="fa fa-check"></i><b>A.4.2</b> Follow-up</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#carrotage"><i class="fa fa-check"></i><b>A.5</b> 10/02 Carrotage</a></li>
<li class="chapter" data-level="A.6" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sixtos-results"><i class="fa fa-check"></i><b>A.6</b> 05/05 Sixto’s results</a></li>
<li class="chapter" data-level="A.7" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutations-days"><i class="fa fa-check"></i><b>A.7</b> 20-21/03 Mutations days</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html"><i class="fa fa-check"></i><b>B</b> Appendix 2: Tree 3D</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation"><i class="fa fa-check"></i><b>B.1</b> Tree Segmentation</a></li>
<li class="chapter" data-level="B.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-architecture"><i class="fa fa-check"></i><b>B.2</b> Tree architecture</a></li>
<li class="chapter" data-level="B.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-light"><i class="fa fa-check"></i><b>B.3</b> Tree light</a></li>
<li class="chapter" data-level="B.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-1"><i class="fa fa-check"></i><b>B.4</b> Tree Segmentation</a>
<ul>
<li class="chapter" data-level="B.4.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeseg"><i class="fa fa-check"></i><b>B.4.1</b> <code>treeseg</code></a></li>
<li class="chapter" data-level="B.4.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#computree"><i class="fa fa-check"></i><b>B.4.2</b> <code>Computree</code></a></li>
<li class="chapter" data-level="B.4.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-with-lidr"><i class="fa fa-check"></i><b>B.4.3</b> Tree Segmentation with <code>lidr</code></a></li>
<li class="chapter" data-level="B.4.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#forest-metrics"><i class="fa fa-check"></i><b>B.4.4</b> Forest Metrics</a></li>
<li class="chapter" data-level="B.4.5" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#sp4e"><i class="fa fa-check"></i><b>B.4.5</b> sp4e</a></li>
<li class="chapter" data-level="B.4.6" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeqsm"><i class="fa fa-check"></i><b>B.4.6</b> TreeQSM</a></li>
<li class="chapter" data-level="B.4.7" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#google-search"><i class="fa fa-check"></i><b>B.4.7</b> Google Search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html"><i class="fa fa-check"></i><b>C</b> Appendix 3: Literature</a>
<ul>
<li class="chapter" data-level="C.0.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#bergeron2023"><i class="fa fa-check"></i><b>C.0.1</b> Bergeron2023</a></li>
<li class="chapter" data-level="C.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#genome-assembly-1"><i class="fa fa-check"></i><b>C.1</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#johnson2020"><i class="fa fa-check"></i><b>C.1.1</b> <span class="citation">Johnson <em>et al.</em> (2020)</span></a></li>
<li class="chapter" data-level="C.1.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#plomion2018"><i class="fa fa-check"></i><b>C.1.2</b> <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.1.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#msc"><i class="fa fa-check"></i><b>C.1.3</b> Msc</a></li>
<li class="chapter" data-level="C.1.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#si"><i class="fa fa-check"></i><b>C.1.4</b> SI</a></li>
<li class="chapter" data-level="C.1.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#httpswww.oakgenome.frpage_id587"><i class="fa fa-check"></i><b>C.1.5</b> https://www.oakgenome.fr/?page_id=587</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#mutation-detection-1"><i class="fa fa-check"></i><b>C.2</b> Mutation detection</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#orr2019"><i class="fa fa-check"></i><b>C.2.1</b> <span class="citation">Orr <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#ramu2013"><i class="fa fa-check"></i><b>C.2.2</b> <span class="citation">Ramu <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#hanlon2019"><i class="fa fa-check"></i><b>C.2.3</b> <span class="citation">Hanlon <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#cibulskis2013"><i class="fa fa-check"></i><b>C.2.4</b> <span class="citation">Cibulskis <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#nicholson2018"><i class="fa fa-check"></i><b>C.2.5</b> <span class="citation">Nicholson <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.2.6" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#alioto2015"><i class="fa fa-check"></i><b>C.2.6</b> <span class="citation">Alioto <em>et al.</em> (2015)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#unclassified"><i class="fa fa-check"></i><b>C.3</b> Unclassified</a>
<ul>
<li class="chapter" data-level="C.3.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#michael2020"><i class="fa fa-check"></i><b>C.3.1</b> <span class="citation">Michael &amp; VanBuren (2020)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#to-read"><i class="fa fa-check"></i><b>C.4</b> To read</a>
<ul>
<li class="chapter" data-level="C.4.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#belser2018"><i class="fa fa-check"></i><b>C.4.1</b> <span class="citation">Belser <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.4.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#edwards1990"><i class="fa fa-check"></i><b>C.4.2</b> <span class="citation">Edwards <em>et al.</em> (1990)</span></a></li>
<li class="chapter" data-level="C.4.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#leebens-mack2019"><i class="fa fa-check"></i><b>C.4.3</b> <span class="citation">Leebens-Mack <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.4.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#links"><i class="fa fa-check"></i><b>C.4.4</b> links</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TreeMutation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutations-swiss" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Mutations Swiss<a href="mutations-swiss.html#mutations-swiss" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter describes the reanalyses of data from <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span> currently done in the <a href="https://github.com/sylvainschmitt/detectMutations/tree/swiss"><code>swiss</code> branch of the <code>detectMutations</code> repository</a>.</p>
<div id="mutations-from-schmid-siegert-on-3p" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Mutations from Schmid-Siegert on 3P<a href="mutations-swiss.html#mutations-from-schmid-siegert-on-3p" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I reported the mutations (Tab. <a href="mutations-swiss.html#tab:napoTab">8.1</a>) from the <a href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41477-017-0066-9/MediaObjects/41477_2017_66_MOESM1_ESM.pdf">supplementary table 2</a> from <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span>, and aligned them back on the 3P genome (Fig. <a href="mutations-swiss.html#fig:napoFig">8.1</a>).
I found back only 14 of the original 17 mutations from Napoleon</p>
<table>
<caption>
<span id="tab:napoTab">Table 8.1: </span>SNVs in the Napoleon Oak. Rerported from <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span>.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Mutation
</th>
<th style="text-align:left;">
CHROM
</th>
<th style="text-align:right;">
POS
</th>
<th style="text-align:left;">
tumor
</th>
<th style="text-align:left;">
REF
</th>
<th style="text-align:left;">
ALT
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SNV1
</td>
<td style="text-align:left;">
Qrob_H2.3_Sc0000161
</td>
<td style="text-align:right;">
602274
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV2
</td>
<td style="text-align:left;">
Qrob_Chr09
</td>
<td style="text-align:right;">
2166134
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV3
</td>
<td style="text-align:left;">
Qrob_Chr05
</td>
<td style="text-align:right;">
56685984
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV4
</td>
<td style="text-align:left;">
Qrob_Chr05
</td>
<td style="text-align:right;">
25205243
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV5
</td>
<td style="text-align:left;">
Qrob_Chr05
</td>
<td style="text-align:right;">
44871901
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV6
</td>
<td style="text-align:left;">
Qrob_Chr01
</td>
<td style="text-align:right;">
39504822
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV7
</td>
<td style="text-align:left;">
Qrob_H2.3_Sc0000061
</td>
<td style="text-align:right;">
381637
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV8
</td>
<td style="text-align:left;">
Qrob_Chr04
</td>
<td style="text-align:right;">
37893394
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV9
</td>
<td style="text-align:left;">
Qrob_H2.3_Sc0000504
</td>
<td style="text-align:right;">
516352
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV10
</td>
<td style="text-align:left;">
Qrob_Chr04
</td>
<td style="text-align:right;">
36278589
</td>
<td style="text-align:left;">
Upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV13
</td>
<td style="text-align:left;">
Qrob_Chr08
</td>
<td style="text-align:right;">
54574817
</td>
<td style="text-align:left;">
Lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV14
</td>
<td style="text-align:left;">
Qrob_Chr09
</td>
<td style="text-align:right;">
30863836
</td>
<td style="text-align:left;">
Lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV16
</td>
<td style="text-align:left;">
Qrob_Chr05
</td>
<td style="text-align:right;">
9535260
</td>
<td style="text-align:left;">
Lower
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV17
</td>
<td style="text-align:left;">
Qrob_Chr07
</td>
<td style="text-align:right;">
215451
</td>
<td style="text-align:left;">
Lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
</tr>
</tbody>
</table>
<div class="figure"><span style="display:block;" id="fig:napoFig"></span>
<img src="treemutation_files/figure-html/napoFig-1.png" alt="Napoleon's original mutations on the 3P genome." width="768" />
<p class="caption">
Figure 8.1: Napoleon’s original mutations on the 3P genome.
</p>
</div>
</div>
<div id="strelka2" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> <code>Strelka2</code><a href="mutations-swiss.html#strelka2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>Strelka2</code> produced 4.2 millions of candidate mutations.</p>
<div id="overlap-with-mutations-from-schmid-siegert" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Overlap with mutations from Schmid-Siegert<a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I tried to find back Napoleon’s original mutations to have a look to their metrics.
I found back only 12 out of the 14 expected mutations (86%) (Tab. <a href="mutations-swiss.html#tab:napoMutTab">8.2</a> and Fig. <a href="mutations-swiss.html#fig:napoMutFig">8.2</a>).
<strong>Beware, <code>Strelka2</code> is detecting putative mutations in the normal sample !</strong>
I looked at different metrics for each (Fig. <a href="mutations-swiss.html#fig:overlapMetrics">8.3</a>):</p>
<ul>
<li><code>mutation_DP</code> and <code>normal_DP</code> are the read depth for the two sample, and shows as expected values between half and two times the mean coverage (60X)</li>
<li><code>normal_altCountT1</code> is the number of alternate allele count in the normal sample, should be 0, but is equal to 3 and 4 (9% of reads) for two SNVs</li>
<li><code>mutation_altCountT1</code> is the number of alternate allele count in the mutated sample, should be not too low, and is most the time above 5</li>
</ul>
<p><strong>The main conclusion is that the mutations detected by <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span> have not always no reads in the “normal” sample and that they show a wide variation of allelic frequency.</strong></p>
<table>
<caption>
<span id="tab:napoMutTab">Table 8.2: </span>Overlap between candidate mutations and Napoleon’s original mutations.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Mutation
</th>
<th style="text-align:left;">
Mutated
</th>
<th style="text-align:left;">
Normal
</th>
<th style="text-align:left;">
Ref
</th>
<th style="text-align:left;">
Alt
</th>
<th style="text-align:right;">
Mutated
AltCount
</th>
<th style="text-align:right;">
Mutated
RefCount
</th>
<th style="text-align:right;">
Normal
AltCount
</th>
<th style="text-align:right;">
Normal
RefCount
</th>
<th style="text-align:right;">
Allelic fraction
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SNV1
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
71
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV1
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
43
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
71
</td>
<td style="text-align:right;">
0.2500000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV2
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
90
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV2
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
90
</td>
<td style="text-align:right;">
0.2631579
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV3
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV3
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.2200000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV4
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
60
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV4
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
0.2500000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV5
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV5
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
63
</td>
<td style="text-align:right;">
0.2826087
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV6
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV6
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
48
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.2545455
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV8
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV8
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
0.2727273
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV9
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV9
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
63
</td>
<td style="text-align:right;">
0.3809524
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV10
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV10
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
71
</td>
<td style="text-align:right;">
0.2500000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV13
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
0.3958333
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV13
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.0833333
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV16
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
42
</td>
<td style="text-align:right;">
0.1694915
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV17
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
55
</td>
<td style="text-align:right;">
0.3287671
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV17
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0.0677966
</td>
</tr>
</tbody>
</table>
<div class="figure"><span style="display:block;" id="fig:napoMutFig"></span>
<img src="treemutation_files/figure-html/napoMutFig-1.png" alt="Overlap between candidate mutations and Napoleon's original mutations: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.2: Overlap between candidate mutations and Napoleon’s original mutations: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:overlapMetrics"></span>
<img src="treemutation_files/figure-html/overlapMetrics-1.png" alt="Evaluation of the overlap between candidate mutations and Napoleon's original mutations." width="768" />
<p class="caption">
Figure 8.3: Evaluation of the overlap between candidate mutations and Napoleon’s original mutations.
</p>
</div>
</div>
<div id="filtering" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Filtering<a href="mutations-swiss.html#filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We filtered mutations with following filters:</p>
<ul>
<li>A read depth for the two sample between half and two times the mean coverage (<code>normal_DP &lt;= 120, normal_DP &gt;= 30, mutation_DP &lt;= 120, mutation_DP &gt;= 30</code>)</li>
<li>A null number of alternate allele count in the normal sample (<code>normal_altCount == 0</code>)</li>
<li>A minimum of 10 alternate allele count in the mutated sample (<code>mutation_altCount &gt;= 10</code>)</li>
</ul>
<p>We obtained 223 candidates (Fig <a href="mutations-swiss.html#fig:mutFilteredS">8.4</a>).
We then used the suggested automatic filter of <code>Strelka2</code>,
resulting in a robust dataset of 87 mutations (Fig <a href="mutations-swiss.html#fig:mutRobustS">8.5</a>).</p>
<div class="figure"><span style="display:block;" id="fig:mutFilteredS"></span>
<img src="treemutation_files/figure-html/mutFilteredS-1.png" alt="Mutations retained after original filtering: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.4: Mutations retained after original filtering: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:mutRobustS"></span>
<img src="treemutation_files/figure-html/mutRobustS-1.png" alt="Mutations retained after robust filtering: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.5: Mutations retained after robust filtering: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
</div>
</div>
<div id="gatk" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> <code>GATK</code><a href="mutations-swiss.html#gatk" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>GATK</code> produced 11.1 millions of candidates!</p>
<div id="overlap-with-mutations-from-schmid-siegert-1" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Overlap with mutations from Schmid-Siegert<a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I tried to find back Napoleon’s original mutations to have a look to their metrics.
I found back only 12 out of the 14 expected mutations (86%) (Tab. <a href="mutations-swiss.html#tab:napoMutTab2">8.3</a> and Fig. <a href="mutations-swiss.html#fig:napoMutFig2">8.6</a>).
I looked at different metrics for each (Fig. <a href="mutations-swiss.html#fig:overlapMetrics">8.3</a>):</p>
<ul>
<li><code>mutation_DP</code> and <code>normal_DP</code> are the read depth for the two sample, and shows as expected values between half and two times the mean coverage (60X)</li>
<li><code>normal_altCountT1</code> is the number of alternate allele count in the normal sample, should be and is 0</li>
<li><code>mutation_altCountT1</code> is the number of alternate allele count in the mutated sample, should be not too low, and is most the time above 10</li>
</ul>
<p><strong>The main conclusion is that the mutations detected by <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span> have no reads in the “normal” sample using <code>GATK</code> with hard filtering which probably already removed low-DP copies in the normal sample, while <code>Strelka2</code> detect them.</strong></p>
<table>
<caption>
<span id="tab:napoMutTab2">Table 8.3: </span>Overlap between candidate mutations and Napoleon’s original mutations.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Mutation
</th>
<th style="text-align:left;">
Mutated
</th>
<th style="text-align:left;">
Normal
</th>
<th style="text-align:right;">
Mutated
AltCount
</th>
<th style="text-align:right;">
Mutated
RefCount
</th>
<th style="text-align:right;">
Normal
AltCount
</th>
<th style="text-align:left;">
Normal
RefCount
</th>
<th style="text-align:right;">
Allelic fraction
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SNV1
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2678571
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV2
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2631579
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV3
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2040816
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV4
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2343750
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV5
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2954545
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV6
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2241379
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV8
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2608696
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV9
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.4146341
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV10
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.2500000
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV13
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.3958333
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV16
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.1694915
</td>
</tr>
<tr>
<td style="text-align:left;">
SNV17
</td>
<td style="text-align:left;">
upper
</td>
<td style="text-align:left;">
lower
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
0.3289474
</td>
</tr>
</tbody>
</table>
<div class="figure"><span style="display:block;" id="fig:napoMutFig2"></span>
<img src="treemutation_files/figure-html/napoMutFig2-1.png" alt="Overlap between candidate mutations and Napoleon's original mutations: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.6: Overlap between candidate mutations and Napoleon’s original mutations: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:overlapMetrics2"></span>
<img src="treemutation_files/figure-html/overlapMetrics2-1.png" alt="Evaluation of the overlap between candidate mutations and Napoleon's original mutations." width="768" />
<p class="caption">
Figure 8.7: Evaluation of the overlap between candidate mutations and Napoleon’s original mutations.
</p>
</div>
</div>
<div id="filtering-1" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Filtering<a href="mutations-swiss.html#filtering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We filtered mutations with following filters:</p>
<ul>
<li>A read depth for the tumor sample between half and two times the mean coverage (<code>tumor_DP &lt;= 120, tumor_DP &gt;= 30</code>)</li>
<li>A null number of alternate allele count in the normal sample (<code>normal_altCount == 0</code>)</li>
<li>A minimum of 10 alternate allele count in the mutated sample (<code>tumor_altCount &gt;= 10</code>)</li>
<li>An allelic frequency inferior to 0.5 (<code>tumor_AF &lt;= 0.5</code>)</li>
</ul>
<p>We obtained 510 611 candidates (Fig <a href="mutations-swiss.html#fig:mutFilteredG">8.8</a>).
We then looked for the overlap between <code>GATK</code> candidates and the suggested automatic filter of <code>Strelka2</code>,
resulting in a robust dataset of 47 mutations (Fig <a href="mutations-swiss.html#fig:mutRobustG">8.9</a>).</p>
<div class="figure"><span style="display:block;" id="fig:mutFilteredG"></span>
<img src="treemutation_files/figure-html/mutFilteredG-1.png" alt="Mutations retained after original filtering: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.8: Mutations retained after original filtering: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:mutRobustG"></span>
<img src="treemutation_files/figure-html/mutRobustG-1.png" alt="Mutations retained after robust filtering: allele frequency (A) and positions on the 3P genome (B)." width="768" />
<p class="caption">
Figure 8.9: Mutations retained after robust filtering: allele frequency (A) and positions on the 3P genome (B).
</p>
</div>
</div>
</div>
<div id="conclusion" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Conclusion<a href="mutations-swiss.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption>
<span id="tab:datasetsswissTab">Table 8.4: </span>Size of the different datasets.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
Number of candidates
</th>
<th style="text-align:right;">
Estimated total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Schmid
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
17
</td>
</tr>
<tr>
<td style="text-align:left;">
GATK
</td>
<td style="text-align:right;">
510 611
</td>
<td style="text-align:right;">
709 182
</td>
</tr>
<tr>
<td style="text-align:left;">
Strelka2
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
Robust
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
57
</td>
</tr>
</tbody>
</table>
<div class="figure"><span style="display:block;" id="fig:datasetsswissAF"></span>
<img src="treemutation_files/figure-html/datasetsswissAF-1.png" alt="Allele frequency for the different datasets." width="768" />
<p class="caption">
Figure 8.10: Allele frequency for the different datasets.
</p>
</div>
<p><img src="treemutation_files/figure-html/datasetsswissVenn2-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/swissMutType-1.png" width="768" /></p>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Schmid-Siegert2017" class="csl-entry">
Schmid-Siegert, E., Sarkar, N., Iseli, C., Calderon, S., Gouhier-Darimont, C., Chrast, J., Cattaneo, P., Schütz, F., Farinelli, L., Pagni, M., Schneider, M., Voumard, J., Jaboyedoff, M., Fankhauser, C., Hardtke, C.S., Keller, L., Pannell, J.R., Reymond, A., Robinson-Rechavi, M., Xenarios, I. &amp; Reymond, P. (2017). Low number of fixed somatic mutations in a long-lived oak tree. <em>Nature Plants</em>, <strong>3</strong>, 926–929. Retrieved from <a href="http://dx.doi.org/10.1038/s41477-017-0066-9">http://dx.doi.org/10.1038/s41477-017-0066-9</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations-in-silico.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mutations-bordeaux.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
