<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Mutations Sixto | TreeMutation</title>
  <meta name="description" content="Mutation in the tropical tree canopy." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Mutations Sixto | TreeMutation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Mutation in the tropical tree canopy." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Mutations Sixto | TreeMutation" />
  
  <meta name="twitter:description" content="Mutation in the tropical tree canopy." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2023-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mutations-angela.html"/>
<link rel="next" href="comparisons-2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.0/leaflet.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.29/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TreeMutation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="species.html"><a href="species.html"><i class="fa fa-check"></i><b>1</b> Species</a>
<ul>
<li class="chapter" data-level="1.1" data-path="species.html"><a href="species.html#dicorynia-guianensis"><i class="fa fa-check"></i><b>1.1</b> <em>Dicorynia guianensis</em></a></li>
<li class="chapter" data-level="1.2" data-path="species.html"><a href="species.html#sextonia-rubra"><i class="fa fa-check"></i><b>1.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="1.3" data-path="species.html"><a href="species.html#eschweilera-coriacea"><i class="fa fa-check"></i><b>1.3</b> <em>Eschweilera coriacea</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="site.html"><a href="site.html"><i class="fa fa-check"></i><b>2</b> Site</a>
<ul>
<li class="chapter" data-level="2.1" data-path="site.html"><a href="site.html#paracou"><i class="fa fa-check"></i><b>2.1</b> Paracou</a></li>
<li class="chapter" data-level="2.2" data-path="site.html"><a href="site.html#régina"><i class="fa fa-check"></i><b>2.2</b> Régina</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="individuals.html"><a href="individuals.html"><i class="fa fa-check"></i><b>3</b> Individuals</a>
<ul>
<li class="chapter" data-level="3.1" data-path="individuals.html"><a href="individuals.html#angela---dicorynia-guyanensis"><i class="fa fa-check"></i><b>3.1</b> <strong>Angela</strong> - <em>Dicorynia guyanensis</em></a></li>
<li class="chapter" data-level="3.2" data-path="individuals.html"><a href="individuals.html#sextonia-rubra-1"><i class="fa fa-check"></i><b>3.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="3.3" data-path="individuals.html"><a href="individuals.html#eschweilera-sp."><i class="fa fa-check"></i><b>3.3</b> <em>Eschweilera sp.</em></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="individuals.html"><a href="individuals.html#mn5505---regina"><i class="fa fa-check"></i><b>3.3.1</b> MN5505 - Regina</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>4</b> Sampling</a></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#sequencing"><i class="fa fa-check"></i><b>5.1</b> Sequencing</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#assembly-workflow"><i class="fa fa-check"></i><b>5.2</b> Assembly workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-assembly.html"><a href="genome-assembly.html#v1"><i class="fa fa-check"></i><b>5.2.1</b> V1</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-assembly.html"><a href="genome-assembly.html#v2"><i class="fa fa-check"></i><b>5.2.2</b> V2</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#annotation-workflow"><i class="fa fa-check"></i><b>5.3</b> Annotation workflow</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#project-description"><i class="fa fa-check"></i><b>5.4</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mutations-detection.html"><a href="mutations-detection.html"><i class="fa fa-check"></i><b>6</b> Mutations detection</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mutations-detection.html"><a href="mutations-detection.html#in-silico-experiment"><i class="fa fa-check"></i><b>6.1</b> <em>In silico</em> experiment</a></li>
<li class="chapter" data-level="6.2" data-path="mutations-detection.html"><a href="mutations-detection.html#swiss-data---schmid-siegert2017"><i class="fa fa-check"></i><b>6.2</b> Swiss data - <span class="citation">Schmid-Siegert <em>et al.</em> (2017)</span></a></li>
<li class="chapter" data-level="6.3" data-path="mutations-detection.html"><a href="mutations-detection.html#bordeaux-data---plomion2018"><i class="fa fa-check"></i><b>6.3</b> Bordeaux data - <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="6.4" data-path="mutations-detection.html"><a href="mutations-detection.html#angela"><i class="fa fa-check"></i><b>6.4</b> Angela</a></li>
<li class="chapter" data-level="6.5" data-path="mutations-detection.html"><a href="mutations-detection.html#sextonia-rubra-2"><i class="fa fa-check"></i><b>6.5</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="6.6" data-path="mutations-detection.html"><a href="mutations-detection.html#hetre"><i class="fa fa-check"></i><b>6.6</b> Hetre</a></li>
<li class="chapter" data-level="6.7" data-path="mutations-detection.html"><a href="mutations-detection.html#project-description-1"><i class="fa fa-check"></i><b>6.7</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mutations-in-silico.html"><a href="mutations-in-silico.html"><i class="fa fa-check"></i><b>7</b> Mutations <em>in silico</em></a></li>
<li class="chapter" data-level="8" data-path="mutations-swiss.html"><a href="mutations-swiss.html"><i class="fa fa-check"></i><b>8</b> Mutations Swiss</a>
<ul>
<li class="chapter" data-level="8.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#mutations-from-schmid-siegert-on-3p"><i class="fa fa-check"></i><b>8.1</b> Mutations from Schmid-Siegert on 3P</a></li>
<li class="chapter" data-level="8.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#strelka2"><i class="fa fa-check"></i><b>8.2</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert"><i class="fa fa-check"></i><b>8.2.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.2.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering"><i class="fa fa-check"></i><b>8.2.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="mutations-swiss.html"><a href="mutations-swiss.html#gatk"><i class="fa fa-check"></i><b>8.3</b> <code>GATK</code></a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert-1"><i class="fa fa-check"></i><b>8.3.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.3.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering-1"><i class="fa fa-check"></i><b>8.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="mutations-swiss.html"><a href="mutations-swiss.html#conclusion"><i class="fa fa-check"></i><b>8.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html"><i class="fa fa-check"></i><b>9</b> Mutations Bordeaux</a>
<ul>
<li class="chapter" data-level="9.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#original-mutations-on-3p"><i class="fa fa-check"></i><b>9.1</b> Original mutations on 3P</a></li>
<li class="chapter" data-level="9.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#coverage"><i class="fa fa-check"></i><b>9.2</b> Coverage</a></li>
<li class="chapter" data-level="9.3" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#strelka2-1"><i class="fa fa-check"></i><b>9.3</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations"><i class="fa fa-check"></i><b>9.3.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.3.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-2"><i class="fa fa-check"></i><b>9.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#mutect2"><i class="fa fa-check"></i><b>9.4</b> <code>Mutect2</code></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations-1"><i class="fa fa-check"></i><b>9.4.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.4.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-3"><i class="fa fa-check"></i><b>9.4.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#conclusion-1"><i class="fa fa-check"></i><b>9.5</b> Conclusion</a></li>
<li class="chapter" data-level="9.6" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#swiss-bordeaux"><i class="fa fa-check"></i><b>9.6</b> Swiss &amp; Bordeaux</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutations-angela.html"><a href="mutations-angela.html"><i class="fa fa-check"></i><b>10</b> Mutations Angela</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mutations-angela.html"><a href="mutations-angela.html#quality-check"><i class="fa fa-check"></i><b>10.1</b> Quality check</a></li>
<li class="chapter" data-level="10.2" data-path="mutations-angela.html"><a href="mutations-angela.html#genome"><i class="fa fa-check"></i><b>10.2</b> Genome</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mutations-angela.html"><a href="mutations-angela.html#annotation"><i class="fa fa-check"></i><b>10.2.1</b> Annotation</a></li>
<li class="chapter" data-level="10.2.2" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations"><i class="fa fa-check"></i><b>10.2.2</b> Mutations</a></li>
<li class="chapter" data-level="10.2.3" data-path="mutations-angela.html"><a href="mutations-angela.html#comp"><i class="fa fa-check"></i><b>10.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mutations-angela.html"><a href="mutations-angela.html#heterozygosity"><i class="fa fa-check"></i><b>10.3</b> Heterozygosity</a></li>
<li class="chapter" data-level="10.4" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-filters"><i class="fa fa-check"></i><b>10.4</b> Mutations filters</a></li>
<li class="chapter" data-level="10.5" data-path="mutations-angela.html"><a href="mutations-angela.html#leaf-mutations"><i class="fa fa-check"></i><b>10.5</b> Leaf mutations</a></li>
<li class="chapter" data-level="10.6" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-architecture"><i class="fa fa-check"></i><b>10.6</b> Mutations architecture</a></li>
<li class="chapter" data-level="10.7" data-path="mutations-angela.html"><a href="mutations-angela.html#phylogeny"><i class="fa fa-check"></i><b>10.7</b> Phylogeny</a></li>
<li class="chapter" data-level="10.8" data-path="mutations-angela.html"><a href="mutations-angela.html#light-annotation"><i class="fa fa-check"></i><b>10.8</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="10.9" data-path="mutations-angela.html"><a href="mutations-angela.html#rate"><i class="fa fa-check"></i><b>10.9</b> Rate</a></li>
<li class="chapter" data-level="10.10" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-for-fruits"><i class="fa fa-check"></i><b>10.10</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="10.11" data-path="mutations-angela.html"><a href="mutations-angela.html#genes"><i class="fa fa-check"></i><b>10.11</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="mutations-sixto.html"><a href="mutations-sixto.html"><i class="fa fa-check"></i><b>11</b> Mutations Sixto</a>
<ul>
<li class="chapter" data-level="11.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#quality-check-1"><i class="fa fa-check"></i><b>11.1</b> Quality check</a></li>
<li class="chapter" data-level="11.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genome-1"><i class="fa fa-check"></i><b>11.2</b> Genome</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#annotation-1"><i class="fa fa-check"></i><b>11.2.1</b> Annotation</a></li>
<li class="chapter" data-level="11.2.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-1"><i class="fa fa-check"></i><b>11.2.2</b> Mutations</a></li>
<li class="chapter" data-level="11.2.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#comp-1"><i class="fa fa-check"></i><b>11.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rubrynolide-enzymes"><i class="fa fa-check"></i><b>11.3</b> Rubrynolide enzymes</a></li>
<li class="chapter" data-level="11.4" data-path="mutations-sixto.html"><a href="mutations-sixto.html#heterozygosity-1"><i class="fa fa-check"></i><b>11.4</b> Heterozygosity</a></li>
<li class="chapter" data-level="11.5" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-filters-1"><i class="fa fa-check"></i><b>11.5</b> Mutations filters</a></li>
<li class="chapter" data-level="11.6" data-path="mutations-sixto.html"><a href="mutations-sixto.html#leaf-mutations-1"><i class="fa fa-check"></i><b>11.6</b> Leaf mutations</a></li>
<li class="chapter" data-level="11.7" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-architecture-1"><i class="fa fa-check"></i><b>11.7</b> Mutations architecture</a></li>
<li class="chapter" data-level="11.8" data-path="mutations-sixto.html"><a href="mutations-sixto.html#phylogeny-1"><i class="fa fa-check"></i><b>11.8</b> Phylogeny</a></li>
<li class="chapter" data-level="11.9" data-path="mutations-sixto.html"><a href="mutations-sixto.html#light-annotation-1"><i class="fa fa-check"></i><b>11.9</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="11.10" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rate-1"><i class="fa fa-check"></i><b>11.10</b> Rate</a></li>
<li class="chapter" data-level="11.11" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-for-fruits-1"><i class="fa fa-check"></i><b>11.11</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="11.12" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genes-1"><i class="fa fa-check"></i><b>11.12</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="comparisons-2.html"><a href="comparisons-2.html"><i class="fa fa-check"></i><b>12</b> Comparisons 2</a>
<ul>
<li class="chapter" data-level="12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#genomes"><i class="fa fa-check"></i><b>12.1</b> Genomes</a></li>
<li class="chapter" data-level="12.2" data-path="comparisons-2.html"><a href="comparisons-2.html#libraries"><i class="fa fa-check"></i><b>12.2</b> Libraries</a></li>
<li class="chapter" data-level="12.3" data-path="comparisons-2.html"><a href="comparisons-2.html#mutations-2"><i class="fa fa-check"></i><b>12.3</b> Mutations</a></li>
<li class="chapter" data-level="12.4" data-path="comparisons-2.html"><a href="comparisons-2.html#frequencies"><i class="fa fa-check"></i><b>12.4</b> Frequencies</a></li>
<li class="chapter" data-level="12.5" data-path="comparisons-2.html"><a href="comparisons-2.html#phylogeny-2"><i class="fa fa-check"></i><b>12.5</b> Phylogeny</a></li>
<li class="chapter" data-level="12.6" data-path="comparisons-2.html"><a href="comparisons-2.html#light"><i class="fa fa-check"></i><b>12.6</b> Light</a></li>
<li class="chapter" data-level="12.7" data-path="comparisons-2.html"><a href="comparisons-2.html#type"><i class="fa fa-check"></i><b>12.7</b> Type</a></li>
<li class="chapter" data-level="12.8" data-path="comparisons-2.html"><a href="comparisons-2.html#spectra"><i class="fa fa-check"></i><b>12.8</b> Spectra</a></li>
<li class="chapter" data-level="12.9" data-path="comparisons-2.html"><a href="comparisons-2.html#rates"><i class="fa fa-check"></i><b>12.9</b> Rates</a></li>
<li class="chapter" data-level="12.10" data-path="comparisons-2.html"><a href="comparisons-2.html#annotations"><i class="fa fa-check"></i><b>12.10</b> Annotations</a></li>
<li class="chapter" data-level="12.11" data-path="comparisons-2.html"><a href="comparisons-2.html#genes-2"><i class="fa fa-check"></i><b>12.11</b> Genes</a></li>
<li class="chapter" data-level="12.12" data-path="comparisons-2.html"><a href="comparisons-2.html#fruits"><i class="fa fa-check"></i><b>12.12</b> Fruits</a>
<ul>
<li class="chapter" data-level="12.12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#from-ssrseq-olivier"><i class="fa fa-check"></i><b>12.12.1</b> From SSRseq Olivier</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="comparisons-5.html"><a href="comparisons-5.html"><i class="fa fa-check"></i><b>13</b> Comparisons 5</a>
<ul>
<li class="chapter" data-level="13.1" data-path="comparisons-5.html"><a href="comparisons-5.html#genomes-1"><i class="fa fa-check"></i><b>13.1</b> Genomes</a></li>
<li class="chapter" data-level="13.2" data-path="comparisons-5.html"><a href="comparisons-5.html#libraries-1"><i class="fa fa-check"></i><b>13.2</b> Libraries</a></li>
<li class="chapter" data-level="13.3" data-path="comparisons-5.html"><a href="comparisons-5.html#mutations-3"><i class="fa fa-check"></i><b>13.3</b> Mutations</a></li>
<li class="chapter" data-level="13.4" data-path="comparisons-5.html"><a href="comparisons-5.html#frequencies-2"><i class="fa fa-check"></i><b>13.4</b> Frequencies</a></li>
<li class="chapter" data-level="13.5" data-path="comparisons-5.html"><a href="comparisons-5.html#types"><i class="fa fa-check"></i><b>13.5</b> Types</a></li>
<li class="chapter" data-level="13.6" data-path="comparisons-5.html"><a href="comparisons-5.html#spectra-1"><i class="fa fa-check"></i><b>13.6</b> Spectra</a></li>
<li class="chapter" data-level="13.7" data-path="comparisons-5.html"><a href="comparisons-5.html#rates-1"><i class="fa fa-check"></i><b>13.7</b> Rates</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="comparisons-5.html"><a href="comparisons-5.html#cummulated-distributions"><i class="fa fa-check"></i><b>13.7.1</b> Cummulated distributions</a></li>
<li class="chapter" data-level="13.7.2" data-path="comparisons-5.html"><a href="comparisons-5.html#lognormal-distributions"><i class="fa fa-check"></i><b>13.7.2</b> Lognormal distributions</a></li>
<li class="chapter" data-level="13.7.3" data-path="comparisons-5.html"><a href="comparisons-5.html#exponential-distrbutions"><i class="fa fa-check"></i><b>13.7.3</b> Exponential distrbutions</a></li>
<li class="chapter" data-level="13.7.4" data-path="comparisons-5.html"><a href="comparisons-5.html#power-law-distribution"><i class="fa fa-check"></i><b>13.7.4</b> Power law distribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="manuscript-figures.html"><a href="manuscript-figures.html"><i class="fa fa-check"></i><b>14</b> Manuscript figures</a>
<ul>
<li class="chapter" data-level="14.0.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-1"><i class="fa fa-check"></i><b>14.0.1</b> Figure 1</a></li>
<li class="chapter" data-level="14.0.2" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-2"><i class="fa fa-check"></i><b>14.0.2</b> Figure 2</a></li>
<li class="chapter" data-level="14.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-3"><i class="fa fa-check"></i><b>14.1</b> Figure 3</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html"><i class="fa fa-check"></i><b>15</b> Manuscript figures EVS</a>
<ul>
<li class="chapter" data-level="15.1" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-2-evs"><i class="fa fa-check"></i><b>15.1</b> Figure 2 EVS</a></li>
<li class="chapter" data-level="15.2" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-3-evs"><i class="fa fa-check"></i><b>15.2</b> Figure 3 EVS</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html"><i class="fa fa-check"></i><b>A</b> Appendix 1: Meetings</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sylvains-kick-off"><i class="fa fa-check"></i><b>A.1</b> 11/02/2021 Sylvain’s Kick-off</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#funding"><i class="fa fa-check"></i><b>A.1.1</b> Funding</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#site-1"><i class="fa fa-check"></i><b>A.1.2</b> Site</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#species-1"><i class="fa fa-check"></i><b>A.1.3</b> Species</a></li>
<li class="chapter" data-level="A.1.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#field-work"><i class="fa fa-check"></i><b>A.1.4</b> Field work</a></li>
<li class="chapter" data-level="A.1.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#genomics"><i class="fa fa-check"></i><b>A.1.5</b> Genomics</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#william-marrande"><i class="fa fa-check"></i><b>A.2</b> 27/05/2021 William Marrande</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sampling-1"><i class="fa fa-check"></i><b>A.2.1</b> Sampling</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#assembly"><i class="fa fa-check"></i><b>A.2.2</b> Assembly</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutation-detection"><i class="fa fa-check"></i><b>A.3</b> 06/07/2021 Mutation detection</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#virtual-experiment"><i class="fa fa-check"></i><b>A.3.1</b> Virtual experiment</a></li>
<li class="chapter" data-level="A.3.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#swiss-data"><i class="fa fa-check"></i><b>A.3.2</b> Swiss data</a></li>
<li class="chapter" data-level="A.3.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#angela-1"><i class="fa fa-check"></i><b>A.3.3</b> Angela</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mid-project-meeting"><i class="fa fa-check"></i><b>A.4</b> 21/01/2022 Mid-project meeting</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#notes"><i class="fa fa-check"></i><b>A.4.1</b> Notes</a></li>
<li class="chapter" data-level="A.4.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#follow-up"><i class="fa fa-check"></i><b>A.4.2</b> Follow-up</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#carrotage"><i class="fa fa-check"></i><b>A.5</b> 10/02 Carrotage</a></li>
<li class="chapter" data-level="A.6" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sixtos-results"><i class="fa fa-check"></i><b>A.6</b> 05/05 Sixto’s results</a></li>
<li class="chapter" data-level="A.7" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutations-days"><i class="fa fa-check"></i><b>A.7</b> 20-21/03 Mutations days</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html"><i class="fa fa-check"></i><b>B</b> Appendix 2: Tree 3D</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation"><i class="fa fa-check"></i><b>B.1</b> Tree Segmentation</a></li>
<li class="chapter" data-level="B.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-architecture"><i class="fa fa-check"></i><b>B.2</b> Tree architecture</a></li>
<li class="chapter" data-level="B.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-light"><i class="fa fa-check"></i><b>B.3</b> Tree light</a></li>
<li class="chapter" data-level="B.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-1"><i class="fa fa-check"></i><b>B.4</b> Tree Segmentation</a>
<ul>
<li class="chapter" data-level="B.4.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeseg"><i class="fa fa-check"></i><b>B.4.1</b> <code>treeseg</code></a></li>
<li class="chapter" data-level="B.4.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#computree"><i class="fa fa-check"></i><b>B.4.2</b> <code>Computree</code></a></li>
<li class="chapter" data-level="B.4.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-with-lidr"><i class="fa fa-check"></i><b>B.4.3</b> Tree Segmentation with <code>lidr</code></a></li>
<li class="chapter" data-level="B.4.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#forest-metrics"><i class="fa fa-check"></i><b>B.4.4</b> Forest Metrics</a></li>
<li class="chapter" data-level="B.4.5" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#sp4e"><i class="fa fa-check"></i><b>B.4.5</b> sp4e</a></li>
<li class="chapter" data-level="B.4.6" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeqsm"><i class="fa fa-check"></i><b>B.4.6</b> TreeQSM</a></li>
<li class="chapter" data-level="B.4.7" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#google-search"><i class="fa fa-check"></i><b>B.4.7</b> Google Search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html"><i class="fa fa-check"></i><b>C</b> Appendix 3: Literature</a>
<ul>
<li class="chapter" data-level="C.0.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#bergeron2023"><i class="fa fa-check"></i><b>C.0.1</b> Bergeron2023</a></li>
<li class="chapter" data-level="C.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#genome-assembly-1"><i class="fa fa-check"></i><b>C.1</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#johnson2020"><i class="fa fa-check"></i><b>C.1.1</b> <span class="citation">Johnson <em>et al.</em> (2020)</span></a></li>
<li class="chapter" data-level="C.1.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#plomion2018"><i class="fa fa-check"></i><b>C.1.2</b> <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.1.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#msc"><i class="fa fa-check"></i><b>C.1.3</b> Msc</a></li>
<li class="chapter" data-level="C.1.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#si"><i class="fa fa-check"></i><b>C.1.4</b> SI</a></li>
<li class="chapter" data-level="C.1.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#httpswww.oakgenome.frpage_id587"><i class="fa fa-check"></i><b>C.1.5</b> https://www.oakgenome.fr/?page_id=587</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#mutation-detection-1"><i class="fa fa-check"></i><b>C.2</b> Mutation detection</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#orr2019"><i class="fa fa-check"></i><b>C.2.1</b> <span class="citation">Orr <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#ramu2013"><i class="fa fa-check"></i><b>C.2.2</b> <span class="citation">Ramu <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#hanlon2019"><i class="fa fa-check"></i><b>C.2.3</b> <span class="citation">Hanlon <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#cibulskis2013"><i class="fa fa-check"></i><b>C.2.4</b> <span class="citation">Cibulskis <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#nicholson2018"><i class="fa fa-check"></i><b>C.2.5</b> <span class="citation">Nicholson <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.2.6" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#alioto2015"><i class="fa fa-check"></i><b>C.2.6</b> <span class="citation">Alioto <em>et al.</em> (2015)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#unclassified"><i class="fa fa-check"></i><b>C.3</b> Unclassified</a>
<ul>
<li class="chapter" data-level="C.3.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#michael2020"><i class="fa fa-check"></i><b>C.3.1</b> <span class="citation">Michael &amp; VanBuren (2020)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#to-read"><i class="fa fa-check"></i><b>C.4</b> To read</a>
<ul>
<li class="chapter" data-level="C.4.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#belser2018"><i class="fa fa-check"></i><b>C.4.1</b> <span class="citation">Belser <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.4.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#edwards1990"><i class="fa fa-check"></i><b>C.4.2</b> <span class="citation">Edwards <em>et al.</em> (1990)</span></a></li>
<li class="chapter" data-level="C.4.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#leebens-mack2019"><i class="fa fa-check"></i><b>C.4.3</b> <span class="citation">Leebens-Mack <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.4.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#links"><i class="fa fa-check"></i><b>C.4.4</b> links</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TreeMutation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutations-sixto" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Mutations Sixto<a href="mutations-sixto.html#mutations-sixto" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter describes the analyses of Sixto mutations currently done in the the <a href="https://github.com/sylvainschmitt/detectMutations/tree/sixto"><code>sixto</code> branch of the <code>detectMutations</code> repository</a>.</p>
<div id="quality-check-1" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Quality check<a href="mutations-sixto.html#quality-check-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The results of multiQC are available for the haplotypes HS1 <a href="data/mutations/sixto/qc/multiqc_report.html">here</a>.
They globally all gave a green light for subsequent analyses.</p>
</div>
<div id="genome-1" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Genome<a href="mutations-sixto.html#genome-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure"><span style="display:block;" id="fig:SixtoGenomFig"></span>
<img src="treemutation_files/figure-html/SixtoGenomFig-1.png" alt="Quality assesment of Sixto's genome. A. Anchored scaffolds size-distribution for anchored haplotypes HS1 in Sixto's genome. B. BUSCO results for haplotypes HS1 and HS2." width="768" />
<p class="caption">
Figure 11.1: Quality assesment of Sixto’s genome. A. Anchored scaffolds size-distribution for anchored haplotypes HS1 in Sixto’s genome. B. BUSCO results for haplotypes HS1 and HS2.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:SixtoCovs"></span>
<img src="treemutation_files/figure-html/SixtoCovs-1.png" alt="Coverage distribution. Distribution of the number of locations in the reference genome with a given depth of coverage." width="768" />
<p class="caption">
Figure 11.2: Coverage distribution. Distribution of the number of locations in the reference genome with a given depth of coverage.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:circosSixtog"></span>
<img src="save/SixtoCircosCovGC.png" alt="Genome coverage and GC-content on a 10-kb windows for the HS1 haplotype (all anchored scaffolds)." width="500" />
<p class="caption">
Figure 11.3: Genome coverage and GC-content on a 10-kb windows for the HS1 haplotype (all anchored scaffolds).
</p>
</div>
<div id="annotation-1" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Annotation<a href="mutations-sixto.html#annotation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="save/SixtoCircosAnnot.png" width="500" /></p>
</div>
<div id="mutations-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Mutations<a href="mutations-sixto.html#mutations-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="save/SixtoCircosmutations.png" width="500" /></p>
</div>
<div id="comp-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> Comp<a href="mutations-sixto.html#comp-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="treemutation_files/figure-html/unnamed-chunk-47-1.png" width="768" /></p>
<pre><code>## 
## Call:
## glm(formula = mutations ~ percentage_gene, family = &quot;poisson&quot;, 
##     data = all)
## 
## Coefficients:
##                 Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)     -1.03310    0.03008 -34.341   &lt;2e-16 ***
## percentage_gene -0.31327    0.17572  -1.783   0.0746 .  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 8665.7  on 8236  degrees of freedom
## Residual deviance: 8662.5  on 8235  degrees of freedom
## AIC: 13107
## 
## Number of Fisher Scoring iterations: 6</code></pre>
<p><img src="treemutation_files/figure-html/unnamed-chunk-48-1.png" width="768" /></p>
<pre><code>## 
## Call:
## glm(formula = mutations ~ percentage_te, family = &quot;poisson&quot;, 
##     data = all)
## 
## Coefficients:
##               Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   -1.50617    0.06899 -21.832  &lt; 2e-16 ***
## percentage_te  0.62403    0.10430   5.983 2.19e-09 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 10017.9  on 9724  degrees of freedom
## Residual deviance:  9982.1  on 9723  degrees of freedom
## AIC: 15066
## 
## Number of Fisher Scoring iterations: 6</code></pre>
</div>
</div>
<div id="rubrynolide-enzymes" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Rubrynolide enzymes<a href="mutations-sixto.html#rubrynolide-enzymes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="mutations-sixto.html#cb6-1" tabindex="-1"></a><span class="ex">On</span> https://img.jgi.doe.gov/cgi-bin/m/main.cgi</span>
<span id="cb6-2"><a href="mutations-sixto.html#cb6-2" tabindex="-1"></a><span class="op">&gt;</span> Find <span class="ex">gene</span></span>
<span id="cb6-3"><a href="mutations-sixto.html#cb6-3" tabindex="-1"></a><span class="op">&gt;</span> dplyr::select <span class="ex">All</span></span>
<span id="cb6-4"><a href="mutations-sixto.html#cb6-4" tabindex="-1"></a><span class="op">&gt;</span> Add <span class="ex">dplyr::selected</span> to Gene Cart</span>
<span id="cb6-5"><a href="mutations-sixto.html#cb6-5" tabindex="-1"></a><span class="op">&gt;</span> Uplad <span class="kw">&amp;</span> <span class="ex">Export</span></span>
<span id="cb6-6"><a href="mutations-sixto.html#cb6-6" tabindex="-1"></a><span class="op">&gt;</span>  FASTA <span class="ex">Nucleic</span> Acid format </span>
<span id="cb6-7"><a href="mutations-sixto.html#cb6-7" tabindex="-1"></a><span class="op">&gt;</span> Show <span class="er">in</span> <span class="bu">export</span> <span class="va">format</span></span>
<span id="cb6-8"><a href="mutations-sixto.html#cb6-8" tabindex="-1"></a><span class="op">&gt;</span> Copy <span class="er">in</span> <span class="ex">rubrynolide.fa</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="mutations-sixto.html#cb7-1" tabindex="-1"></a><span class="bu">cd</span> data/mutations/sixto/genome/</span>
<span id="cb7-2"><a href="mutations-sixto.html#cb7-2" tabindex="-1"></a><span class="ex">~/Tools/blatSrc/bin/blat</span> HS1_Sru_omap1_hap1_HYBRID_SCAFFOLD.fa rubrynolide.fa rubrynolide_on_sixto_anchored.psl</span></code></pre></div>
<p><strong>No match!</strong></p>
</div>
<div id="heterozygosity-1" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Heterozygosity<a href="mutations-sixto.html#heterozygosity-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We used k-mer analyses (21-mers) using <code>jellyfish</code> and the <code>GenomeScope</code> to estimate genetic diversity <span class="math inline">\(\pi=0.901%\)</span> (Fig. <a href="mutations-sixto.html#fig:genomescopeSixto">11.4</a>).
We detected and filtered SNPs as follow:</p>
<ul>
<li>Raw: raw result from <code>GATK HaplotypeCaller + GenomicsDBImport + GenotypeGVCFs</code></li>
<li>Biallelic: biallelic sites with <code>bcftools</code></li>
<li>SNP SNPs with <code>GATK</code></li>
<li>Filtered: SNPs with QUAL &lt; 30, QD &lt; 2, FS &gt; 60, SOR &gt; 3 using <code>GATK</code></li>
<li>Non-missing: SNPs in all genotypes and individuals with <code>plink</code></li>
<li>Shared=8M : shared SNPs by at least 32 out of 33 individuals with <code>bcftools</code></li>
</ul>
<div class="figure"><span style="display:block;" id="fig:genomescopeSixto"></span>
<img src="data/mutations/sixto/hz/genomescope.png" alt="GenomeScope Profile. Full results here: http://genomescope.org/analysis.php?code=Esy8w8C9yQa4rmw87o6y." width="1000" />
<p class="caption">
Figure 11.4: GenomeScope Profile. Full results here: <a href="http://genomescope.org/analysis.php?code=Esy8w8C9yQa4rmw87o6y" class="uri">http://genomescope.org/analysis.php?code=Esy8w8C9yQa4rmw87o6y</a>.
</p>
</div>
</div>
<div id="mutations-filters-1" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Mutations filters<a href="mutations-sixto.html#mutations-filters-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We removed all candidate mutations present in the heterozygous sites and in the raw candidate mutations from the cambium.
We kept only mutations with no copy of the mutated allele in the normal sample (NAC, <code>normal_altCount == 0</code>).
We tested independently the effect of 4 filters and look their effect on (1) the percentage kept, (2) the distribution of allelic frequencies, and (3) the overlap between filters:</p>
<ol style="list-style-type: decimal">
<li><strong>MAC</strong>: a minimum of 5 copies of the mutated allele in the mutated sample (<code>mutation_altCount &gt;= 5</code>)</li>
<li><strong>DP</strong>: a read depth for the two sample between the 5th quantile and the 95th quantile of the coverage of the corresponding library (<code>normal_DP &lt;= high_cov, normal_DP &gt;= low_cov, mutation_DP &lt;= high_cov, mutation_DP &gt;= low_cov</code>)</li>
<li><strong>BIO</strong>: the mutation is present in at least two biological replicates (2 leaves from the crown)</li>
<li><strong>EVS</strong>: <code>Strelka 2</code> automatic filtering based on the empiric variant score (<code>Filter == "PASS"</code>)</li>
</ol>
<p><strong>MAC</strong> and <strong>EVS</strong> are the most stringent filters (Fig. <a href="mutations-angela.html#fig:pctmutfilter">10.7</a>,
and mostly <strong>MAC</strong> filter changes the allele frequencies distribution (Fig. <a href="mutations-angela.html#fig:afmutfilter">10.8</a>.
Most filters share mutations, except <strong>EVS</strong> and <strong>BIO</strong> that rejected individually a lot mutations shared by the two others (Fig. <a href="mutations-angela.html#fig:vennmutfilter">10.9</a>.
And all filters except <strong>MAC</strong> are not sensitive to the library coverage (Fig. <a href="mutations-angela.html#fig:nmutcovfilter">10.10</a>.
We will thus use two filters for next steps:</p>
<ol style="list-style-type: decimal">
<li><strong>base</strong>:
<ul>
<li><strong>NAC</strong>: no copy of the mutated allele in the normal sample (<code>normal_altCountT1 == 0</code>)</li>
<li><strong>MAC</strong>: a minimum of 5 copies of the mutated allele in the mutated sample (<code>mutation_altCount &gt;= 5</code>)</li>
<li><strong>DP</strong>: a read depth for the two sample between the 10th quantile and the 90th quantile of the coverage of the mutated library (<code>normal_DP &lt;= high_cov, normal_DP &gt;= low_cov, mutation_DP &lt;= high_cov, mutation_DP &gt;= low_cov</code>)</li>
<li><strong>BIO</strong>: the mutation is present in at least two biological replicates (2 leaves from the crown)</li>
</ul></li>
<li><strong>evs</strong>:
<ul>
<li><strong>base</strong></li>
<li><strong>EVS</strong>: <code>Strelka 2</code> automatic filtering based on the empiric variant score (<code>Filter == "PASS"</code>)</li>
</ul></li>
</ol>
<div class="figure"><span style="display:block;" id="fig:pctmutfilterSixto"></span>
<img src="treemutation_files/figure-html/pctmutfilterSixto-1.png" alt="Percentage of kept mutations per filter." width="768" />
<p class="caption">
Figure 11.5: Percentage of kept mutations per filter.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:afmutfilterSixto"></span>
<img src="data/mutations/sixto/filters/filters_mutations_af.png" alt="Alleles frequencies of filtered mutations." width="600" />
<p class="caption">
Figure 11.6: Alleles frequencies of filtered mutations.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:vennmutfilterSixto"></span>
<img src="data/mutations/sixto/filters/filters_mutations_venn.png" alt="Filtered mutations sharing across filters." width="1050" />
<p class="caption">
Figure 11.7: Filtered mutations sharing across filters.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:nmutcovfilterSixto"></span>
<img src="treemutation_files/figure-html/nmutcovfilterSixto-1.png" alt="Link between the number of detected mutations and the coverage across samples depending on the filter used." width="768" />
<p class="caption">
Figure 11.8: Link between the number of detected mutations and the coverage across samples depending on the filter used.
</p>
</div>
</div>
<div id="leaf-mutations-1" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Leaf mutations<a href="mutations-sixto.html#leaf-mutations-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>14 to 59 mutations passed evs filtering across samples (Fig. <a href="#fig:nmutTrunkFiltTabSixto"><strong>??</strong></a>).
Most of mutations are low frequency (Fig. <a href="mutations-sixto.html#fig:afLeafFiltFigSixto">11.10</a>).
Most of mutations are not shared by biological replicates (Fig. <a href="mutations-sixto.html#fig:vennLeafFiltevsFigSixto">11.11</a> and Fig. <a href="mutations-sixto.html#fig:vennLeafFiltbaseFigSixto">11.12</a>).</p>
<div class="figure"><span style="display:block;" id="fig:nmutLeafFiltTabSixto"></span>
<img src="treemutation_files/figure-html/nmutLeafFiltTabSixto-1.png" alt="Number of filtered mutations per leaf." width="768" />
<p class="caption">
Figure 11.9: Number of filtered mutations per leaf.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:afLeafFiltFigSixto"></span>
<img src="treemutation_files/figure-html/afLeafFiltFigSixto-1.png" alt="Alleles-frequencies of filtered mutations in leaf samples." width="768" />
<p class="caption">
Figure 11.10: Alleles-frequencies of filtered mutations in leaf samples.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:vennLeafFiltevsFigSixto"></span>
<img src="treemutation_files/figure-html/vennLeafFiltevsFigSixto-1.png" alt="Filtered mutations sharing across leaf samples within tips for evs filtering." width="768" />
<p class="caption">
Figure 11.11: Filtered mutations sharing across leaf samples within tips for evs filtering.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:vennLeafFiltbaseFigSixto"></span>
<img src="treemutation_files/figure-html/vennLeafFiltbaseFigSixto-1.png" alt="Filtered mutations sharing across leaf samples within tips for base filtering." width="768" />
<p class="caption">
Figure 11.12: Filtered mutations sharing across leaf samples within tips for base filtering.
</p>
</div>
</div>
<div id="mutations-architecture-1" class="section level2 hasAnchor" number="11.7">
<h2><span class="header-section-number">11.7</span> Mutations architecture<a href="mutations-sixto.html#mutations-architecture-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="treemutation_files/figure-html/archiMutationsEVSFigSixto-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/archiMutationsEVSFigAFSixto-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/archiMutationsBaseFigSixto-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/archiMutationsEVSFig2AFSixto-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/archiMutationsEVSFig2AF2Sixto-1.png" width="768" /></p>
</div>
<div id="phylogeny-1" class="section level2 hasAnchor" number="11.8">
<h2><span class="header-section-number">11.8</span> Phylogeny<a href="mutations-sixto.html#phylogeny-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="treemutation_files/figure-html/angleatipsphyloSixto-1.png" width="1152" /></p>
</div>
<div id="light-annotation-1" class="section level2 hasAnchor" number="11.9">
<h2><span class="header-section-number">11.9</span> Light &amp; annotation<a href="mutations-sixto.html#light-annotation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-57"></span>
<img src="treemutation_files/figure-html/unnamed-chunk-57-1.png" alt="Number of mutations per filter and light condition across branches." width="768" />
<p class="caption">
Figure 11.13: Number of mutations per filter and light condition across branches.
</p>
</div>
<p><img src="treemutation_files/figure-html/unnamed-chunk-58-1.png" width="768" /></p>
<p><img src="treemutation_files/figure-html/unnamed-chunk-59-1.png" width="1152" /></p>
</div>
<div id="rate-1" class="section level2 hasAnchor" number="11.10">
<h2><span class="header-section-number">11.10</span> Rate<a href="mutations-sixto.html#rate-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="treemutation_files/figure-html/unnamed-chunk-60-1.png" width="768" /></p>
</div>
<div id="mutations-for-fruits-1" class="section level2 hasAnchor" number="11.11">
<h2><span class="header-section-number">11.11</span> Mutations for Fruits<a href="mutations-sixto.html#mutations-for-fruits-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Selection of mutations for transmission in fruits.</p>
<table>
<caption>
<span id="tab:fruitMutationsTabSixto">Table 11.1: </span>Number of mutations per branch.
</caption>
<thead>
<tr>
<th style="text-align:right;">
replicate
</th>
<th style="text-align:right;">
N
</th>
<th style="text-align:right;">
minAF
</th>
<th style="text-align:right;">
meanAF
</th>
<th style="text-align:right;">
maxAF
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.37
</td>
<td style="text-align:right;">
0.37
</td>
<td style="text-align:right;">
0.37
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
0.11
</td>
<td style="text-align:right;">
0.11
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.09
</td>
<td style="text-align:right;">
0.11
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.09
</td>
<td style="text-align:right;">
0.10
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.09
</td>
<td style="text-align:right;">
0.13
</td>
</tr>
</tbody>
</table>
<p><img src="treemutation_files/figure-html/fruitMutationsFigSixto-1.png" width="768" /></p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-c059fb26a312b99af895" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c059fb26a312b99af895">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["Super-Scaffold_3","Super-Scaffold_6","Super-Scaffold_6","Super-Scaffold_2","Super-Scaffold_14","Super-Scaffold_46","Super-Scaffold_14","Super-Scaffold_46","Super-Scaffold_14","Super-Scaffold_6","Super-Scaffold_13","Super-Scaffold_35","Super-Scaffold_9","Super-Scaffold_43","Super-Scaffold_5","Super-Scaffold_14","Super-Scaffold_13","Super-Scaffold_26","Super-Scaffold_5","Super-Scaffold_9","Super-Scaffold_31","Super-Scaffold_2","Super-Scaffold_3","Super-Scaffold_2","Super-Scaffold_2","Super-Scaffold_54","Super-Scaffold_16","Super-Scaffold_13","Super-Scaffold_7","Super-Scaffold_43","Super-Scaffold_7","Super-Scaffold_46","Super-Scaffold_12","Super-Scaffold_2","Super-Scaffold_26","Super-Scaffold_7"],[23125604,22702523,22702521,123353752,31198009,26523027,60409142,10603162,10788485,6578384,1079125,19262955,39012920,55230935,15434392,25196252,26308975,239886,15434420,19776398,2768761,120249675,3310977,107394803,23747078,12089144,40356854,26308937,2312265,14926191,5943497,33206267,34955729,33471162,26768587,5783490],["B1, C1, C3, E2, E4, F2","B1, C1, C3, E4, F2","B1, C1, C3, E4, F2","B1, C1, E2, F2","B1, C1, E2, E4, F2","B1, C1, E2, F2","B1, C1, E2","B1, C1, C3","C1, C3, E2, E4","C3, E2, E4","B1, C1, F2","C1, C3, E4, F2","B1, C1, E2, F2","E2, F2","C3, E4, F2","B1, E4, F2","C3, F2","B1, E2, E4","B1, C3, E4","C1, C3, F2","C1, E4, F2","B1, C1, C3","C1, C3, E2","B1, E4, F2","C1","C1, E2","B1","C3, F2","B1, C3, F2","C1","C1","C1","C1","C1","C1","B1"],[18,6,6,6,5,5,5,5,5,4,4,4,4,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3],[0.37,0.11,0.11,0.1,0.11,0.11,0.07000000000000001,0.07000000000000001,0.07000000000000001,0.1,0.09,0.08,0.08,0.13,0.11,0.1,0.1,0.1,0.09,0.09,0.09,0.09,0.08,0.08,0.08,0.08,0.08,0.08,0.08,0.08,0.08,0.07000000000000001,0.07000000000000001,0.07000000000000001,0.07000000000000001,0.07000000000000001],[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Super scaffold<\/th>\n      <th>position<\/th>\n      <th>branch(es)<\/th>\n      <th>replicate<\/th>\n      <th>allele frequency<\/th>\n      <th>rank<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="genes-1" class="section level2 hasAnchor" number="11.12">
<h2><span class="header-section-number">11.12</span> Genes<a href="mutations-sixto.html#genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="mutations-sixto.html#cb8-1" tabindex="-1"></a><span class="bu">cd</span> data/mutations/sixto/annotation</span>
<span id="cb8-2"><a href="mutations-sixto.html#cb8-2" tabindex="-1"></a><span class="ex">~/Tools/gffread/gffread</span> trsc.fa.transdecoder.gff3 <span class="at">-T</span> <span class="at">-o</span> trsc.fa.transdecoder.gtf</span>
<span id="cb8-3"><a href="mutations-sixto.html#cb8-3" tabindex="-1"></a><span class="fu">perl</span> ~/Tools/SNPdat_package_v1.0.5/SNPdat_v1.0.5.pl <span class="dt">\</span></span>
<span id="cb8-4"><a href="mutations-sixto.html#cb8-4" tabindex="-1"></a>  <span class="at">-i</span> SNPsOnTrsc.tsv <span class="dt">\</span></span>
<span id="cb8-5"><a href="mutations-sixto.html#cb8-5" tabindex="-1"></a>  <span class="at">-g</span> trsc.fa.transdecoder.gtf <span class="dt">\</span></span>
<span id="cb8-6"><a href="mutations-sixto.html#cb8-6" tabindex="-1"></a>  <span class="at">-f</span> trsc.fa <span class="dt">\</span></span>
<span id="cb8-7"><a href="mutations-sixto.html#cb8-7" tabindex="-1"></a>  <span class="at">-s</span> synonymy.summary <span class="dt">\</span></span>
<span id="cb8-8"><a href="mutations-sixto.html#cb8-8" tabindex="-1"></a>  <span class="at">-o</span> synonymy.output</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-64"></span>
<img src="treemutation_files/figure-html/unnamed-chunk-64-1.png" alt="Mutations types." width="768" />
<p class="caption">
Figure 11.14: Mutations types.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-65"></span>
<img src="treemutation_files/figure-html/unnamed-chunk-65-1.png" alt="Mutation allelic fraction per type." width="768" />
<p class="caption">
Figure 11.15: Mutation allelic fraction per type.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-66"></span>
<img src="treemutation_files/figure-html/unnamed-chunk-66-1.png" alt="Synonymy with allelic fraction." width="768" />
<p class="caption">
Figure 11.16: Synonymy with allelic fraction.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-67"></span>
<img src="treemutation_files/figure-html/unnamed-chunk-67-1.png" alt="GO enrichment for Angela non-synonymous mutations." width="768" />
<p class="caption">
Figure 11.17: GO enrichment for Angela non-synonymous mutations.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mutations-angela.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="comparisons-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
