<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Mutations detection | TreeMutation</title>
  <meta name="description" content="Mutation in the tropical tree canopy." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Mutations detection | TreeMutation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Mutation in the tropical tree canopy." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Mutations detection | TreeMutation" />
  
  <meta name="twitter:description" content="Mutation in the tropical tree canopy." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2023-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-assembly.html"/>
<link rel="next" href="mutations-in-silico.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.0/leaflet.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.29/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TreeMutation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="species.html"><a href="species.html"><i class="fa fa-check"></i><b>1</b> Species</a>
<ul>
<li class="chapter" data-level="1.1" data-path="species.html"><a href="species.html#dicorynia-guianensis"><i class="fa fa-check"></i><b>1.1</b> <em>Dicorynia guianensis</em></a></li>
<li class="chapter" data-level="1.2" data-path="species.html"><a href="species.html#sextonia-rubra"><i class="fa fa-check"></i><b>1.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="1.3" data-path="species.html"><a href="species.html#eschweilera-coriacea"><i class="fa fa-check"></i><b>1.3</b> <em>Eschweilera coriacea</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="site.html"><a href="site.html"><i class="fa fa-check"></i><b>2</b> Site</a>
<ul>
<li class="chapter" data-level="2.1" data-path="site.html"><a href="site.html#paracou"><i class="fa fa-check"></i><b>2.1</b> Paracou</a></li>
<li class="chapter" data-level="2.2" data-path="site.html"><a href="site.html#régina"><i class="fa fa-check"></i><b>2.2</b> Régina</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="individuals.html"><a href="individuals.html"><i class="fa fa-check"></i><b>3</b> Individuals</a>
<ul>
<li class="chapter" data-level="3.1" data-path="individuals.html"><a href="individuals.html#angela---dicorynia-guyanensis"><i class="fa fa-check"></i><b>3.1</b> <strong>Angela</strong> - <em>Dicorynia guyanensis</em></a></li>
<li class="chapter" data-level="3.2" data-path="individuals.html"><a href="individuals.html#sextonia-rubra-1"><i class="fa fa-check"></i><b>3.2</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="3.3" data-path="individuals.html"><a href="individuals.html#eschweilera-sp."><i class="fa fa-check"></i><b>3.3</b> <em>Eschweilera sp.</em></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="individuals.html"><a href="individuals.html#mn5505---regina"><i class="fa fa-check"></i><b>3.3.1</b> MN5505 - Regina</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sampling.html"><a href="sampling.html"><i class="fa fa-check"></i><b>4</b> Sampling</a></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#sequencing"><i class="fa fa-check"></i><b>5.1</b> Sequencing</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#assembly-workflow"><i class="fa fa-check"></i><b>5.2</b> Assembly workflow</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-assembly.html"><a href="genome-assembly.html#v1"><i class="fa fa-check"></i><b>5.2.1</b> V1</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-assembly.html"><a href="genome-assembly.html#v2"><i class="fa fa-check"></i><b>5.2.2</b> V2</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#annotation-workflow"><i class="fa fa-check"></i><b>5.3</b> Annotation workflow</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#project-description"><i class="fa fa-check"></i><b>5.4</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mutations-detection.html"><a href="mutations-detection.html"><i class="fa fa-check"></i><b>6</b> Mutations detection</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mutations-detection.html"><a href="mutations-detection.html#in-silico-experiment"><i class="fa fa-check"></i><b>6.1</b> <em>In silico</em> experiment</a></li>
<li class="chapter" data-level="6.2" data-path="mutations-detection.html"><a href="mutations-detection.html#swiss-data---schmid-siegert2017"><i class="fa fa-check"></i><b>6.2</b> Swiss data - <span class="citation">Schmid-Siegert <em>et al.</em> (2017)</span></a></li>
<li class="chapter" data-level="6.3" data-path="mutations-detection.html"><a href="mutations-detection.html#bordeaux-data---plomion2018"><i class="fa fa-check"></i><b>6.3</b> Bordeaux data - <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="6.4" data-path="mutations-detection.html"><a href="mutations-detection.html#angela"><i class="fa fa-check"></i><b>6.4</b> Angela</a></li>
<li class="chapter" data-level="6.5" data-path="mutations-detection.html"><a href="mutations-detection.html#sextonia-rubra-2"><i class="fa fa-check"></i><b>6.5</b> <em>Sextonia rubra</em></a></li>
<li class="chapter" data-level="6.6" data-path="mutations-detection.html"><a href="mutations-detection.html#hetre"><i class="fa fa-check"></i><b>6.6</b> Hetre</a></li>
<li class="chapter" data-level="6.7" data-path="mutations-detection.html"><a href="mutations-detection.html#project-description-1"><i class="fa fa-check"></i><b>6.7</b> Project description</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mutations-in-silico.html"><a href="mutations-in-silico.html"><i class="fa fa-check"></i><b>7</b> Mutations <em>in silico</em></a></li>
<li class="chapter" data-level="8" data-path="mutations-swiss.html"><a href="mutations-swiss.html"><i class="fa fa-check"></i><b>8</b> Mutations Swiss</a>
<ul>
<li class="chapter" data-level="8.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#mutations-from-schmid-siegert-on-3p"><i class="fa fa-check"></i><b>8.1</b> Mutations from Schmid-Siegert on 3P</a></li>
<li class="chapter" data-level="8.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#strelka2"><i class="fa fa-check"></i><b>8.2</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert"><i class="fa fa-check"></i><b>8.2.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.2.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering"><i class="fa fa-check"></i><b>8.2.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="mutations-swiss.html"><a href="mutations-swiss.html#gatk"><i class="fa fa-check"></i><b>8.3</b> <code>GATK</code></a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="mutations-swiss.html"><a href="mutations-swiss.html#overlap-with-mutations-from-schmid-siegert-1"><i class="fa fa-check"></i><b>8.3.1</b> Overlap with mutations from Schmid-Siegert</a></li>
<li class="chapter" data-level="8.3.2" data-path="mutations-swiss.html"><a href="mutations-swiss.html#filtering-1"><i class="fa fa-check"></i><b>8.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="mutations-swiss.html"><a href="mutations-swiss.html#conclusion"><i class="fa fa-check"></i><b>8.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html"><i class="fa fa-check"></i><b>9</b> Mutations Bordeaux</a>
<ul>
<li class="chapter" data-level="9.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#original-mutations-on-3p"><i class="fa fa-check"></i><b>9.1</b> Original mutations on 3P</a></li>
<li class="chapter" data-level="9.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#coverage"><i class="fa fa-check"></i><b>9.2</b> Coverage</a></li>
<li class="chapter" data-level="9.3" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#strelka2-1"><i class="fa fa-check"></i><b>9.3</b> <code>Strelka2</code></a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations"><i class="fa fa-check"></i><b>9.3.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.3.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-2"><i class="fa fa-check"></i><b>9.3.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#mutect2"><i class="fa fa-check"></i><b>9.4</b> <code>Mutect2</code></a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#overlap-with-original-mutations-1"><i class="fa fa-check"></i><b>9.4.1</b> Overlap with original mutations</a></li>
<li class="chapter" data-level="9.4.2" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#filtering-3"><i class="fa fa-check"></i><b>9.4.2</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#conclusion-1"><i class="fa fa-check"></i><b>9.5</b> Conclusion</a></li>
<li class="chapter" data-level="9.6" data-path="mutations-bordeaux.html"><a href="mutations-bordeaux.html#swiss-bordeaux"><i class="fa fa-check"></i><b>9.6</b> Swiss &amp; Bordeaux</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mutations-angela.html"><a href="mutations-angela.html"><i class="fa fa-check"></i><b>10</b> Mutations Angela</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mutations-angela.html"><a href="mutations-angela.html#quality-check"><i class="fa fa-check"></i><b>10.1</b> Quality check</a></li>
<li class="chapter" data-level="10.2" data-path="mutations-angela.html"><a href="mutations-angela.html#genome"><i class="fa fa-check"></i><b>10.2</b> Genome</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mutations-angela.html"><a href="mutations-angela.html#annotation"><i class="fa fa-check"></i><b>10.2.1</b> Annotation</a></li>
<li class="chapter" data-level="10.2.2" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations"><i class="fa fa-check"></i><b>10.2.2</b> Mutations</a></li>
<li class="chapter" data-level="10.2.3" data-path="mutations-angela.html"><a href="mutations-angela.html#comp"><i class="fa fa-check"></i><b>10.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mutations-angela.html"><a href="mutations-angela.html#heterozygosity"><i class="fa fa-check"></i><b>10.3</b> Heterozygosity</a></li>
<li class="chapter" data-level="10.4" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-filters"><i class="fa fa-check"></i><b>10.4</b> Mutations filters</a></li>
<li class="chapter" data-level="10.5" data-path="mutations-angela.html"><a href="mutations-angela.html#leaf-mutations"><i class="fa fa-check"></i><b>10.5</b> Leaf mutations</a></li>
<li class="chapter" data-level="10.6" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-architecture"><i class="fa fa-check"></i><b>10.6</b> Mutations architecture</a></li>
<li class="chapter" data-level="10.7" data-path="mutations-angela.html"><a href="mutations-angela.html#phylogeny"><i class="fa fa-check"></i><b>10.7</b> Phylogeny</a></li>
<li class="chapter" data-level="10.8" data-path="mutations-angela.html"><a href="mutations-angela.html#light-annotation"><i class="fa fa-check"></i><b>10.8</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="10.9" data-path="mutations-angela.html"><a href="mutations-angela.html#rate"><i class="fa fa-check"></i><b>10.9</b> Rate</a></li>
<li class="chapter" data-level="10.10" data-path="mutations-angela.html"><a href="mutations-angela.html#mutations-for-fruits"><i class="fa fa-check"></i><b>10.10</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="10.11" data-path="mutations-angela.html"><a href="mutations-angela.html#genes"><i class="fa fa-check"></i><b>10.11</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="mutations-sixto.html"><a href="mutations-sixto.html"><i class="fa fa-check"></i><b>11</b> Mutations Sixto</a>
<ul>
<li class="chapter" data-level="11.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#quality-check-1"><i class="fa fa-check"></i><b>11.1</b> Quality check</a></li>
<li class="chapter" data-level="11.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genome-1"><i class="fa fa-check"></i><b>11.2</b> Genome</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="mutations-sixto.html"><a href="mutations-sixto.html#annotation-1"><i class="fa fa-check"></i><b>11.2.1</b> Annotation</a></li>
<li class="chapter" data-level="11.2.2" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-1"><i class="fa fa-check"></i><b>11.2.2</b> Mutations</a></li>
<li class="chapter" data-level="11.2.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#comp-1"><i class="fa fa-check"></i><b>11.2.3</b> Comp</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rubrynolide-enzymes"><i class="fa fa-check"></i><b>11.3</b> Rubrynolide enzymes</a></li>
<li class="chapter" data-level="11.4" data-path="mutations-sixto.html"><a href="mutations-sixto.html#heterozygosity-1"><i class="fa fa-check"></i><b>11.4</b> Heterozygosity</a></li>
<li class="chapter" data-level="11.5" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-filters-1"><i class="fa fa-check"></i><b>11.5</b> Mutations filters</a></li>
<li class="chapter" data-level="11.6" data-path="mutations-sixto.html"><a href="mutations-sixto.html#leaf-mutations-1"><i class="fa fa-check"></i><b>11.6</b> Leaf mutations</a></li>
<li class="chapter" data-level="11.7" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-architecture-1"><i class="fa fa-check"></i><b>11.7</b> Mutations architecture</a></li>
<li class="chapter" data-level="11.8" data-path="mutations-sixto.html"><a href="mutations-sixto.html#phylogeny-1"><i class="fa fa-check"></i><b>11.8</b> Phylogeny</a></li>
<li class="chapter" data-level="11.9" data-path="mutations-sixto.html"><a href="mutations-sixto.html#light-annotation-1"><i class="fa fa-check"></i><b>11.9</b> Light &amp; annotation</a></li>
<li class="chapter" data-level="11.10" data-path="mutations-sixto.html"><a href="mutations-sixto.html#rate-1"><i class="fa fa-check"></i><b>11.10</b> Rate</a></li>
<li class="chapter" data-level="11.11" data-path="mutations-sixto.html"><a href="mutations-sixto.html#mutations-for-fruits-1"><i class="fa fa-check"></i><b>11.11</b> Mutations for Fruits</a></li>
<li class="chapter" data-level="11.12" data-path="mutations-sixto.html"><a href="mutations-sixto.html#genes-1"><i class="fa fa-check"></i><b>11.12</b> Genes</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="comparisons-2.html"><a href="comparisons-2.html"><i class="fa fa-check"></i><b>12</b> Comparisons 2</a>
<ul>
<li class="chapter" data-level="12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#genomes"><i class="fa fa-check"></i><b>12.1</b> Genomes</a></li>
<li class="chapter" data-level="12.2" data-path="comparisons-2.html"><a href="comparisons-2.html#libraries"><i class="fa fa-check"></i><b>12.2</b> Libraries</a></li>
<li class="chapter" data-level="12.3" data-path="comparisons-2.html"><a href="comparisons-2.html#mutations-2"><i class="fa fa-check"></i><b>12.3</b> Mutations</a></li>
<li class="chapter" data-level="12.4" data-path="comparisons-2.html"><a href="comparisons-2.html#frequencies"><i class="fa fa-check"></i><b>12.4</b> Frequencies</a></li>
<li class="chapter" data-level="12.5" data-path="comparisons-2.html"><a href="comparisons-2.html#phylogeny-2"><i class="fa fa-check"></i><b>12.5</b> Phylogeny</a></li>
<li class="chapter" data-level="12.6" data-path="comparisons-2.html"><a href="comparisons-2.html#light"><i class="fa fa-check"></i><b>12.6</b> Light</a></li>
<li class="chapter" data-level="12.7" data-path="comparisons-2.html"><a href="comparisons-2.html#type"><i class="fa fa-check"></i><b>12.7</b> Type</a></li>
<li class="chapter" data-level="12.8" data-path="comparisons-2.html"><a href="comparisons-2.html#spectra"><i class="fa fa-check"></i><b>12.8</b> Spectra</a></li>
<li class="chapter" data-level="12.9" data-path="comparisons-2.html"><a href="comparisons-2.html#rates"><i class="fa fa-check"></i><b>12.9</b> Rates</a></li>
<li class="chapter" data-level="12.10" data-path="comparisons-2.html"><a href="comparisons-2.html#annotations"><i class="fa fa-check"></i><b>12.10</b> Annotations</a></li>
<li class="chapter" data-level="12.11" data-path="comparisons-2.html"><a href="comparisons-2.html#genes-2"><i class="fa fa-check"></i><b>12.11</b> Genes</a></li>
<li class="chapter" data-level="12.12" data-path="comparisons-2.html"><a href="comparisons-2.html#fruits"><i class="fa fa-check"></i><b>12.12</b> Fruits</a>
<ul>
<li class="chapter" data-level="12.12.1" data-path="comparisons-2.html"><a href="comparisons-2.html#from-ssrseq-olivier"><i class="fa fa-check"></i><b>12.12.1</b> From SSRseq Olivier</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="comparisons-5.html"><a href="comparisons-5.html"><i class="fa fa-check"></i><b>13</b> Comparisons 5</a>
<ul>
<li class="chapter" data-level="13.1" data-path="comparisons-5.html"><a href="comparisons-5.html#genomes-1"><i class="fa fa-check"></i><b>13.1</b> Genomes</a></li>
<li class="chapter" data-level="13.2" data-path="comparisons-5.html"><a href="comparisons-5.html#libraries-1"><i class="fa fa-check"></i><b>13.2</b> Libraries</a></li>
<li class="chapter" data-level="13.3" data-path="comparisons-5.html"><a href="comparisons-5.html#mutations-3"><i class="fa fa-check"></i><b>13.3</b> Mutations</a></li>
<li class="chapter" data-level="13.4" data-path="comparisons-5.html"><a href="comparisons-5.html#frequencies-2"><i class="fa fa-check"></i><b>13.4</b> Frequencies</a></li>
<li class="chapter" data-level="13.5" data-path="comparisons-5.html"><a href="comparisons-5.html#types"><i class="fa fa-check"></i><b>13.5</b> Types</a></li>
<li class="chapter" data-level="13.6" data-path="comparisons-5.html"><a href="comparisons-5.html#spectra-1"><i class="fa fa-check"></i><b>13.6</b> Spectra</a></li>
<li class="chapter" data-level="13.7" data-path="comparisons-5.html"><a href="comparisons-5.html#rates-1"><i class="fa fa-check"></i><b>13.7</b> Rates</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="comparisons-5.html"><a href="comparisons-5.html#cummulated-distributions"><i class="fa fa-check"></i><b>13.7.1</b> Cummulated distributions</a></li>
<li class="chapter" data-level="13.7.2" data-path="comparisons-5.html"><a href="comparisons-5.html#lognormal-distributions"><i class="fa fa-check"></i><b>13.7.2</b> Lognormal distributions</a></li>
<li class="chapter" data-level="13.7.3" data-path="comparisons-5.html"><a href="comparisons-5.html#exponential-distrbutions"><i class="fa fa-check"></i><b>13.7.3</b> Exponential distrbutions</a></li>
<li class="chapter" data-level="13.7.4" data-path="comparisons-5.html"><a href="comparisons-5.html#power-law-distribution"><i class="fa fa-check"></i><b>13.7.4</b> Power law distribution</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="manuscript-figures.html"><a href="manuscript-figures.html"><i class="fa fa-check"></i><b>14</b> Manuscript figures</a>
<ul>
<li class="chapter" data-level="14.0.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-1"><i class="fa fa-check"></i><b>14.0.1</b> Figure 1</a></li>
<li class="chapter" data-level="14.0.2" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-2"><i class="fa fa-check"></i><b>14.0.2</b> Figure 2</a></li>
<li class="chapter" data-level="14.1" data-path="manuscript-figures.html"><a href="manuscript-figures.html#figure-3"><i class="fa fa-check"></i><b>14.1</b> Figure 3</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html"><i class="fa fa-check"></i><b>15</b> Manuscript figures EVS</a>
<ul>
<li class="chapter" data-level="15.1" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-2-evs"><i class="fa fa-check"></i><b>15.1</b> Figure 2 EVS</a></li>
<li class="chapter" data-level="15.2" data-path="manuscript-figures-evs.html"><a href="manuscript-figures-evs.html#figure-3-evs"><i class="fa fa-check"></i><b>15.2</b> Figure 3 EVS</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html"><i class="fa fa-check"></i><b>A</b> Appendix 1: Meetings</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sylvains-kick-off"><i class="fa fa-check"></i><b>A.1</b> 11/02/2021 Sylvain’s Kick-off</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#funding"><i class="fa fa-check"></i><b>A.1.1</b> Funding</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#site-1"><i class="fa fa-check"></i><b>A.1.2</b> Site</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#species-1"><i class="fa fa-check"></i><b>A.1.3</b> Species</a></li>
<li class="chapter" data-level="A.1.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#field-work"><i class="fa fa-check"></i><b>A.1.4</b> Field work</a></li>
<li class="chapter" data-level="A.1.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#genomics"><i class="fa fa-check"></i><b>A.1.5</b> Genomics</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#william-marrande"><i class="fa fa-check"></i><b>A.2</b> 27/05/2021 William Marrande</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sampling-1"><i class="fa fa-check"></i><b>A.2.1</b> Sampling</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#assembly"><i class="fa fa-check"></i><b>A.2.2</b> Assembly</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutation-detection"><i class="fa fa-check"></i><b>A.3</b> 06/07/2021 Mutation detection</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#virtual-experiment"><i class="fa fa-check"></i><b>A.3.1</b> Virtual experiment</a></li>
<li class="chapter" data-level="A.3.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#swiss-data"><i class="fa fa-check"></i><b>A.3.2</b> Swiss data</a></li>
<li class="chapter" data-level="A.3.3" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#angela-1"><i class="fa fa-check"></i><b>A.3.3</b> Angela</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mid-project-meeting"><i class="fa fa-check"></i><b>A.4</b> 21/01/2022 Mid-project meeting</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#notes"><i class="fa fa-check"></i><b>A.4.1</b> Notes</a></li>
<li class="chapter" data-level="A.4.2" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#follow-up"><i class="fa fa-check"></i><b>A.4.2</b> Follow-up</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#carrotage"><i class="fa fa-check"></i><b>A.5</b> 10/02 Carrotage</a></li>
<li class="chapter" data-level="A.6" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#sixtos-results"><i class="fa fa-check"></i><b>A.6</b> 05/05 Sixto’s results</a></li>
<li class="chapter" data-level="A.7" data-path="appendix-1-meetings.html"><a href="appendix-1-meetings.html#mutations-days"><i class="fa fa-check"></i><b>A.7</b> 20-21/03 Mutations days</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html"><i class="fa fa-check"></i><b>B</b> Appendix 2: Tree 3D</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation"><i class="fa fa-check"></i><b>B.1</b> Tree Segmentation</a></li>
<li class="chapter" data-level="B.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-architecture"><i class="fa fa-check"></i><b>B.2</b> Tree architecture</a></li>
<li class="chapter" data-level="B.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-light"><i class="fa fa-check"></i><b>B.3</b> Tree light</a></li>
<li class="chapter" data-level="B.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-1"><i class="fa fa-check"></i><b>B.4</b> Tree Segmentation</a>
<ul>
<li class="chapter" data-level="B.4.1" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeseg"><i class="fa fa-check"></i><b>B.4.1</b> <code>treeseg</code></a></li>
<li class="chapter" data-level="B.4.2" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#computree"><i class="fa fa-check"></i><b>B.4.2</b> <code>Computree</code></a></li>
<li class="chapter" data-level="B.4.3" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#tree-segmentation-with-lidr"><i class="fa fa-check"></i><b>B.4.3</b> Tree Segmentation with <code>lidr</code></a></li>
<li class="chapter" data-level="B.4.4" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#forest-metrics"><i class="fa fa-check"></i><b>B.4.4</b> Forest Metrics</a></li>
<li class="chapter" data-level="B.4.5" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#sp4e"><i class="fa fa-check"></i><b>B.4.5</b> sp4e</a></li>
<li class="chapter" data-level="B.4.6" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#treeqsm"><i class="fa fa-check"></i><b>B.4.6</b> TreeQSM</a></li>
<li class="chapter" data-level="B.4.7" data-path="appendix-2-tree-3d.html"><a href="appendix-2-tree-3d.html#google-search"><i class="fa fa-check"></i><b>B.4.7</b> Google Search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html"><i class="fa fa-check"></i><b>C</b> Appendix 3: Literature</a>
<ul>
<li class="chapter" data-level="C.0.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#bergeron2023"><i class="fa fa-check"></i><b>C.0.1</b> Bergeron2023</a></li>
<li class="chapter" data-level="C.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#genome-assembly-1"><i class="fa fa-check"></i><b>C.1</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#johnson2020"><i class="fa fa-check"></i><b>C.1.1</b> <span class="citation">Johnson <em>et al.</em> (2020)</span></a></li>
<li class="chapter" data-level="C.1.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#plomion2018"><i class="fa fa-check"></i><b>C.1.2</b> <span class="citation">Plomion <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.1.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#msc"><i class="fa fa-check"></i><b>C.1.3</b> Msc</a></li>
<li class="chapter" data-level="C.1.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#si"><i class="fa fa-check"></i><b>C.1.4</b> SI</a></li>
<li class="chapter" data-level="C.1.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#httpswww.oakgenome.frpage_id587"><i class="fa fa-check"></i><b>C.1.5</b> https://www.oakgenome.fr/?page_id=587</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#mutation-detection-1"><i class="fa fa-check"></i><b>C.2</b> Mutation detection</a>
<ul>
<li class="chapter" data-level="C.2.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#orr2019"><i class="fa fa-check"></i><b>C.2.1</b> <span class="citation">Orr <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#ramu2013"><i class="fa fa-check"></i><b>C.2.2</b> <span class="citation">Ramu <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#hanlon2019"><i class="fa fa-check"></i><b>C.2.3</b> <span class="citation">Hanlon <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.2.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#cibulskis2013"><i class="fa fa-check"></i><b>C.2.4</b> <span class="citation">Cibulskis <em>et al.</em> (2013)</span></a></li>
<li class="chapter" data-level="C.2.5" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#nicholson2018"><i class="fa fa-check"></i><b>C.2.5</b> <span class="citation">Nicholson <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.2.6" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#alioto2015"><i class="fa fa-check"></i><b>C.2.6</b> <span class="citation">Alioto <em>et al.</em> (2015)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#unclassified"><i class="fa fa-check"></i><b>C.3</b> Unclassified</a>
<ul>
<li class="chapter" data-level="C.3.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#michael2020"><i class="fa fa-check"></i><b>C.3.1</b> <span class="citation">Michael &amp; VanBuren (2020)</span></a></li>
</ul></li>
<li class="chapter" data-level="C.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#to-read"><i class="fa fa-check"></i><b>C.4</b> To read</a>
<ul>
<li class="chapter" data-level="C.4.1" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#belser2018"><i class="fa fa-check"></i><b>C.4.1</b> <span class="citation">Belser <em>et al.</em> (2018)</span></a></li>
<li class="chapter" data-level="C.4.2" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#edwards1990"><i class="fa fa-check"></i><b>C.4.2</b> <span class="citation">Edwards <em>et al.</em> (1990)</span></a></li>
<li class="chapter" data-level="C.4.3" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#leebens-mack2019"><i class="fa fa-check"></i><b>C.4.3</b> <span class="citation">Leebens-Mack <em>et al.</em> (2019)</span></a></li>
<li class="chapter" data-level="C.4.4" data-path="appendix-3-literature.html"><a href="appendix-3-literature.html#links"><i class="fa fa-check"></i><b>C.4.4</b> links</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TreeMutation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mutations-detection" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Mutations detection<a href="mutations-detection.html#mutations-detection" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter briefly introduced the next chapters on the detection of mutations:</p>
<ul>
<li>Using an <em>in silico</em> virtual experiment</li>
<li>Using swiss data from <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span></li>
<li>Using french data from <span class="citation">Plomion <em>et al.</em> (<a href="#ref-Plomion2018">2018</a>)</span></li>
<li>Using Angela sequences</li>
<li>Using sequences from the <em>Sextonia rubra</em></li>
<li>Using sequences from the <a href="https://en.wikipedia.org/wiki/Faux_de_Verzy">“Faux de Verzy”</a></li>
</ul>
<p>The three first analyses resulted in a published manuscript: <a href="https://peercommunityjournal.org/articles/10.24072/pcjournal.187/" class="uri">https://peercommunityjournal.org/articles/10.24072/pcjournal.187/</a></p>
<div id="in-silico-experiment" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> <em>In silico</em> experiment<a href="mutations-detection.html#in-silico-experiment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <em>in silico</em> experiment uses the <a href="https://github.com/sylvainschmitt/generateMutations">generateMutations</a> and the <a href="https://github.com/sylvainschmitt/detectMutations">detectMutations</a> workflows to test the effect of coverage/sequencing depth and allelic frequency on the performance of 7 tools to detect mutations either generalist or specific to mutations.</p>
<div class="figure"><span style="display:block;" id="fig:generateMutations"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/generateMutations/main/dag/dag.minimal.svg" alt="Generate mutations workflow."  />
<p class="caption">
Figure 6.1: Generate mutations workflow.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:detectMutations"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/main/dag/dag.svg" alt="Detect mutations workflow."  />
<p class="caption">
Figure 6.2: Detect mutations workflow.
</p>
</div>
</div>
<div id="swiss-data---schmid-siegert2017" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Swiss data - <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span><a href="mutations-detection.html#swiss-data---schmid-siegert2017" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We re-analyzed mutations from <span class="citation">Schmid-Siegert <em>et al.</em> (<a href="#ref-Schmid-Siegert2017">2017</a>)</span> with <code>Strelka2</code> and <code>GATK</code> to compare original and obtained mutations to show the interest of <code>Strelka2</code> and high sequencing depth.</p>
<div class="figure"><span style="display:block;" id="fig:detectMutationsSwiss"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/swiss/dag/dag.minimal.svg" alt="Detect mutations workflow for Swiss data."  />
<p class="caption">
Figure 6.3: Detect mutations workflow for Swiss data.
</p>
</div>
</div>
<div id="bordeaux-data---plomion2018" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Bordeaux data - <span class="citation">Plomion <em>et al.</em> (<a href="#ref-Plomion2018">2018</a>)</span><a href="mutations-detection.html#bordeaux-data---plomion2018" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We re-analyzed mutations from <span class="citation">Plomion <em>et al.</em> (<a href="#ref-Plomion2018">2018</a>)</span> with <code>Strelka2</code> and <code>Mutect2</code> to compare original and obtained mutations to show the interest of <code>Strelka2</code> and high sequencing depth.</p>
<div class="figure"><span style="display:block;" id="fig:detectMutationsBdx"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/bordeaux/dag/dag.minimal.svg" alt="Detect mutations workflow for Swiss data."  />
<p class="caption">
Figure 6.4: Detect mutations workflow for Swiss data.
</p>
</div>
</div>
<div id="angela" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Angela<a href="mutations-detection.html#angela" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Genome, heterozygous sites, cambial mutations and leaf mutations from sequences of Angela.</p>
<div class="figure"><span style="display:block;" id="fig:detectMutationsAngela"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/angela/dag/dag.minimal.svg" alt="Detect mutations workflow for Angela."  />
<p class="caption">
Figure 6.5: Detect mutations workflow for Angela.
</p>
</div>
</div>
<div id="sextonia-rubra-2" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> <em>Sextonia rubra</em><a href="mutations-detection.html#sextonia-rubra-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Genome, heterozygous sites, cambial mutations and leaf mutations from sequences of Sixto</p>
<div class="figure"><span style="display:block;" id="fig:detectMutationsSixto"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/sixto/dag/dag.minimal.svg" alt="Detect mutations workflow for Sixto."  />
<p class="caption">
Figure 6.6: Detect mutations workflow for Sixto.
</p>
</div>
</div>
<div id="hetre" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Hetre<a href="mutations-detection.html#hetre" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mutations from sequences of the <a href="https://en.wikipedia.org/wiki/Faux_de_Verzy">“Faux de Verzy”</a>.</p>
<div class="figure"><span style="display:block;" id="fig:detectMutationsVerzy"></span>
<img src="https://raw.githubusercontent.com/sylvainschmitt/detectMutations/hetre/dag/dag.svg" alt="Detect mutations workflow for Angela."  />
<p class="caption">
Figure 6.7: Detect mutations workflow for Angela.
</p>
</div>
</div>
<div id="project-description-1" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Project description<a href="mutations-detection.html#project-description-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>For each species, we will extract DNA and construct 27 individually tagged genomic libraries to produce short read sequencing (see sampling strategy, WP2 above) with NovaSeq6000 SP technology, targeting aminimum of 50x coverage per library (70-90x should be realistic, we will go for 100x if possible). The possible sequencing depth affordable in the project will depend on the genome size of chosen species and some extent on fluctuating sequencing prices. Lab work will be conducted at Ecofog and PGTB (Biogeco); the sequencing will be subcontracted to CEA Genoscope, Evry.</p>
</blockquote>
<blockquote>
<p>Detection of mutations (T4.2): For mutation detection, we will minimise the impact of library preparation and sequencing errors by retaining only mutations detected by comparison with the “zero mutation reference” (Hanlon et al 2019) and in each branch (Orr et al. 2019), giving a total of 9 genotype sampling points in each tree. We will quantify the occurrence of mutations in each of the 24 leaf libraries per tree using a high-sensitivity method (Cibulskis et al., 2013). De novomutations appear in single meristem cells and their abundance in tree tissues will depend on the specific cell’s divisions and contribution to generating tree tissues. Fixation of de novo mutations in tissues is expected to be rare and the expected time to fixation is long, see (Nicholson et al., 2018)for an example in humans. The detection of novel mutations must thus be sensitive to low allele copy number in tissues. For this reason, we will test and use mutation detection methods typically used in cancer mutation research, that are optimized for strong allelic frequency skews (Alioto et al., 2015).The artificial inclusion of in silicomutationsinto the reads and permutations among branch labels allows the estimation of the recovery rate of true mutations, the false negative rates, and the false positive rates (Orr et al., 2019). We will test whether the rate of accumulation of mutations is increased in high-light-exposed branches compared to those with low-light exposure accounting for the physical tree branches (tree architecture -WP3), the mutation history, and light environment (WP3).</p>
</blockquote>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Plomion2018" class="csl-entry">
Plomion, C., Aury, J.M., Amselem, J., Leroy, T., Murat, F., Duplessis, S., Faye, S., Francillonne, N., Labadie, K., Provost, G.L., Lesur, I., Bartholomé, J., Faivre-Rampant, P., Kohler, A., Leplé, J.C., Chantret, N., Chen, J., Diévart, A., Alaeitabar, T., Barbe, V., Belser, C., Bergès, H., Bodénès, C., Bogeat-Triboulot, M.B., Bouffaud, M.L., Brachi, B., Chancerel, E., Cohen, D., Couloux, A., Silva, C.D., Dossat, C., Ehrenmann, F., Gaspin, C., Grima-Pettenati, J., Guichoux, E., Hecker, A., Herrmann, S., Hugueney, P., Hummel, I., Klopp, C., Lalanne, C., Lascoux, M., Lasserre, E., Lemainque, A., Desprez-Loustau, M.L., Luyten, I., Madoui, M.A., Mangenot, S., Marchal, C., Maumus, F., Mercier, J., Michotey, C., Panaud, O., Picault, N., Rouhier, N., Rué, O., Rustenholz, C., Salin, F., Soler, M., Tarkka, M., Velt, A., Zanne, A.E., Martin, F., Wincker, P., Quesneville, H., Kremer, A. &amp; Salse, J. (2018). Oak genome reveals facets of long lifespan. <em>Nature Plants</em>, <strong>4</strong>, 440–452. Retrieved from <a href="http://dx.doi.org/10.1038/s41477-018-0172-3">http://dx.doi.org/10.1038/s41477-018-0172-3</a>
</div>
<div id="ref-Schmid-Siegert2017" class="csl-entry">
Schmid-Siegert, E., Sarkar, N., Iseli, C., Calderon, S., Gouhier-Darimont, C., Chrast, J., Cattaneo, P., Schütz, F., Farinelli, L., Pagni, M., Schneider, M., Voumard, J., Jaboyedoff, M., Fankhauser, C., Hardtke, C.S., Keller, L., Pannell, J.R., Reymond, A., Robinson-Rechavi, M., Xenarios, I. &amp; Reymond, P. (2017). Low number of fixed somatic mutations in a long-lived oak tree. <em>Nature Plants</em>, <strong>3</strong>, 926–929. Retrieved from <a href="http://dx.doi.org/10.1038/s41477-017-0066-9">http://dx.doi.org/10.1038/s41477-017-0066-9</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genome-assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mutations-in-silico.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
